<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-09-05" />

<title>Analysis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-09-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree54fa9d53fcf21807e7b992f7bc59e68f4f5173a7targetblank54fa9d5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/54fa9d53fcf21807e7b992f7bc59e68f4f5173a7" target="_blank">54fa9d5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree54fa9d53fcf21807e7b992f7bc59e68f4f5173a7targetblank54fa9d5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/54fa9d53fcf21807e7b992f7bc59e68f4f5173a7" target="_blank">54fa9d5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/Formatted_Candidate_ETG_20200305_TAD10Kb.tsv
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Ignored:    data/single_replicate_peaks/.DS_Store
    Ignored:    data/single_replicate_peaks/control/
    Ignored:    data/single_replicate_peaks/dex12h/
    Ignored:    data/single_replicate_peaks/dex2h/
    Ignored:    data/single_replicate_peaks/peaks/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  MR-paper.pdf
    Untracked:  data/reddy_bed_JUNB_DEX2h_r23.bed.gz

Unstaged changes:
    Modified:   analysis/01_Results.Rmd
    Modified:   analysis/Q2-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/Q4-GR-binding-PROTAC-genes.Rmd
    Modified:   analysis/Q9-Differential_binding.Rmd
    Modified:   analysis/admin.Rmd
    Modified:   data/results/differential_binding_results.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/01_Results.Rmd</code>) and HTML (<code>docs/01_Results.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/54fa9d53fcf21807e7b992f7bc59e68f4f5173a7/analysis/01_Results.Rmd" target="_blank">54fa9d5</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/54fa9d53fcf21807e7b992f7bc59e68f4f5173a7/docs/01_Results.html" target="_blank">54fa9d5</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/analysis/01_Results.Rmd" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/01_Results.html" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>The overall goal is to (1)understand the effect of activated glucocorticoid receptors on gene regulation and (2) relate the epigenetic data from the Reddy lab with our expression data. To this end Mahshid and developed seven initial questions which define the primary scope of the analysis. In a discussion on these results further questions arised which are answered further below. As some of these questions/answers relate to results obtained in other work by Katharina, Mahshid and Pierre-Luc the flow of questions may not be fully coherent in themselves.</p>
</div>
<div id="question-1-similarity-of-our-rna-data-with-the-reddy-data" class="section level1">
<h1>Question 1: Similarity of our RNA data with the Reddy data</h1>
<p><strong>Q: Which gene expressions have similar patterns of up-regulation (control vs. Dex) in our data compared to the Reddy data?</strong></p>
<p>In our experiments we compared samples from a control experiment with samples that were treated for 18 hours with DEX. Besides a control, the Reddy lab performed a similar experiment where they treated cells with DEX for 12 hours. This is the closest comparison to our data.</p>
<p>As a first step we hence want to understand whether the results are similar/comparable.</p>
<div id="comparison-via-venn-diagram" class="section level2">
<h2>Comparison via Venn diagram</h2>
<pre class="r"><code>q1_data &lt;- readRDS(file=&quot;../data/EOS_files/Q1_EOS.rds&quot;)

n_genes_plot &lt;- dim(q1_data[[&quot;logFC_plot_data&quot;]])[1]
n_genes_union &lt;- length(q1_data[[&quot;union_DE_genes&quot;]])</code></pre>
<p>We first inspect a Venn diagram to understand the overlap of differentially expressed genes from both datasets. <strong>We see that a large part of the genes do overlap, while there are also many genes that are specific to the respective analyses.</strong> However, this is only a binary comparison and so a we next take a look at how the exact log-fold changes compare.</p>
<p><img src="assets/1_venn_diagram.png" /></p>
</div>
<div id="comparison-via-log-fold-changes" class="section level2">
<h2>Comparison via log-fold changes</h2>
<p>Second, we compare the log fold changes for the union of all genes identified as significantly differentially expressed. The union of all DE genes counts 8121 genes of which 7342 genes (90.41%) were also in the intersection of both datasets and could be plotted.</p>
<p><strong>We see that the results of our differential expression (DE) analysis from the 18 DEX treatment compare well with the 12h DEX treatment DE results of the Reddy lab.</strong></p>
<p><img src="assets/2_logFC_Mahshid_Reddy.png" /></p>
</div>
</div>
<div id="question-2-binding-of-gr-gc-complex-to-de-genes-from-our-analysis" class="section level1 tabset">
<h1 class="tabset">Question 2: Binding of GR-GC complex to DE genes from our analysis</h1>
<p><strong>Q: For the genes that our RNA-seq data shows to be DE after treatment with DEX as well as the inhibitors and control: do we see any direct TF binding to these sites? For which of these is there a difference compared to the control ChIP-seq experiment?</strong></p>
<p><strong>Take-away:</strong> Genes which are differentially expressed at a significant level are associated with GR peaks close to them. Genes with GR binding close to them are also associated with lower p-values in the DE analysis. A higher number of GR peaks close to the TSS of a gene is associated with a higher absolute log-fold change in gene expression. Genes where GR binds close to or directly at the TSS are associated with the higher absolute log-fold changes.</p>
<p>A table with the DE results from our 18h DEX treatment which is augmented by information on peaks identified nearby can be found in the downloads section.</p>
<p>An analysis on peaks which are known to interact with distant genes (long-range interactions) can be found further below in this file.</p>
<div id="distribution-of-logfc-and-p.value" class="section level2">
<h2>Distribution of logFC and p.value</h2>
<p><strong>What does this plot show?</strong></p>
<p>We select all genes that are differentially expressed - both for Mahshid’s data and for the DE analysis on the Reddy data. We then split the data into two groups based on whether the Reddy data showed that the glucocorticoid receptor was found to bind near the gene or not. We plot the distributions of (a) the log fold changes in the DE analysis and (b) the p-value of the DE analysis.</p>
<p><img src="assets/Q2_3_mahshid_reddy.png" /></p>
</div>
<div id="boxplots-by-gr-peaks-number" class="section level2">
<h2>Boxplots by GR peaks number</h2>
<p><img src="assets/Q2_4_boxplots.png" /></p>
</div>
<div id="boxplots-by-type-of-closest-gr-peak" class="section level2">
<h2>Boxplots by type of closest GR peak</h2>
<p><em>Note:</em> NA means that there was no peak for which the gene’s TSS was the TSS closest to the peak.</p>
<p><img src="assets/Q2_6_boxplots_class.png" /></p>
</div>
<div id="logfc-vs.-distance-to-closest-peak" class="section level2">
<h2>logFC vs. distance to closest peak</h2>
<p><img src="assets/Q2_5_TSS_distances.png" /></p>
</div>
</div>
<div id="question-3-inhibition-unique-to-protac-compared-to-cort-and-mif" class="section level1 tabset">
<h1 class="tabset">Question 3: Inhibition unique to protac compared to Cort and MIF</h1>
<p><strong>Q: Are there any genes that treated by the Protac before/after DEX that could be reversed but which could not by the inhibitors or vice versa?</strong></p>
<p>We have performed experiments where we first exposed cells to either (1) DEX (2) inhibitors or (3) KH PROTAC for a substantial time and then added inhibitors or the KH PROTAC in case (1) or DEX in cases (2) and (3). We would like to see which genes are normally activated by DEX but where the effect could be blocked by the PROTAC, but NOT the inhibitors.</p>
<p>The plot for the 2h&gt;16h treatment has been split into (1) genes with a positive log-fold change and (2) genes with a negative log-fold change to prevent the plots from becoming even more crowded.</p>
<p>A file with the list of genes can be found in the downloads section.</p>
<div id="h-16h" class="section level2">
<h2>2h &gt; 16h</h2>
<p><img src="assets/Q3_1_reversal216.png" /></p>
</div>
<div id="h-2h-logfc-0" class="section level2">
<h2>16h &gt; 2h (logFC &gt; 0)</h2>
<p><img src="assets/Q3_2_reversal162_p1.png" /></p>
</div>
<div id="h-2h-logfc-0-1" class="section level2">
<h2>16h &gt; 2h (logFC &lt; 0)</h2>
<p><img src="assets/Q3_3_reversal162_p2.png" /></p>
</div>
</div>
<div id="question-4-does-the-gr-bind-close-to-any-of-the-13-de-genes-from-the-18h-kh-protac-treatment" class="section level1 tabset">
<h1 class="tabset">Question 4: Does the GR bind close to any of the 13 DE genes from the 18h KH-PROTAC treatment?</h1>
<p>Cells that were exposed to KH-103 for 18h showed 13 genes to be differentially expressed at an &lt;0.05 FDR level. We want to understand whether GR binds close to these genes in a case where such cells are treated with DEX.</p>
<p>The Reddy lab has collected data on where the glucocorticoid receptor (NR3C1) binds to after treatment with dexamethasone for 12h.</p>
<p><strong>Take-away:</strong> We find that the GR is found to bind close to five out of the 13 differentially expressed genes from our experiment. Interestingly, we also observe that some GR peaks are stronger after 2 hours of DEX treatment than 12 hours of continuous DEX treatment.</p>
<div id="table-view" class="section level2">
<h2>Table view</h2>
<pre class="r"><code>kh.protac &lt;- read.xlsx(&quot;../docs/downloads/Q4_KH_DE_GR_bound.xlsx&quot;)

# only keep the significant genes
kh.protac.sig &lt;- kh.protac[kh.protac$FDR&lt;0.05,]
kh.protac.sig[,c(&quot;Gene&quot;, &quot;logFC&quot;, &quot;FDR&quot;, &quot;chip_peaks&quot;, &quot;gc_bound&quot;)] %&gt;%
  kableExtra::kable() %&gt;%
  kableExtra::kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
chip_peaks
</th>
<th style="text-align:left;">
gc_bound
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PER1
</td>
<td style="text-align:right;">
1.1428657
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
TGFBR1
</td>
<td style="text-align:right;">
0.5510412
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
FKBP5
</td>
<td style="text-align:right;">
0.7301845
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
NBEAL2
</td>
<td style="text-align:right;">
0.3850139
</td>
<td style="text-align:right;">
0.0001572
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
CAVIN2
</td>
<td style="text-align:right;">
0.3182665
</td>
<td style="text-align:right;">
0.0006525
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
SLC38A2
</td>
<td style="text-align:right;">
0.2546515
</td>
<td style="text-align:right;">
0.0007788
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
TEF
</td>
<td style="text-align:right;">
-0.9140413
</td>
<td style="text-align:right;">
0.0010003
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
CRY1
</td>
<td style="text-align:right;">
-0.3858283
</td>
<td style="text-align:right;">
0.0021419
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
C5AR2
</td>
<td style="text-align:right;">
0.6589461
</td>
<td style="text-align:right;">
0.0039443
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
ANGPTL4
</td>
<td style="text-align:right;">
0.4185862
</td>
<td style="text-align:right;">
0.0064158
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
CRY2
</td>
<td style="text-align:right;">
-0.4867045
</td>
<td style="text-align:right;">
0.0097262
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
CHML
</td>
<td style="text-align:right;">
0.2746874
</td>
<td style="text-align:right;">
0.0135275
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
PER3
</td>
<td style="text-align:right;">
-0.7575232
</td>
<td style="text-align:right;">
0.0398908
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
</tbody>
</table>
<p><img src="assets/Q4_1_kh_gr_binding.png" /> ## p-value vs. logFC</p>
</div>
<div id="signal-tracks-for-peaks-that-could-be-matched-with-the-13-genes" class="section level2">
<h2>Signal tracks for peaks that could be matched with the 13 genes</h2>
<p>We plot peaks for which the closest TSS was found to be one of the 13 genes. The tracks of the peaks shown are <em>centered at these peaks</em> and <em>not</em> at the TSS of the respective gene.</p>
<p><img src="assets/Q4_2_kh_peak_tracks.png" /></p>
</div>
<div id="tracks-around-the-all-tsss-of-the-13-genes" class="section level2">
<h2>Tracks around the all TSSs of the 13 genes</h2>
<p>Here we first obtain all known transcripts of the 13 genes and plot the signals <em>centered around the respective TSS sites</em>. As a gene can have multiple TSSs this plot is a bit crowded.</p>
<p><img src="assets/Q4_3_kh_all_txs.png" /></p>
</div>
<div id="tracks-around-main-tss-for-each-of-the-13-genes" class="section level2">
<h2>Tracks around main TSS for each of the 13 genes</h2>
<p>To simplify the previous plot (all TSSs) we just choose a single TSS per gene (the first one in the list) and plot the signal around this TSS.</p>
<p><img src="assets/Q4_4_kh_unique_gene_tss.png" /></p>
</div>
</div>
<div id="question-5-binding-to-selected-genes" class="section level1">
<h1>Question 5: Binding to selected genes</h1>
<p><strong>Q: Does GR bind to BRCA1, Hsd11b1 and CH25H?</strong></p>
<p>These are specific genes requested to be examined by Mahshid. The background of the biological question is not known to me.</p>
<p><strong>Take-away:</strong> We see that there is one peak in the ChIP data close to the BRCA1 gene and two peaks close to the HSD11B1 gene. There is no peak that is closest to the CH25H gene. However, the CD25H was also not found in the RNA seq data.</p>
<pre class="r"><code>res_int &lt;- read.xlsx(&quot;../docs/downloads/Q2_GR_bound_DE_genes.xlsx&quot;)
plt_data &lt;- res_int[res_int$Gene %in% c(&quot;BRCA1&quot;, &quot;HSD11B1&quot;, &quot;CH25H&quot;),c(&quot;Gene&quot;, &quot;logFC&quot;, &quot;FDR&quot;, &quot;chip_peaks&quot;, &quot;gc_bound&quot;)]
plt_data %&gt;%
  ggplot() +
  geom_point(aes(logFC, -log10(FDR), colour=gc_bound, size=chip_peaks)) +
  labs(title=&quot;GR binding activity for diff. expressed PROTAC genes&quot;) +
  ggrepel::geom_text_repel(data=plt_data, aes(logFC, -log10(FDR), label=Gene, size=2)) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/01_Results.Rmd/-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig--1-1">
Past versions of -1-1.png
</button>
</p>
<div id="fig--1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/-1-1.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>CH25H was not found. Is it in the RNA-seq experiment?</p>
<pre class="r"><code>se &lt;- readRDS(&quot;../data/EOS_files/Q1_SummExp.rds&quot;)
&quot;ENSG00000138135&quot; %in% rowData(se)$gene_id</code></pre>
<pre><code>[1] FALSE</code></pre>
<p>No, it is not in the RNA-seq experiment.</p>
<p>We also check whether the gene is present in the table with the peaks. If the gene were to be missing in our DE data then a given peak would not show up in the results above, as it could not have been merged onto the DE results.</p>
<p>We search for the closest match to be sure that its not due to capitalisation or a small typo</p>
<pre class="r"><code># We also have to check the other way around by looking at all the peaks.
# If it was in the peaks then it could not be merged onto the RNA data
peakfreq &lt;- readRDS(file=&quot;../data/EOS_files/Q2_gene_peaks.rds&quot;)
peakfreq &lt;- peakfreq[2:dim(peakfreq)[1],] # first entry is empty

# we search for the closest match to be sure that its not due to capitalisation or a small typo
peakfreq$Gene[amatch(&quot;CH25H&quot;, peakfreq$Gene, maxDist = Inf)]</code></pre>
<pre><code>[1] C5
2939 Levels:  AACS AADAC AADACL2-AS1 AADACP1 ABCA1 ABCA4 ABCA5 ABCC1 ... ZYG11B</code></pre>
<p><strong>Result:</strong> there is no match for CH25h</p>
<p>Lastly, we show the information we have for the two genes found in the data:</p>
<pre class="r"><code>plt_data %&gt;%
  kableExtra::kable() %&gt;%
  kableExtra::kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
chip_peaks
</th>
<th style="text-align:left;">
gc_bound
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
118
</td>
<td style="text-align:left;">
HSD11B1
</td>
<td style="text-align:right;">
6.0333836
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
210
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:right;">
-0.9559991
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
</tbody>
</table>
</div>
<div id="question-6-common-binding-sites-of-junb-and-gr" class="section level1">
<h1>Question 6: Common binding sites of JUNB and GR</h1>
<p><strong>Q: What are common binding sites between JUNB and GR?</strong> -&gt; Generally not a priority though</p>
<p><strong>Take-aways:</strong></p>
<ul>
<li>Interestingly there are more peaks after 2h treatment with DEX than 12h of treatment with DEX</li>
<li>Most of the 12h GR peaks are shared with the JUNB peaks, only a small fraction is unique to the 12h GR ChIP seq data</li>
<li>The 2h GR data has more peaks that are unique and not shared with the 12h GR or 12h JUNB data</li>
</ul>
<div id="unfiltered" class="section level2 tabset">
<h2 class="tabset">Unfiltered</h2>
<pre class="r"><code># Load GR
peaks_gr_12h &lt;- import.bed15(&quot;../data/reddy_bed_DEX12h_r123.bed&quot;, format=&quot;narrowPeak&quot;)
peaks_gr_2h &lt;- import.bed15(&quot;../data/reddy_bed_DEX2h_r123.bed.gz&quot;, format=&quot;narrowPeak&quot;)
# load JUNB
peaks_junb_2h &lt;- import.bed15(&quot;../data/reddy_bed_JUNB_DEX2h_r23.bed.gz&quot;, format=&quot;narrowPeak&quot;)
peaks_junb_12h &lt;- import.bed15(&quot;../data/reddy_bed_JUNB_DEX12h_r123.bed&quot;, format=&quot;narrowPeak&quot;)

# Show tile diagram
list_of_regions &lt;- list(GR_2h=peaks_gr_2h, GR_12h=peaks_gr_12h, JUNB_12h=peaks_junb_12h) # Does not work with only 
cat(&quot;\n\n### Overlap \n\n&quot;)</code></pre>
<div id="overlap" class="section level3">
<h3>Overlap</h3>
<pre class="r"><code>regionOverlaps(list_of_regions)</code></pre>
<p><img src="figure/01_Results.Rmd/q6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-q6-1">
Past versions of q6-1.png
</button>
</p>
<div id="fig-q6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/q6-1.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Show upset plot
cat(&quot;\n\n### Upset plot \n\n&quot;)</code></pre>
</div>
<div id="upset-plot" class="section level3">
<h3>Upset plot</h3>
<pre class="r"><code>regionUpset(list_of_regions)</code></pre>
<p><img src="figure/01_Results.Rmd/q6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-q6-2">
Past versions of q6-2.png
</button>
</p>
<div id="fig-q6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/q6-2.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="what-if-we-filter-the-peaks-by-their-score" class="section level2 tabset">
<h2 class="tabset">What if we filter the peaks by their score?</h2>
<p><strong>Take-away:</strong> There is no significant change in the results/plots if we filter for a minimum peak score of 900 (max value is 1000).</p>
<pre class="r"><code># Does it make a difference if we only choose peaks with high scores?
## What is the range of scores
# qplot(peaks_gr_2h$score, geom=&quot;histogram&quot;)

# Show tile diagram
list_of_regions_v2 &lt;- list(GR_2h=peaks_gr_2h[peaks_gr_2h$score&gt;900], 
                           GR_12h=peaks_gr_12h[peaks_gr_12h$score&gt;900], 
                           JUNB_12h=peaks_junb_12h[peaks_junb_12h$score&gt;900]) # Does not work with only two 
cat(&quot;\n\n### Overlap \n\n&quot;)</code></pre>
<div id="overlap-1" class="section level3">
<h3>Overlap</h3>
<pre class="r"><code>regionOverlaps(list_of_regions_v2)</code></pre>
<p><img src="figure/01_Results.Rmd/q6%20filtered-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-q6-filtered-1">
Past versions of “q6 filtered-1.png”
</button>
</p>
<div id="fig-no-spaces-q6-filtered-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/q6 filtered-1.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Show upset plot
cat(&quot;\n\n### Upset plot \n\n&quot;)</code></pre>
</div>
<div id="upset-plot-1" class="section level3">
<h3>Upset plot</h3>
<pre class="r"><code>regionUpset(list_of_regions_v2)</code></pre>
<p><img src="figure/01_Results.Rmd/q6%20filtered-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-q6-filtered-2">
Past versions of “q6 filtered-2.png”
</button>
</p>
<div id="fig-no-spaces-q6-filtered-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/q6 filtered-2.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="which-genes-have-a-gr-peak-and-a-junb-peak-within-2500bp-distance-to-the-tss" class="section level2 tabset">
<h2 class="tabset">Which genes have a GR peak AND a JUNB peak within 2500bp distance to the TSS?</h2>
<p>Annotate GR 2h and 12h and JUNB 2h</p>
<pre class="r"><code>library(AnnotationHub)</code></pre>
<pre><code>Warning: package &#39;AnnotationHub&#39; was built under R version 4.1.2</code></pre>
<pre><code>Loading required package: BiocFileCache</code></pre>
<pre><code>Warning: package &#39;BiocFileCache&#39; was built under R version 4.1.2</code></pre>
<pre><code>Loading required package: dbplyr</code></pre>
<pre><code>Warning: package &#39;dbplyr&#39; was built under R version 4.1.2</code></pre>
<pre><code>
Attaching package: &#39;dbplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    ident, sql</code></pre>
<pre><code>
Attaching package: &#39;AnnotationHub&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:rtracklayer&#39;:

    hubUrl</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    cache</code></pre>
<pre class="r"><code>ah &lt;- AnnotationHub()</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre class="r"><code>ensdb &lt;- ah[[&quot;AH98047&quot;]] # AH89211</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>require(&quot;ensembldb&quot;)</code></pre>
<pre><code>Warning: package &#39;GenomicFeatures&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;AnnotationFilter&#39; was built under R version 4.1.1</code></pre>
<pre class="r"><code>gr_peaks_annotated_2h &lt;- annotateRegions(peaks_gr_2h, ensdb)</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre class="r"><code>gr_peaks_annotated_12h &lt;- readRDS(&quot;../data/EOS_files/Q2_annotated_peaks.rds&quot;)
JUNB_peaks_annotated_2h &lt;- annotateRegions(peaks_junb_2h, ensdb)</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre class="r"><code>JUNB_peaks_annotated_12h &lt;- annotateRegions(peaks_junb_12h, ensdb)</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre class="r"><code>### filter for 2500 bp distance to gene TSS
# GR 2h
gr_peaks_annotated_2h$nearestTSS.gene_name.2500r &lt;- gr_peaks_annotated_2h$nearestTSS.gene_name
gr_peaks_annotated_2h[abs(gr_peaks_annotated_2h$distance2nearestTSS) &gt;= 2500, ]$nearestTSS.gene_name.2500r &lt;- &quot;&quot;
# JUNB 2h 
JUNB_peaks_annotated_2h$nearestTSS.gene_name.2500r &lt;- JUNB_peaks_annotated_2h$nearestTSS.gene_name
JUNB_peaks_annotated_2h[abs(JUNB_peaks_annotated_2h$distance2nearestTSS) &gt;= 2500, ]$nearestTSS.gene_name.2500r &lt;- &quot;&quot;
# JUNB 12h 
JUNB_peaks_annotated_12h$nearestTSS.gene_name.2500r &lt;- JUNB_peaks_annotated_12h$nearestTSS.gene_name
JUNB_peaks_annotated_12h[abs(JUNB_peaks_annotated_12h$distance2nearestTSS) &gt;= 2500, ]$nearestTSS.gene_name.2500r &lt;- &quot;&quot;

# subset
gr_2h_subset &lt;- gr_peaks_annotated_2h[gr_peaks_annotated_2h$nearestTSS.gene_name.2500r != &quot;&quot;,]
gr_12h_subset &lt;- gr_peaks_annotated_12h[gr_peaks_annotated_12h$nearestTSS.gene_name.2500r != &quot;&quot;,]
junb_2h_subset &lt;- JUNB_peaks_annotated_2h[JUNB_peaks_annotated_2h$nearestTSS.gene_name.2500r != &quot;&quot;,]
junb_12h_subset &lt;- JUNB_peaks_annotated_12h[JUNB_peaks_annotated_12h$nearestTSS.gene_name.2500r != &quot;&quot;,]

list_of_regions_v3 &lt;- list(GR_2h=gr_2h_subset, 
                           GR_12h=gr_12h_subset,
                           JUNB_2h=junb_2h_subset,
                           JUNB_12h=junb_12h_subset)

cat(&quot;\n\n### Overlap \n\n&quot;)</code></pre>
<div id="overlap-2" class="section level3">
<h3>Overlap</h3>
<pre class="r"><code>regionOverlaps(list_of_regions_v3)</code></pre>
<p><img src="figure/01_Results.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/unnamed-chunk-3-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Show upset plot
cat(&quot;\n\n### Upset plot \n\n&quot;)</code></pre>
</div>
<div id="upset-plot-2" class="section level3">
<h3>Upset plot</h3>
<pre class="r"><code>regionUpset(list_of_regions_v3)</code></pre>
<p><img src="figure/01_Results.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/unnamed-chunk-3-2.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="visualise-the-genes-that-have-both-a-junb-and-gr-peak-within-2500bp-of-their-tss" class="section level4">
<h4>Visualise the genes that have both a JUNB and GR peak within 2500bp of their TSS</h4>
<pre class="r"><code>gr_junb_2h_intersection_genes &lt;- intersect(gr_2h_subset$nearestTSS.gene_name.2500r, junb_2h_subset$nearestTSS.gene_name.2500r)

# cat(gr_junb_2h_intersection_genes)

length(gr_junb_2h_intersection_genes)</code></pre>
<pre><code>[1] 1139</code></pre>
<p>There is an overlap of 1139 genes.</p>
</div>
</div>
</div>
</div>
<div id="question-7-pxr-sequences" class="section level1">
<h1>Question 7: PXR Sequences</h1>
<p><strong>Q: Does GC-GR bind to PXR sequences?</strong></p>
<p>Other groups have observed that a GR inhibitor activates CYP genes. CYP genes are target genes of PXR TF that have PXR recognition sites. the hypothesis here is that GR might also bind to PXR sites directly or probably is regulating CYP genes via interaction with PXR TF on these genes. For example, based on our RNA-seq dex triggers many CYP genes and the inhibitors too, but protac doesn’t.</p>
<p>The PXR motif looks like this (from Hocomoco):</p>
<p><img src="assets/Q7_1_motif.png" /></p>
<pre class="r"><code>gr_peaks &lt;- readRDS(&quot;../data/EOS_files/Q2_annotated_peaks.rds&quot;)
moi &lt;- readRDS(file=&quot;../data/EOS_files/pxr_peaks.rds&quot;)

# cat(paste0(&quot;We have &quot;, length(gr_peaks), &quot; GR peaks, of which &quot;, length(moi), &quot; (&quot;, round(length(moi)/length(gr_peaks),4)*100, &quot;%) contain a PXR motif&quot;))</code></pre>
<div id="we-have-7920-gr-peaks-of-which-238-3.01-contain-a-pxr-motif" class="section level3">
<h3>We have 7920 GR peaks, of which 238 (3.01%) contain a PXR motif</h3>
<p>We can also look at the signals around the PXR motifs found in the accessible chromatin (identified using the 12h ATAC seq data from the Reddy lab)</p>
<p><img src="assets/Q7_2_signal.png" /></p>
</div>
</div>
<div id="q8-how-do-our-signals-compare-to-our-differential-expression-results" class="section level1">
<h1>Q8: How do our signals compare to our differential expression results?</h1>
<p>In the following we plot the signal for a subset of genes, IF these genes are associated with a peak the in the GR data for the 12h DEX ChIP data.</p>
<div id="signals-and-de-results-for-all-genes-with-associated-peaks" class="section level3">
<h3>Signals and DE results for all genes with associated peaks</h3>
<p>We first plot the signal tracks for all genes for the sake of completeness. We plot around the TSS of the gene, not around the centre of the peak. Reminder: The peaks are labelled with the gene name the TSS of which is closest to the peak.</p>
<pre class="r"><code># load summarised experiment
se &lt;- readRDS(file=&quot;../data/results/SE.processed.rds&quot;)
# removes the row without name
se &lt;- se[2:dim(se)[1]]
res_int_all &lt;- readRDS(file=&quot;../data/results/DE_our_data.rds&quot;)</code></pre>
<pre class="r"><code>dds &lt;- readRDS(&quot;../data/results/dds.rds&quot;)

tracks &lt;-  list(&quot;2h-DEX&quot;=&quot;../data/reddy_bigwig_DEX2h_r123.bigWig&quot;,
                &quot;6h-DEX&quot;=&quot;../data/reddy_bigwig_DEX6h_r123.bigWig&quot;,
                &quot;12h-DEX&quot;=&quot;../data/reddy_bigwig_DEX12h_r123.bigWig&quot;)

plot_signals_and_sechm &lt;- function(peaks, genes_subset=NULL, plot_around_gene_tss=FALSE){
  # first subset for a given selection of genes
  if (is.null(genes_subset) == FALSE) {
    peaks &lt;- peaks[peaks$nearestTSS.gene_name %in% genes_subset]
    if (length(peaks) == 0) return(paste0(&quot;No peaks found for the genes:&quot;, paste(genes_subset, collapse = &#39;, &#39;)))
  }
  
  # get rid of peaks that don&#39;t have an assigned gene that was quantified
  peaks &lt;- sort(peaks[peaks$nearestTSS.gene_name %in% row.names(se)])
  # give unique names to the peaks
  names(peaks) &lt;- make.unique(peaks$nearestTSS.gene_name)
  # select (and duplicate) the entries in R, and give them the names
  se2 &lt;- se[as.character(peaks$nearestTSS.gene_name),]
  row.names(se2) &lt;- names(peaks)
  
  if (plot_around_gene_tss == TRUE){
    ah &lt;- AnnotationHub()
    ensdb &lt;- ah[[&quot;AH98047&quot;]] # AH89211

    mygenes.transcripts &lt;-  transcripts(ensdb, columns=c(&quot;tx_id&quot;, &quot;tx_name&quot;, &quot;tx_biotype&quot;,
                                                                &quot;tx_support_level&quot;,
                                                                &quot;gene_id&quot;,&quot;gene_name&quot;))
    mygenes.transcripts &lt;- subset(mygenes.transcripts, gene_name %in% peaks$nearestTSS.gene_name)
    mygenes.tss &lt;-  resize(mygenes.transcripts, width=2000, fix=&#39;start&#39;)
    seqlevelsStyle(mygenes.tss) &lt;- &quot;UCSC&quot;
    mygenes_unique &lt;- mygenes.tss[!duplicated(mygenes.tss$gene_name),]
    
    # adjust the se2 genes to only show the ones for which we have the TSS
    genes &lt;- mygenes_unique$gene_name
    se2 &lt;- se[genes,]
    # get the signals
    signals &lt;- signal2Matrix(tracks, mygenes_unique, w=5, extend=1000)
  } 
  else
  {
    signals &lt;- signal2Matrix(tracks, peaks, w=5, extend=500)  
  }
  
  
  h1 &lt;- plotEnrichedHeatmaps(signals, width=unit(8, &quot;cm&quot;))
  set.seed(100) # for colours
  h2 &lt;- sechm::sechm(se2, genes=row.names(se2), assayName=&quot;log2FC&quot;, sortRowsOn=NULL, 
                     cluster_rows=FALSE, width=unit(12, &quot;cm&quot;), top_annotation=c(&quot;condition&quot;,&quot;batch&quot;), 
                     gaps_at=&quot;condition&quot;, column_title_gp=gpar(fontsize=10), column_title_rot=90)
  # (for this second one it&#39;s important that rows are in the input order)
  h1 + h2
}

plt &lt;- plot_signals_and_sechm(gr_peaks, plot_around_gene_tss=TRUE)</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>plt</code></pre>
<p><img src="figure/01_Results.Rmd/unnamed-chunk-6-1.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/unnamed-chunk-6-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/unnamed-chunk-6-1.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="signaltracks-de-plot-for-side-effects-of-the-inhibitors" class="section level2 tabset">
<h2 class="tabset">Signaltracks + DE plot for Side effects of the inhibitors</h2>
<p>In the following we show the same plots, but for smaller subsets of genes. We choose at most the 200 most significantly differentially expressed genes from a given comparison named below.</p>
<p>The signals are plotted around the gene TSS ± 1000bp.</p>
<pre class="r"><code>se.processed &lt;- readRDS(file=&quot;../data/results/SE.processed.rds&quot;)

cat(&quot;\n\n### Genes triggered by all inhibitors \n\n&quot;)</code></pre>
<div id="genes-triggered-by-all-inhibitors" class="section level3">
<h3>Genes triggered by all inhibitors</h3>
<pre class="r"><code># Genes triggered by all inhibitors
w &lt;- se$condition2 %in% c(&quot;control&quot;,&quot;18h Cort113&quot;,&quot;18h KH-103&quot;,&quot;18h MIF&quot;)
dds2 &lt;- dds[,w]
mm &lt;- model.matrix(~se$isInhibited[w])
fit &lt;- glmFit(dds2,mm)
res &lt;- as.data.frame(topTags(glmLRT(fit),Inf))
rnames_all_inhibs &lt;- head(row.names(res)[res$FDR&lt;0.05],200)
plt_all_inhibs &lt;- plot_signals_and_sechm(gr_peaks, genes_subset=rnames_all_inhibs, plot_around_gene_tss=TRUE)</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>plt_all_inhibs</code></pre>
<p><img src="figure/01_Results.Rmd/unnamed-chunk-7-1.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/unnamed-chunk-7-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n### Genes specifically altered by KH-103 \n\n&quot;)</code></pre>
</div>
<div id="genes-specifically-altered-by-kh-103" class="section level3">
<h3>Genes specifically altered by KH-103</h3>
<pre class="r"><code>## Genes specifically altered by KH-103
isKh &lt;- se$condition[w]==&quot;18h KH-103&quot;
mm &lt;- model.matrix(~isKh)
fit &lt;- glmFit(estimateDisp(dds2,mm),mm)
res &lt;- as.data.frame(topTags(glmLRT(fit, &quot;isKhTRUE&quot;),Inf))
rnames_kh103 &lt;- row.names(res)[res$FDR&lt;0.05]
plt_kh103  &lt;- plot_signals_and_sechm(gr_peaks, genes_subset=rnames_kh103, plot_around_gene_tss=TRUE)
plt_kh103</code></pre>
<p>[1] “No peaks found for the genes:TEF, CRY2”</p>
<pre class="r"><code>cat(&quot;\n\n### Genes triggered by other inhibitors and not (or less) KH-103 \n\n&quot;)</code></pre>
</div>
<div id="genes-triggered-by-other-inhibitors-and-not-or-less-kh-103" class="section level3">
<h3>Genes triggered by other inhibitors and not (or less) KH-103</h3>
<pre class="r"><code>## Genes triggered by other inhibitors and not (or less) KH-103
mm &lt;- model.matrix(~se$isInhibited[w]+isKh)
fit &lt;- glmFit(estimateDisp(dds2,mm),mm)
res &lt;- as.data.frame(topTags(glmLRT(fit, contrast=c(0,1,-1)),Inf))
rnames_inhibs_not_kh &lt;- head(row.names(res)[res$FDR&lt;0.05],200)
plt_inhibs_not_kh  &lt;- plot_signals_and_sechm(gr_peaks, genes_subset=rnames_inhibs_not_kh, plot_around_gene_tss=TRUE)</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>plt_inhibs_not_kh</code></pre>
<p><img src="figure/01_Results.Rmd/unnamed-chunk-7-2.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/unnamed-chunk-7-2.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="signaltracks-de-plot-for-inhibition-effects" class="section level2 tabset">
<h2 class="tabset">Signaltracks + DE plot for inhibition effects</h2>
<p>The signals are plotted around the gene TSS ± 1000bp.</p>
<pre class="r"><code>cat(&quot;\n\n### DEX-triggered genes that are inhibited (prior) \n\n&quot;)</code></pre>
<div id="dex-triggered-genes-that-are-inhibited-prior" class="section level3">
<h3>DEX-triggered genes that are inhibited (prior)</h3>
<pre class="r"><code>## DEX-triggered genes that are inhibited (prior)
wPrior &lt;- which(se$condition2 %in% c(&quot;control&quot;, &quot;16h Cort113 &gt; 2h DEX+Cort113&quot;, &quot;16h DMSO &gt; 2h DEX+DMSO&quot;, 
&quot;16h KH-103 &gt; 2h DEX+KH-103&quot;, &quot;16h MIF  &gt; 2h DEX+MIF&quot;))
mm &lt;- model.matrix(~droplevels(se$condition2[wPrior]))
fit &lt;- glmFit(estimateDisp(dds[,wPrior],mm),mm)
res &lt;- as.data.frame(topTags(glmLRT(fit, contrast=c(0,0,1,0,0)), Inf))
rnames_dex_triggered &lt;-  head(row.names(res)[res$FDR&lt;0.05],200)
plt_dex_triggered  &lt;- plot_signals_and_sechm(gr_peaks, genes_subset=rnames_dex_triggered, plot_around_gene_tss=TRUE)</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>plt_dex_triggered</code></pre>
<p><img src="figure/01_Results.Rmd/inhibition%20effects-1.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-inhibition-effects-1">
Past versions of “inhibition effects-1.png”
</button>
</p>
<div id="fig-no-spaces-inhibition-effects-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/inhibition effects-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/inhibition effects-1.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n### Significantly more inhibited by KH-103 than by other inhibitors \n\n&quot;)</code></pre>
</div>
<div id="significantly-more-inhibited-by-kh-103-than-by-other-inhibitors" class="section level3">
<h3>Significantly more inhibited by KH-103 than by other inhibitors</h3>
<pre class="r"><code>### Significantly more inhibited by KH-103 than by other inhibitors
tmp &lt;- grepl(&quot;KH-103|control&quot;,se$condition2)[wPrior]
mm &lt;- model.matrix(~tmp)
fit &lt;- glmFit(estimateDisp(dds[,wPrior],mm),mm)
res &lt;- as.data.frame(topTags(glmLRT(fit), Inf))
rnames_kh103_inhibited &lt;-  head(row.names(res)[res$FDR&lt;0.05],200)
plt_kh103_inhibited  &lt;- plot_signals_and_sechm(gr_peaks, genes_subset=rnames_kh103_inhibited, plot_around_gene_tss=TRUE)</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>plt_kh103_inhibited</code></pre>
<p><img src="figure/01_Results.Rmd/inhibition%20effects-2.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-inhibition-effects-2">
Past versions of “inhibition effects-2.png”
</button>
</p>
<div id="fig-no-spaces-inhibition-effects-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/inhibition effects-2.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/inhibition effects-2.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n### Significantly less inhibited by KH-103 than by other inhibitors \n\n&quot;)</code></pre>
</div>
<div id="significantly-less-inhibited-by-kh-103-than-by-other-inhibitors" class="section level3">
<h3>Significantly less inhibited by KH-103 than by other inhibitors</h3>
<pre class="r"><code>### Significantly less inhibited by KH-103 than by other inhibitors
tmp &lt;- grepl(&quot;KH-103|DEX\\+DMSO&quot;,se$condition2)[wPrior]
mm &lt;- model.matrix(~tmp)
fit &lt;- glmFit(estimateDisp(dds[,wPrior],mm),mm)
res &lt;- as.data.frame(topTags(glmLRT(fit), Inf))
rnames_kh103_less_inhibited &lt;-  head(row.names(res)[res$FDR&lt;0.05],200)
plt_kh103_less_inhibited  &lt;- plot_signals_and_sechm(gr_peaks, genes_subset=rnames_kh103_less_inhibited, plot_around_gene_tss=TRUE)</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>plt_kh103_less_inhibited</code></pre>
<p><img src="figure/01_Results.Rmd/inhibition%20effects-3.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-inhibition-effects-3">
Past versions of “inhibition effects-3.png”
</button>
</p>
<div id="fig-no-spaces-inhibition-effects-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/inhibition effects-3.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/inhibition effects-3.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n### DEX-triggered genes that are inhibited (after)\n\n&quot;)</code></pre>
</div>
<div id="dex-triggered-genes-that-are-inhibited-after" class="section level3">
<h3>DEX-triggered genes that are inhibited (after)</h3>
<pre class="r"><code>## DEX-triggered genes that are inhibited (after)
wAfter &lt;- which(se$condition2 %in% c(&quot;control&quot;, &quot;2h DEX &gt; 16h DEX+Cort113&quot;, &quot;2h DEX &gt; 16h DEX+DMSO&quot;, 
&quot;2h DEX &gt; 16h DEX+KH-103&quot;, &quot;2h DEX &gt; 16h DEX+MIF&quot;))
mm &lt;- model.matrix(~droplevels(se$condition2[wAfter]))
fit &lt;- glmFit(estimateDisp(dds[,wAfter],mm),mm)
res &lt;- as.data.frame(topTags(glmLRT(fit, contrast=c(0,0,1,0,0)), Inf))
attr(res, &quot;description&quot;) &lt;- &quot;Significantly-inhibited DEX-responsive genes by any all inhibitors, when administered after DEX&quot;
rnames_inhibited &lt;-  head(row.names(res)[res$FDR&lt;0.05],200)
plt_inhibited   &lt;- plot_signals_and_sechm(gr_peaks, genes_subset=rnames_inhibited, plot_around_gene_tss=TRUE)</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>plt_inhibited </code></pre>
<p><img src="figure/01_Results.Rmd/inhibition%20effects-4.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-inhibition-effects-4">
Past versions of “inhibition effects-4.png”
</button>
</p>
<div id="fig-no-spaces-inhibition-effects-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/inhibition effects-4.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/inhibition effects-4.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="q9-what-is-the-relation-of-differential-expression-of-genes-and-differential-binding-of-nearby-regions" class="section level1 tabset">
<h1 class="tabset">Q9: What is the relation of differential expression of genes and differential binding of nearby regions?</h1>
<pre class="r"><code>et &lt;- readRDS(file=&quot;../data/results/differential_binding_results.rds&quot;)
merged_peaks &lt;- readRDS(file=&quot;../data/results/merged_peaks.rds&quot;)</code></pre>
<p>309 genes are significant at a 0.05 FDR level.</p>
<p>An overview of the differential binding peaks is available in the downloads section.</p>
<div id="tabular-overview" class="section level2">
<h2>Tabular overview:</h2>
<pre class="r"><code># print table
et %&gt;% 
  dplyr::select(nearestTSS.gene_name, nearestTSS, FDR,  everything()) %&gt;% 
  head(20) %&gt;%
  kable() %&gt;% 
  kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
nearestTSS.gene_name
</th>
<th style="text-align:left;">
nearestTSS
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:left;">
name
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
logCPM
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:right;">
distance2nearestTSS
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ERGIC1
</td>
<td style="text-align:left;">
ENST00000686449
</td>
<td style="text-align:right;">
0.0000449
</td>
<td style="text-align:left;">
chr5:172869049-172869468
</td>
<td style="text-align:right;">
8.196641
</td>
<td style="text-align:right;">
3.139451
</td>
<td style="text-align:right;">
1.0e-07
</td>
<td style="text-align:right;">
-8807
</td>
</tr>
<tr>
<td style="text-align:left;">
SCNN1A
</td>
<td style="text-align:left;">
ENST00000536411
</td>
<td style="text-align:right;">
0.0001185
</td>
<td style="text-align:left;">
chr12:6376197-6376471
</td>
<td style="text-align:right;">
9.442922
</td>
<td style="text-align:right;">
1.932770
</td>
<td style="text-align:right;">
2.0e-06
</td>
<td style="text-align:right;">
169
</td>
</tr>
<tr>
<td style="text-align:left;">
ZBTB16
</td>
<td style="text-align:left;">
ENST00000683006
</td>
<td style="text-align:right;">
0.0001185
</td>
<td style="text-align:left;">
chr11:114154481-114154905
</td>
<td style="text-align:right;">
6.603147
</td>
<td style="text-align:right;">
3.326866
</td>
<td style="text-align:right;">
2.1e-06
</td>
<td style="text-align:right;">
-5522
</td>
</tr>
<tr>
<td style="text-align:left;">
MIR3193
</td>
<td style="text-align:left;">
ENST00000578262
</td>
<td style="text-align:right;">
0.0001185
</td>
<td style="text-align:left;">
chr20:31610720-31611048
</td>
<td style="text-align:right;">
7.391412
</td>
<td style="text-align:right;">
2.354293
</td>
<td style="text-align:right;">
2.3e-06
</td>
<td style="text-align:right;">
3534
</td>
</tr>
<tr>
<td style="text-align:left;">
ZBTB16
</td>
<td style="text-align:left;">
ENST00000545851
</td>
<td style="text-align:right;">
0.0001185
</td>
<td style="text-align:left;">
chr11:114179197-114179671
</td>
<td style="text-align:right;">
6.317432
</td>
<td style="text-align:right;">
3.572129
</td>
<td style="text-align:right;">
2.5e-06
</td>
<td style="text-align:right;">
-1095
</td>
</tr>
<tr>
<td style="text-align:left;">
NANOGP9
</td>
<td style="text-align:left;">
ENST00000439890
</td>
<td style="text-align:right;">
0.0001185
</td>
<td style="text-align:left;">
chrX:65772821-65773162
</td>
<td style="text-align:right;">
6.559845
</td>
<td style="text-align:right;">
2.871221
</td>
<td style="text-align:right;">
2.7e-06
</td>
<td style="text-align:right;">
-470
</td>
</tr>
<tr>
<td style="text-align:left;">
ADGRG6
</td>
<td style="text-align:left;">
ENST00000508295
</td>
<td style="text-align:right;">
0.0001185
</td>
<td style="text-align:left;">
chr6:142376184-142376447
</td>
<td style="text-align:right;">
9.314629
</td>
<td style="text-align:right;">
1.809602
</td>
<td style="text-align:right;">
3.0e-06
</td>
<td style="text-align:right;">
-13865
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000523502
</td>
<td style="text-align:right;">
0.0001185
</td>
<td style="text-align:left;">
chr8:130762051-130762455
</td>
<td style="text-align:right;">
7.318173
</td>
<td style="text-align:right;">
2.255812
</td>
<td style="text-align:right;">
3.1e-06
</td>
<td style="text-align:right;">
106339
</td>
</tr>
<tr>
<td style="text-align:left;">
ZBTB7B
</td>
<td style="text-align:left;">
ENST00000292176
</td>
<td style="text-align:right;">
0.0001226
</td>
<td style="text-align:left;">
chr1:155017282-155017586
</td>
<td style="text-align:right;">
7.240682
</td>
<td style="text-align:right;">
2.180893
</td>
<td style="text-align:right;">
4.0e-06
</td>
<td style="text-align:right;">
2834
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000650258
</td>
<td style="text-align:right;">
0.0001226
</td>
<td style="text-align:left;">
chr5:172775589-172775944
</td>
<td style="text-align:right;">
6.422277
</td>
<td style="text-align:right;">
2.756907
</td>
<td style="text-align:right;">
4.1e-06
</td>
<td style="text-align:right;">
-521
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000669692
</td>
<td style="text-align:right;">
0.0001226
</td>
<td style="text-align:left;">
chr11:114162823-114163184
</td>
<td style="text-align:right;">
6.423372
</td>
<td style="text-align:right;">
2.738228
</td>
<td style="text-align:right;">
4.4e-06
</td>
<td style="text-align:right;">
1657
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000527351
</td>
<td style="text-align:right;">
0.0001231
</td>
<td style="text-align:left;">
chr11:123122426-123122655
</td>
<td style="text-align:right;">
9.139182
</td>
<td style="text-align:right;">
1.641666
</td>
<td style="text-align:right;">
4.8e-06
</td>
<td style="text-align:right;">
-13721
</td>
</tr>
<tr>
<td style="text-align:left;">
VMP1
</td>
<td style="text-align:left;">
ENST00000587470
</td>
<td style="text-align:right;">
0.0001231
</td>
<td style="text-align:left;">
chr17:59837426-59837958
</td>
<td style="text-align:right;">
5.922668
</td>
<td style="text-align:right;">
3.587849
</td>
<td style="text-align:right;">
5.2e-06
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
RNA5SP40
</td>
<td style="text-align:left;">
ENST00000517147
</td>
<td style="text-align:right;">
0.0001244
</td>
<td style="text-align:left;">
chr1:9436312-9436621
</td>
<td style="text-align:right;">
6.628821
</td>
<td style="text-align:right;">
2.419730
</td>
<td style="text-align:right;">
5.6e-06
</td>
<td style="text-align:right;">
-1157
</td>
</tr>
<tr>
<td style="text-align:left;">
CASC8
</td>
<td style="text-align:left;">
ENST00000502082
</td>
<td style="text-align:right;">
0.0001254
</td>
<td style="text-align:left;">
chr8:127500769-127501057
</td>
<td style="text-align:right;">
7.111911
</td>
<td style="text-align:right;">
2.056748
</td>
<td style="text-align:right;">
6.4e-06
</td>
<td style="text-align:right;">
18630
</td>
</tr>
<tr>
<td style="text-align:left;">
NUDT9P1
</td>
<td style="text-align:left;">
ENST00000477230
</td>
<td style="text-align:right;">
0.0001254
</td>
<td style="text-align:left;">
chr10:91144907-91145205
</td>
<td style="text-align:right;">
7.079109
</td>
<td style="text-align:right;">
2.053142
</td>
<td style="text-align:right;">
6.5e-06
</td>
<td style="text-align:right;">
-8144
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000656621
</td>
<td style="text-align:right;">
0.0001346
</td>
<td style="text-align:left;">
chr14:94599130-94599528
</td>
<td style="text-align:right;">
6.058585
</td>
<td style="text-align:right;">
3.089651
</td>
<td style="text-align:right;">
7.4e-06
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
ENST00000531569
</td>
<td style="text-align:right;">
0.0001408
</td>
<td style="text-align:left;">
chr11:35908078-35908413
</td>
<td style="text-align:right;">
6.213104
</td>
<td style="text-align:right;">
2.553969
</td>
<td style="text-align:right;">
8.6e-06
</td>
<td style="text-align:right;">
-10305
</td>
</tr>
<tr>
<td style="text-align:left;">
CBLB
</td>
<td style="text-align:left;">
ENST00000643403
</td>
<td style="text-align:right;">
0.0001408
</td>
<td style="text-align:left;">
chr3:105792326-105792693
</td>
<td style="text-align:right;">
6.133745
</td>
<td style="text-align:right;">
2.886163
</td>
<td style="text-align:right;">
9.0e-06
</td>
<td style="text-align:right;">
-60953
</td>
</tr>
<tr>
<td style="text-align:left;">
ZMYND8
</td>
<td style="text-align:left;">
ENST00000441977
</td>
<td style="text-align:right;">
0.0001408
</td>
<td style="text-align:left;">
chr20:47334115-47334473
</td>
<td style="text-align:right;">
5.941042
</td>
<td style="text-align:right;">
3.005947
</td>
<td style="text-align:right;">
9.9e-06
</td>
<td style="text-align:right;">
-13599
</td>
</tr>
</tbody>
</table>
</div>
<div id="volcano-plot" class="section level2">
<h2>Volcano plot</h2>
<pre class="r"><code>ggplot(et, aes(logFC, -log10(FDR))) + geom_point() + 
  geom_hline(yintercept=-log10(0.05), linetype=&quot;dashed&quot;) +
  theme_ipsum_rc() +
  labs(title=paste0(&quot;Differential binding results for &quot;, dim(et)[1], &quot; genes&quot;)) +
  ggrepel::geom_text_repel(data=et,
                           aes(x=logFC, y=-log10(FDR), label=nearestTSS.gene_name, size=4),
                           show.legend=F,
                           max.overlaps = 30)</code></pre>
<pre><code>Warning: ggrepel: 210 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/01_Results.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/unnamed-chunk-10-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/unnamed-chunk-10-1.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="signal-tracks" class="section level2">
<h2>Signal tracks</h2>
<pre class="r"><code>tracks &lt;- list(&quot;C2&quot;=&quot;../data/single_replicate_peaks/control/Chip_GR_control_s2.bigWig&quot;,
               &quot;C3&quot;=&quot;../data/single_replicate_peaks/control/Chip_GR_control_s3.bigWig&quot;,
               &quot;t12_1&quot;=&quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1.bigWig&quot;,
               &quot;t12_2&quot;=&quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2.bigWig&quot;,
               &quot;t12_3&quot;=&quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s3.bigWig&quot;)
m &lt;- signal2Matrix(tracks, merged_peaks)</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/control/Chip_GR_control_s2.bigWig</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/control/Chip_GR_control_s3.bigWig</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1.bigWig</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2.bigWig</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s3.bigWig</code></pre>
<pre class="r"><code>plotEnrichedHeatmaps(m)</code></pre>
<p><img src="figure/01_Results.Rmd/unnamed-chunk-11-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/unnamed-chunk-11-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/unnamed-chunk-11-1.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="average-signal-plots" class="section level2">
<h2>Average signal plots</h2>
<pre class="r"><code>d1 &lt;- meltSignals(m)
d1$condition &lt;- gsub(&quot;[0-9]&quot;,&quot;&quot;,d1$sample)
d1$condition &lt;- ifelse(d1$condition==&quot;C&quot;, &quot;Control&quot;, &quot;12hDEX&quot;)
p1 &lt;- ggplot(d1, aes(position, mean, group=sample, colour=condition)) + geom_line() + ggtitle(&quot;Average signal plots&quot;) + theme_ipsum_rc()
p1</code></pre>
<p><img src="figure/01_Results.Rmd/unnamed-chunk-12-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/unnamed-chunk-12-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># d2 &lt;- meltSignals(m2)
# d2$condition &lt;- gsub(&quot;[0-9]&quot;,&quot;&quot;,d2$sample)
# d2$condition &lt;- ifelse(d2$condition==&quot;C&quot;, &quot;Control&quot;, &quot;12hDEX&quot;)
# p2 &lt;- ggplot(d2, aes(position, mean, group=sample, colour=condition)) + geom_line() + ggtitle(&quot;Top peaks normalization&quot;) + theme_ipsum_rc()
# ggpubr::ggarrange(p1, p2, ncol=2)</code></pre>
</div>
</div>
<div id="q10-dataset-of-known-enhancer---target-gene-interactions-by-salviato-et-al." class="section level1">
<h1>Q10: Dataset of known enhancer - target gene interactions by Salviato et al.</h1>
<p><strong>Q: Are there some GR peaks which influence expression of genes despite not being close to these genes?</strong></p>
<p><a href="https://academic.oup.com/nar/article/49/17/e97/6312759">Salviato et al.</a> have curated a dataset of genomic regions which are known to act with distal genes. By comparing these regions with our peaks we can identify potential long-distance relationships between GR peaks and genes.</p>
<pre class="r"><code>interactions &lt;- read.csv(&quot;../data/Formatted_Candidate_ETG_20200305_TAD10Kb.tsv&quot;, sep=&quot;\t&quot;)
interactions_gr &lt;- makeGRangesFromDataFrame(interactions, keep.extra.columns=T)

# subset for peaks which are in a range of 2500bp of a TSS
# gr_peaks &lt;- gr_peaks[gr_peaks$nearestTSS.gene_name.2500r != &quot;&quot;]

# find peaks from our data peak dataset that overlap with the known ETG interactions
overlap_peaks &lt;- findOverlaps(gr_peaks, interactions_gr)

# get a granges object of all peaks which have an associated long-range interaction in the table
new_peaks_lri &lt;- gr_peaks[queryHits(overlap_peaks)]
new_peaks_lri$long_range_interaction &lt;- TRUE
new_peaks_lri$long_range_gene &lt;- interactions_gr[subjectHits(overlap_peaks)]$symbol
new_peaks_lri$dist2lr_target &lt;- interactions_gr[subjectHits(overlap_peaks)]$distance

# get peaks from our dataset which have no known interactions with distal genes
mask_peaks_with_no_lr_interactions &lt;- !(1:length(gr_peaks) %in% unique(queryHits(overlap_peaks)))
new_peaks_no_lri &lt;- gr_peaks[mask_peaks_with_no_lr_interactions]
new_peaks_no_lri$long_range_interaction &lt;- FALSE
new_peaks_no_lri$long_range_gene &lt;- NA
new_peaks_no_lri$dist2lr_target &lt;- NA

# merge the granges objects together. This object will contain duplicates
peaks &lt;- c(new_peaks_lri, new_peaks_no_lri)

# Subset the granges object for </code></pre>
<div id="how-many-significant-genes-are-there-in-our-18h-dex-treatment-depending-on-the-type-of-peaks-associated-with-the-gene" class="section level3">
<h3>How many significant genes are there in our 18h DEX treatment depending on the type of peaks associated with the gene?</h3>
<p>TRUE/FALSE indicates whether the gene was found to be differentially expressed in our 18h DEX treatment at a 0.05 FDR level.</p>
<pre class="r"><code>### How many peaks have no TSS nearby but are known for long-range interactions?
de_res &lt;- read.xlsx(&quot;../docs/downloads/Q2_GR_bound_DE_genes.xlsx&quot;)
de_res &lt;- de_res %&gt;% left_join(as.data.frame(peaks) %&gt;% dplyr::select(&quot;long_range_gene&quot;, &quot;long_range_interaction&quot;, &quot;dist2lr_target&quot;) %&gt;% dplyr::rename(Gene=long_range_gene),
                               by=&quot;Gene&quot;) %&gt;%
  group_by(Gene) %&gt;%
  mutate(min_dist2lr_target=min(abs(dist2lr_target))) %&gt;%
  dplyr::select(-dist2lr_target) %&gt;%
  distinct() %&gt;%
  tidyr::replace_na(list(long_range_interaction=F))

de_res$category &lt;- NA
de_res[(de_res$chip_peaks.2500r &gt; 0) &amp; (de_res$long_range_interaction == FALSE), &quot;category&quot;] &lt;- &quot;&lt;2500 to TSS only&quot;
de_res[(de_res$chip_peaks.2500r == 0) &amp; (de_res$long_range_interaction == TRUE), &quot;category&quot;] &lt;- &quot;LRI only&quot;
de_res[(de_res$chip_peaks.2500r == 0) &amp; (de_res$long_range_interaction == FALSE), &quot;category&quot;] &lt;- &quot;Neither&quot;
de_res[(de_res$chip_peaks.2500r &gt; 0) &amp; (de_res$long_range_interaction == TRUE), &quot;category&quot;] &lt;- &quot;&lt;2500 to TSS+LRI&quot;
de_res$category &lt;- factor(de_res$category, levels=c(&quot;Neither&quot;, &quot;LRI only&quot;, &quot;&lt;2500 to TSS only&quot;,&quot;&lt;2500 to TSS+LRI&quot;))

kableExtra::kable(table(de_res$is_significant, de_res$category)) %&gt;% kableExtra::kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Neither
</th>
<th style="text-align:right;">
LRI only
</th>
<th style="text-align:right;">
&lt;2500 to TSS only
</th>
<th style="text-align:right;">
&lt;2500 to TSS+LRI
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
5582
</td>
<td style="text-align:right;">
1616
</td>
<td style="text-align:right;">
90
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
4477
</td>
<td style="text-align:right;">
1671
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
202
</td>
</tr>
</tbody>
</table>
</div>
<div id="do-we-see-different-log-fold-changes-depending-on-the-type-of-associated-peaks" class="section level3">
<h3>Do we see different log-fold changes depending on the type of associated peaks?</h3>
<pre class="r"><code># Idea: A table of our peaks with a the gene it is influencing and the distance to the gene
### boxplot: no closeby peak, but known lr-interaction VS. no peak at all, closeby peak and long-range interaction, closeby peak only
de_res %&gt;%
  dplyr::filter(FDR &lt; 0.05) %&gt;%
  ggplot(aes(x=category, y=abs(logFC))) + 
  geom_boxplot() +
  theme_ipsum_rc() +#
  ylim(c(0,2)) +
  labs(title=&quot;Absolute logFC differences based on types of associated peaks&quot;, subtitle=&quot;DE genes with FDR &lt; 0.05 only&quot;, y=&quot;Absolute logFC&quot;)</code></pre>
<pre><code>Warning: Removed 206 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/01_Results.Rmd/boxplot%20LRI-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-LRI-1">
Past versions of “boxplot LRI-1.png”
</button>
</p>
<div id="fig-no-spaces-boxplot-LRI-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/01_Results.Rmd/boxplot LRI-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/figure/01_Results.Rmd/boxplot LRI-1.png" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>###
# Idea 2: A table of peaks with a column containing a string. This string is the comma-separated gene names
peaks_short &lt;- as.data.frame(peaks) %&gt;% 
  dplyr::mutate(id=paste0(seqnames, &quot;:&quot;, start, &quot;-&quot;, end)) %&gt;%
  # select(&quot;id&quot;, everything()) %&gt;%
  group_by(id) %&gt;% 
  mutate(lr_genes = paste0(long_range_gene, collapse = &quot;, &quot;))  %&gt;%
  dplyr::select(-c(&quot;long_range_gene&quot;,&quot;dist2lr_target&quot;)) %&gt;%
  distinct() %&gt;%
  dplyr::select(&quot;id&quot;, &quot;score&quot;, &quot;signalValue&quot;, &quot;pValue&quot;, &quot;distance2nearestTSS&quot;, &quot;nearestTSS.gene_name&quot;, #&quot;nearestTSS&quot;, 
         &quot;class&quot;,&quot;long_range_interaction&quot;, &quot;lr_genes&quot;)</code></pre>
</div>
<div id="is-the-gene-closest-to-a-peak-included-in-the-list-of-genes-acting-at-long-range-based-on-the-salviato-dataset" class="section level3">
<h3>Is the gene closest to a peak included in the list of genes acting at long-range, based on the Salviato dataset?</h3>
<pre class="r"><code>peaks_short$nearestTSS.in.LRI_genes &lt;- mapply(grepl, peaks_short$nearestTSS.gene_name, peaks_short$lr_genes)
peaks_short[peaks_short$nearestTSS.gene_name == &quot;&quot;, &quot;nearestTSS.in.LRI_genes&quot;] &lt;-  NA
table(as.character(peaks_short$nearestTSS.in.LRI_genes), useNA=&quot;always&quot;) %&gt;% 
  as.data.frame() %&gt;% 
  dplyr::rename(&quot;Nearest TSS is one of the LRI genes&quot;=Var1) %&gt;% 
  kable() %&gt;% 
  kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Nearest TSS is one of the LRI genes
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
5503
</td>
</tr>
<tr>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
405
</td>
</tr>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
2012
</td>
</tr>
</tbody>
</table>
<p>A table of the genes as displayed below can be accessed as an excel file in the download section.</p>
<p>It includes the DE results from Mahshid’s 18h DEX experiment augmented by information on whether a peak in the ChIP data was found to be closest to this gene, as well as whether a peak was found to interact with the TSS of the gene based on the Salviato dataset.</p>
<pre class="r"><code>wb &lt;- createWorkbook()
sheet_name &lt;- &quot;Augmented_18h_DE&quot;
addWorksheet(wb, sheet_name)
writeData(wb, sheet = sheet_name, de_res)
saveWorkbook(wb, &quot;../docs/downloads/Q10_DE_genes_long_range_interactions.xlsx&quot;, overwrite = TRUE)

de_res %&gt;% head(20) %&gt;% kable() %&gt;% kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
logCPM
</th>
<th style="text-align:right;">
LR
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
chip_peaks
</th>
<th style="text-align:right;">
chip_peaks.2500r
</th>
<th style="text-align:left;">
gc_bound
</th>
<th style="text-align:left;">
is_significant
</th>
<th style="text-align:left;">
long_range_interaction
</th>
<th style="text-align:right;">
min_dist2lr_target
</th>
<th style="text-align:left;">
category
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ANGPTL4
</td>
<td style="text-align:right;">
4.301591
</td>
<td style="text-align:right;">
7.074175
</td>
<td style="text-align:right;">
2374.2119
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
207532
</td>
<td style="text-align:left;">
&lt;2500 to TSS+LRI
</td>
</tr>
<tr>
<td style="text-align:left;">
FKBP5
</td>
<td style="text-align:right;">
4.250025
</td>
<td style="text-align:right;">
6.170062
</td>
<td style="text-align:right;">
2090.9533
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
35155
</td>
<td style="text-align:left;">
&lt;2500 to TSS+LRI
</td>
</tr>
<tr>
<td style="text-align:left;">
ATP1A1
</td>
<td style="text-align:right;">
1.601091
</td>
<td style="text-align:right;">
8.169135
</td>
<td style="text-align:right;">
1554.3518
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
Neither
</td>
</tr>
<tr>
<td style="text-align:left;">
PYGB
</td>
<td style="text-align:right;">
1.955580
</td>
<td style="text-align:right;">
8.771525
</td>
<td style="text-align:right;">
1433.3839
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
5542
</td>
<td style="text-align:left;">
LRI only
</td>
</tr>
<tr>
<td style="text-align:left;">
TFCP2L1
</td>
<td style="text-align:right;">
3.792744
</td>
<td style="text-align:right;">
4.908050
</td>
<td style="text-align:right;">
1382.0798
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
&lt;2500 to TSS only
</td>
</tr>
<tr>
<td style="text-align:left;">
CIDEC
</td>
<td style="text-align:right;">
4.310337
</td>
<td style="text-align:right;">
4.888107
</td>
<td style="text-align:right;">
1343.4096
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
&lt;2500 to TSS only
</td>
</tr>
<tr>
<td style="text-align:left;">
MT2A
</td>
<td style="text-align:right;">
3.534036
</td>
<td style="text-align:right;">
8.088070
</td>
<td style="text-align:right;">
1319.6656
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
&lt;2500 to TSS only
</td>
</tr>
<tr>
<td style="text-align:left;">
TSC22D3
</td>
<td style="text-align:right;">
3.356876
</td>
<td style="text-align:right;">
5.640700
</td>
<td style="text-align:right;">
1277.8539
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
&lt;2500 to TSS only
</td>
</tr>
<tr>
<td style="text-align:left;">
RHCG
</td>
<td style="text-align:right;">
3.079983
</td>
<td style="text-align:right;">
4.700196
</td>
<td style="text-align:right;">
1161.4860
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
379498
</td>
<td style="text-align:left;">
&lt;2500 to TSS+LRI
</td>
</tr>
<tr>
<td style="text-align:left;">
TGM2
</td>
<td style="text-align:right;">
1.489977
</td>
<td style="text-align:right;">
9.882849
</td>
<td style="text-align:right;">
1155.4558
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
928082
</td>
<td style="text-align:left;">
&lt;2500 to TSS+LRI
</td>
</tr>
<tr>
<td style="text-align:left;">
FAM43A
</td>
<td style="text-align:right;">
3.147715
</td>
<td style="text-align:right;">
3.949821
</td>
<td style="text-align:right;">
961.5880
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
109577
</td>
<td style="text-align:left;">
LRI only
</td>
</tr>
<tr>
<td style="text-align:left;">
ACSL1
</td>
<td style="text-align:right;">
3.005821
</td>
<td style="text-align:right;">
5.447785
</td>
<td style="text-align:right;">
959.8093
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
&lt;2500 to TSS only
</td>
</tr>
<tr>
<td style="text-align:left;">
STOM
</td>
<td style="text-align:right;">
1.954002
</td>
<td style="text-align:right;">
8.047979
</td>
<td style="text-align:right;">
953.1021
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
Neither
</td>
</tr>
<tr>
<td style="text-align:left;">
CAVIN2
</td>
<td style="text-align:right;">
1.744690
</td>
<td style="text-align:right;">
5.871643
</td>
<td style="text-align:right;">
939.0613
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
&lt;2500 to TSS only
</td>
</tr>
<tr>
<td style="text-align:left;">
NBEAL2
</td>
<td style="text-align:right;">
2.001726
</td>
<td style="text-align:right;">
6.834884
</td>
<td style="text-align:right;">
873.5847
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
Neither
</td>
</tr>
<tr>
<td style="text-align:left;">
SHC1
</td>
<td style="text-align:right;">
1.039135
</td>
<td style="text-align:right;">
8.659475
</td>
<td style="text-align:right;">
866.4488
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
&lt;2500 to TSS only
</td>
</tr>
<tr>
<td style="text-align:left;">
MT1X
</td>
<td style="text-align:right;">
3.348676
</td>
<td style="text-align:right;">
6.433053
</td>
<td style="text-align:right;">
821.6947
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
&lt;2500 to TSS only
</td>
</tr>
<tr>
<td style="text-align:left;">
SEC14L2
</td>
<td style="text-align:right;">
2.425641
</td>
<td style="text-align:right;">
4.776408
</td>
<td style="text-align:right;">
811.4586
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
Neither
</td>
</tr>
<tr>
<td style="text-align:left;">
OTULINL
</td>
<td style="text-align:right;">
3.882485
</td>
<td style="text-align:right;">
4.702683
</td>
<td style="text-align:right;">
806.0085
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
7867
</td>
<td style="text-align:left;">
LRI only
</td>
</tr>
<tr>
<td style="text-align:left;">
ITGA5
</td>
<td style="text-align:right;">
2.133065
</td>
<td style="text-align:right;">
6.007132
</td>
<td style="text-align:right;">
721.2871
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
Neither
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="section" class="section level1">
<h1></h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.18.4            AnnotationFilter_1.18.0    
 [3] GenomicFeatures_1.46.5      AnnotationDbi_1.56.2       
 [5] AnnotationHub_3.2.2         BiocFileCache_2.2.1        
 [7] dbplyr_2.2.1                kableExtra_1.3.4           
 [9] epiwraps_0.99.42            EnrichedHeatmap_1.24.0     
[11] ComplexHeatmap_2.10.0       rtracklayer_1.54.0         
[13] stringdist_0.9.8            openxlsx_4.2.5             
[15] hrbrthemes_0.8.6            VennDiagram_1.7.3          
[17] futile.logger_1.4.3         SEtools_1.8.0              
[19] edgeR_3.36.0                limma_3.50.3               
[21] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[23] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[25] IRanges_2.28.0              S4Vectors_0.32.4           
[27] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[29] matrixStats_0.62.0          dplyr_1.0.9                
[31] biomaRt_2.50.3              ggplot2_3.3.6              

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                tidyr_1.2.0                  
  [3] bit64_4.0.5                   knitr_1.39                   
  [5] DelayedArray_0.20.0           data.table_1.14.2            
  [7] rpart_4.1.16                  KEGGREST_1.34.0              
  [9] RCurl_1.98-1.7                doParallel_1.0.17            
 [11] generics_0.1.3                cowplot_1.1.1                
 [13] lambda.r_1.2.4                RSQLite_2.2.14               
 [15] bit_4.0.4                     webshot_0.5.3                
 [17] xml2_1.3.3                    httpuv_1.6.5                 
 [19] assertthat_0.2.1              viridis_0.6.2                
 [21] xfun_0.31                     hms_1.1.1                    
 [23] jquerylib_0.1.4               evaluate_0.15                
 [25] promises_1.2.0.1              TSP_1.2-0                    
 [27] fansi_1.0.3                   restfulr_0.0.15              
 [29] progress_1.2.2                DBI_1.1.3                    
 [31] geneplotter_1.72.0            htmlwidgets_1.5.4            
 [33] purrr_0.3.4                   ellipsis_0.3.2               
 [35] backports_1.4.1               V8_4.2.0                     
 [37] annotate_1.72.0               deldir_1.0-6                 
 [39] vctrs_0.4.1                   cachem_1.0.6                 
 [41] withr_2.5.0                   Gviz_1.38.4                  
 [43] BSgenome_1.62.0               checkmate_2.1.0              
 [45] GenomicAlignments_1.30.0      prettyunits_1.1.1            
 [47] svglite_2.1.0                 cluster_2.1.3                
 [49] lazyeval_0.2.2                crayon_1.5.1                 
 [51] genefilter_1.76.0             pkgconfig_2.0.3              
 [53] labeling_0.4.2                nlme_3.1-158                 
 [55] ProtGenerics_1.26.0           seriation_1.3.5              
 [57] nnet_7.3-17                   rlang_1.0.3                  
 [59] lifecycle_1.0.1               registry_0.5-1               
 [61] filelock_1.0.2                extrafontdb_1.0              
 [63] dichromat_2.0-0.1             rprojroot_2.0.3              
 [65] Matrix_1.4-1                  base64enc_0.1-3              
 [67] whisker_0.4                   GlobalOptions_0.1.2          
 [69] png_0.1-7                     viridisLite_0.4.0            
 [71] rjson_0.2.21                  bitops_1.0-7                 
 [73] Biostrings_2.62.0             blob_1.2.3                   
 [75] workflowr_1.7.0               shape_1.4.6                  
 [77] stringr_1.4.0                 jpeg_0.1-9                   
 [79] scales_1.2.0                  memoise_2.0.1                
 [81] magrittr_2.0.3                plyr_1.8.7                   
 [83] zlibbioc_1.40.0               compiler_4.1.0               
 [85] BiocIO_1.4.0                  RColorBrewer_1.1-3           
 [87] clue_0.3-61                   DESeq2_1.34.0                
 [89] Rsamtools_2.10.0              cli_3.3.0                    
 [91] XVector_0.34.0                htmlTable_2.4.0              
 [93] formatR_1.12                  Formula_1.2-4                
 [95] mgcv_1.8-40                   tidyselect_1.1.2             
 [97] stringi_1.7.6                 highr_0.9                    
 [99] yaml_2.3.5                    locfit_1.5-9.5               
[101] latticeExtra_0.6-30           ggrepel_0.9.1                
[103] sass_0.4.1                    VariantAnnotation_1.40.0     
[105] randomcoloR_1.1.0.1           tools_4.1.0                  
[107] parallel_4.1.0                circlize_0.4.15              
[109] rstudioapi_0.13               foreach_1.5.2                
[111] foreign_0.8-82                git2r_0.30.1                 
[113] gridExtra_2.3                 farver_2.1.1                 
[115] Rtsne_0.16                    sechm_1.2.0                  
[117] digest_0.6.29                 BiocManager_1.30.18          
[119] shiny_1.7.1                   Rcpp_1.0.8.3                 
[121] BiocVersion_3.14.0            later_1.3.0                  
[123] httr_1.4.3                    gdtools_0.2.4                
[125] biovizBase_1.42.0             colorspace_2.0-3             
[127] rvest_1.0.2                   XML_3.99-0.10                
[129] fs_1.5.2                      splines_4.1.0                
[131] systemfonts_1.0.4             xtable_1.8-4                 
[133] jsonlite_1.8.0                futile.options_1.0.1         
[135] GenomicFiles_1.30.0           UpSetR_1.4.0                 
[137] R6_2.5.1                      Hmisc_4.7-0                  
[139] mime_0.12                     pillar_1.7.0                 
[141] htmltools_0.5.2               glue_1.6.2                   
[143] fastmap_1.1.0                 BiocParallel_1.28.3          
[145] interactiveDisplayBase_1.32.0 codetools_0.2-18             
[147] utf8_1.2.2                    lattice_0.20-45              
[149] bslib_0.3.1                   tibble_3.1.7                 
[151] sva_3.42.0                    curl_4.3.2                   
[153] magick_2.7.3                  zip_2.2.0                    
[155] Rttf2pt1_1.3.10               interp_1.1-2                 
[157] survival_3.3-1                rmarkdown_2.14               
[159] munsell_0.5.0                 GetoptLong_1.0.5             
[161] GenomeInfoDbData_1.2.7        iterators_1.0.14             
[163] gtable_0.3.0                  extrafont_0.18               </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
