<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-06-22" />

<title>Analysis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-06-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree207afa653f1bc6b51cf5f07acfe8c498be77c57btargetblank207afa6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/207afa653f1bc6b51cf5f07acfe8c498be77c57b" target="_blank">207afa6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree207afa653f1bc6b51cf5f07acfe8c498be77c57btargetblank207afa6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/207afa653f1bc6b51cf5f07acfe8c498be77c57b" target="_blank">207afa6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/

Untracked files:
    Untracked:  code/helper_functions.R
    Untracked:  data/reddy_bed_ATAC_12h_DEX.bed.gz
    Untracked:  data/reddy_bed_ATAC_control.bed.gz
    Untracked:  data/reddy_bed_DEX12h_r123.bed
    Untracked:  data/reddy_bed_DEX2h_r123.bed.gz
    Untracked:  data/reddy_bed_JUNB_DEX12h_r123.bed
    Untracked:  data/reddy_bigwig_DEX12h_r123.bigWig
    Untracked:  data/reddy_bigwig_DEX2h_r123.bigWig
    Untracked:  data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Untracked:  data/reddy_rna_DEX12h.tsv
    Untracked:  data/reddy_rna_DEX12h_r1.tsv
    Untracked:  data/reddy_rna_DEX12h_r2.tsv
    Untracked:  data/reddy_rna_DEX12h_r3.tsv
    Untracked:  data/reddy_rna_DEX12h_r4.tsv
    Untracked:  data/reddy_rna_DEX12h_transcript.tsv
    Untracked:  data/reddy_rna_control.tsv
    Untracked:  data/reddy_rna_control_r1.tsv
    Untracked:  data/reddy_rna_control_r2.tsv
    Untracked:  data/reddy_rna_control_r3.tsv
    Untracked:  data/reddy_rna_control_r4.tsv
    Untracked:  data/results/
    Untracked:  output/Q2_GR_bound_DE_genes.xlsx
    Untracked:  output/Q4_KH_DE_GR_bound.xlsx

Unstaged changes:
    Modified:   .gitignore
    Modified:   _workflowr.yml
    Modified:   analysis/01_download_files.Rmd
    Modified:   analysis/02_exploratory_analysis.Rmd
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q2-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/Q4-GR-binding-PROTAC-genes.Rmd
    Deleted:    analysis/Q6-JUNB-GR-overlap.Rmd
    Modified:   analysis/Q7-PRX-sequences.Rmd
    Modified:   analysis/about.Rmd
    Modified:   analysis/admin.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/02_exploratory_analysis.Rmd</code>) and HTML (<code>docs/02_exploratory_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/analysis/02_exploratory_analysis.Rmd" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/02_exploratory_analysis.html" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="question-1-similarity-of-our-rna-data-with-the-reddy-data" class="section level1">
<h1>Question 1: Similarity of our RNA data with the Reddy data</h1>
<p><strong>Q: Which gene expressions have similar patterns of up-regulation (control vs. Dex) in our data compared to the Reddy data?</strong></p>
<p>In our experiments we compared samples from a control experiment with samples that were treated for 18 hours with DEX. Besides a control, the Reddy lab performed a similar experiment where they treated cells with DEX for 12 hours. This is the closest comparison to our data.</p>
<p><em>Planned approach:</em> Check which genes are differentially expressed in their data and create a Venn-diagram for comparable experiments with the differentially expressed genes from our analysis.</p>
<div id="results" class="section level2">
<h2>Results</h2>
<pre class="r"><code>q1_data &lt;- readRDS(file=&quot;../data/EOS_files/Q1_EOS.rds&quot;)
names(q1_data)</code></pre>
<pre><code>[1] &quot;Reddy_all&quot;           &quot;Reddy_significant&quot;   &quot;Mahshid_all&quot;        
[4] &quot;Mahshid_significant&quot; &quot;union_DE_genes&quot;      &quot;logFC_plot_data&quot;    </code></pre>
<pre class="r"><code>n_genes_plot &lt;- dim(q1_data[[&quot;logFC_plot_data&quot;]])[1]
n_genes_union &lt;- length(q1_data[[&quot;union_DE_genes&quot;]])</code></pre>
<p>We first inspect a Venn diagram to understand the overlap of differentially expressed genes from both datasets. We see that a large part of the genes do overlap, while there are also many genes that are specific to the respective analyses.</p>
<p><img src="assets/1_venn_diagram.png" /></p>
<p>Second, we compare the log fold changes for the union of all genes identified as significantly differentially expressed. The union of all DE genes counts 8121 genes of which 7342 genes (0.9%) were also in the intersection of both datasets and could be plotted.</p>
<p><img src="assets/2_logFC_Mahshid_Reddy.png" /></p>
</div>
</div>
<div id="question-2-binding-of-gr-gc-complex-to-de-genes-from-our-analysis" class="section level1">
<h1>Question 2: Binding of GR-GC complex to DE genes from our analysis</h1>
<p><strong>Q: For the genes that our RNA-seq data shows to be DE after treatment with DEX as well as the inhibitors and control: do we see any direct TF binding to these sites? For which of these is there a difference compared to the control ChIP-seq experiment?</strong></p>
<p>We see a significant change in terms of both the p-value and the logFC for genes that are bound by the GC-GR complex.</p>
<p>Plot: we select all genes that are differentially expressed - both for M = shid’s data and for the DE analysis on the Reddy data. We then split the data into two groups based on whether the Reddy data showed that the glucorticoid receptor was found to bind near the gene or not. We plot the distributions of (a) the log fold changes in the DE analysis and (b) the p-value of the DE analysis.</p>
<p><strong>Idea for next step:</strong> Does it make sense to look at pathways of associated genes?</p>
<p><img src="assets/Q2_3_mahshid_reddy.png" /></p>
</div>
<div id="question-3-inhibition-unique-to-protac-compared-to-cort-and-mif" class="section level1 tabset">
<h1 class="tabset">Question 3: Inhibition unique to protac compared to Cort and MIF</h1>
<p>Are there any genes that treated by the Protac before/after DEX that could be reversed but which could not by the inhibitors or vice versa?</p>
<p>We have performed experiments where we first exposed cells to either (1) DEX (2) inhibitors or (3) KH PROTAC for a substantial time and then added inhibitors or the KH PROTAC in case (1) or DEX in cases (2) and (3). We would like to see whether the genes which were activated by DEX but the returned to a normal expression by the KH PROTAC, where a similar treatment by the inhibitors could not lead to such an effect and vice versa.</p>
<div id="h-16h" class="section level2">
<h2>2h &gt; 16h</h2>
<p><img src="assets/Q3_1_reversal216.png" /></p>
</div>
<div id="h-16h-logfc-0" class="section level2">
<h2>2h &gt; 16h (logFC &gt; 0)</h2>
<p><img src="assets/Q3_2_reversal162_p1.png" /></p>
</div>
<div id="h-16h-logfc-0-1" class="section level2">
<h2>2h &gt; 16h (logFC &lt; 0)</h2>
<p><img src="assets/Q3_3_reversal162_p2.png" /></p>
<p>Download excel file with data <a href="assets/protact_flipped_DE.xlsx">here</a></p>
</div>
</div>
<div id="question-4-does-the-gr-bind-close-to-any-of-the-13-de-genes-from-the-12h-kh-protac-treatment" class="section level1">
<h1>Question 4: Does the GR bind close to any of the 13 DE genes from the 12h KH-PROTAC treatment?</h1>
<p>We have 13 genes that are differentially expressed at an &lt;0.05 FDR level. The Reddy lab has collected data on where the glucorticoid receptor (NR3C1) binds to after treatment with dexamethasone for 12h. We find that the GR is found to bind close to five out of the 13 differentially expressed genes.</p>
<pre class="r"><code>kh.protac &lt;- read.xlsx(&quot;../output/Q4_KH_DE_GR_bound.xlsx&quot;)

# only keep the significant genes
kh.protac.sig &lt;- kh.protac[kh.protac$FDR&lt;0.05,]
kh.protac.sig[,c(&quot;Gene&quot;, &quot;logFC&quot;, &quot;FDR&quot;, &quot;chip_peaks&quot;, &quot;gc_bound&quot;)] %&gt;%
  kableExtra::kable() %&gt;%
  kableExtra::kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
chip_peaks
</th>
<th style="text-align:left;">
gc_bound
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PER1
</td>
<td style="text-align:right;">
1.1428657
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
TGFBR1
</td>
<td style="text-align:right;">
0.5510412
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
FKBP5
</td>
<td style="text-align:right;">
0.7301845
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
NBEAL2
</td>
<td style="text-align:right;">
0.3850139
</td>
<td style="text-align:right;">
0.0001572
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
CAVIN2
</td>
<td style="text-align:right;">
0.3182665
</td>
<td style="text-align:right;">
0.0006525
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
SLC38A2
</td>
<td style="text-align:right;">
0.2546515
</td>
<td style="text-align:right;">
0.0007788
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
TEF
</td>
<td style="text-align:right;">
-0.9140413
</td>
<td style="text-align:right;">
0.0010003
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
CRY1
</td>
<td style="text-align:right;">
-0.3858283
</td>
<td style="text-align:right;">
0.0021419
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
C5AR2
</td>
<td style="text-align:right;">
0.6589461
</td>
<td style="text-align:right;">
0.0039443
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
ANGPTL4
</td>
<td style="text-align:right;">
0.4185862
</td>
<td style="text-align:right;">
0.0064158
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
CRY2
</td>
<td style="text-align:right;">
-0.4867045
</td>
<td style="text-align:right;">
0.0097262
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
CHML
</td>
<td style="text-align:right;">
0.2746874
</td>
<td style="text-align:right;">
0.0135275
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
<tr>
<td style="text-align:left;">
PER3
</td>
<td style="text-align:right;">
-0.7575232
</td>
<td style="text-align:right;">
0.0398908
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
FALSE
</td>
</tr>
</tbody>
</table>
<p><img src="assets/Q4_1_kh_gr_binding.png" /></p>
</div>
<div id="question-5-binding-to-selected-genes" class="section level1">
<h1>Question 5: Binding to selected genes</h1>
<p><strong>Q: Does GR bind to BRCA1, Hsd11b1 and CH25H?</strong></p>
<p>–&gt; We see that there is one peak in the ChIP data close to the BRCA1 gene and two close to the HSD11B1 gene. There is no peak that is closest to the CH25H gene. However, the CD25H was also not found in the RNA seq data.</p>
<pre class="r"><code>res_int &lt;- read.xlsx(&quot;../output/Q2_GR_bound_DE_genes.xlsx&quot;)
plt_data &lt;- res_int[res_int$Gene %in% c(&quot;BRCA1&quot;, &quot;HSD11B1&quot;, &quot;CH25H&quot;),c(&quot;Gene&quot;, &quot;logFC&quot;, &quot;FDR&quot;, &quot;chip_peaks&quot;, &quot;gc_bound&quot;)]
plt_data %&gt;%
  ggplot() +
  geom_point(aes(logFC, -log10(FDR), colour=gc_bound, size=chip_peaks)) +
  labs(title=&quot;GR binding activity for diff. expressed PROTAC genes&quot;) +
  ggrepel::geom_text_repel(data=plt_data, aes(logFC, -log10(FDR), label=Gene, size=2)) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/02_exploratory_analysis.Rmd/Q8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Q8-1">
Past versions of Q8-1.png
</button>
</p>
<div id="fig-Q8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/figure/02_exploratory_analysis.Rmd/Q8-1.png" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># CH25H was not found. Is it in the RNA-seq experiment?
se &lt;- readRDS(&quot;../data/EOS_files/Q1_SummExp.rds&quot;)
&quot;ENSG00000138135&quot; %in% rowData(se)$gene_id</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code># We also have to check the other way around by looking at all the peaks.
# If it was in the peaks then it could not be merged onto the RNA data
peakfreq &lt;- readRDS(file=&quot;../data/EOS_files/Q2_gene_peaks.rds&quot;)
peakfreq &lt;- peakfreq[2:dim(peakfreq)[1],]

# we search for the closest match to be sure that its not due to capitalisation or a small typo
peakfreq$Gene[amatch(&quot;CH25H&quot;, peakfreq$Gene, maxDist = Inf)]</code></pre>
<pre><code>[1] C5
2939 Levels:  AACS AADAC AADACL2-AS1 AADACP1 ABCA1 ABCA4 ABCA5 ABCC1 ... ZYG11B</code></pre>
<pre class="r"><code># --&gt; No match for CH25h

# show results
# peakfreq[peakfreq$Gene %in% c(&quot;BRCA1&quot;, &quot;HSD11B1&quot;, &quot;CH25H&quot;),] %&gt;%
#   kableExtra::kable() %&gt;%
#   kableExtra::kable_styling()

plt_data %&gt;%
  kableExtra::kable() %&gt;%
  kableExtra::kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
chip_peaks
</th>
<th style="text-align:left;">
gc_bound
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
118
</td>
<td style="text-align:left;">
HSD11B1
</td>
<td style="text-align:right;">
6.0333836
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
<tr>
<td style="text-align:left;">
210
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:right;">
-0.9559991
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
TRUE
</td>
</tr>
</tbody>
</table>
</div>
<div id="question-6-common-binding-sites-of-junb-and-gr" class="section level1">
<h1>Question 6: Common binding sites of JUNB and GR</h1>
<p><strong>Q: What are common binding sites between JUNB and GR?</strong> -&gt; Generally not a priority though</p>
<p><strong>Take-aways:</strong></p>
<ul>
<li>Interestingly there are more peaks after 2h treatment with DEX than 12h of treatment with DEX</li>
<li>Most of the 12h GR peaks are shared with the JUNB peaks, only a small fraction is unique to the 12h GR ChIP seq data</li>
<li>The 2h GR data has more peaks that are unique and not shared with the 12h GR or 12h JUNB data</li>
</ul>
<pre class="r"><code># Load GR
peaks_gr_12h &lt;- import.bed15(&quot;../data/reddy_bed_DEX12h_r123.bed&quot;, format=&quot;narrowPeak&quot;)
peaks_gr_2h &lt;- import.bed15(&quot;../data/reddy_bed_DEX2h_r123.bed.gz&quot;, format=&quot;narrowPeak&quot;)
# load JUNB
peaks_junb_12h &lt;- import.bed15(&quot;../data/reddy_bed_JUNB_DEX12h_r123.bed&quot;, format=&quot;narrowPeak&quot;)

# Show tile diagram
list_of_regions &lt;- list(GR_2h=peaks_gr_2h, GR_12h=peaks_gr_12h, JUNB_12h=peaks_junb_12h) # Does not work with only two
regionOverlaps(list_of_regions)</code></pre>
<p><img src="figure/02_exploratory_analysis.Rmd/q6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Show upset plot
regionUpset(list_of_regions)</code></pre>
<p><img src="figure/02_exploratory_analysis.Rmd/q6-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="what-if-we-filter-the-peaks-by-their-score" class="section level3">
<h3>What if we filter the peaks by their score?</h3>
<p>There is no significant change in the results/plots if we filter for a minimum peak score of 900 (max value is 1000).</p>
<pre class="r"><code># Does it make a difference if we only choose peaks with high scores?
## What is the range of scores
# qplot(peaks_gr_2h$score, geom=&quot;histogram&quot;)

# Show tile diagram
list_of_regions_v2 &lt;- list(GR_2h=peaks_gr_2h[peaks_gr_2h$score&gt;900], 
                           GR_12h=peaks_gr_12h[peaks_gr_12h$score&gt;900], 
                           JUNB_12h=peaks_junb_12h[peaks_junb_12h$score&gt;900]) # Does not work with only two 
regionOverlaps(list_of_regions_v2)</code></pre>
<p><img src="figure/02_exploratory_analysis.Rmd/q6%20filtered-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Show upset plot
regionUpset(list_of_regions_v2)</code></pre>
<p><img src="figure/02_exploratory_analysis.Rmd/q6%20filtered-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="question-7-pxr-sequences" class="section level1">
<h1>Question 7: PXR Sequences</h1>
<p>Q: Does GC-GR bind to PXR sequences</p>
<p>Other groups have observed that a GR inhibitor activates CYP genes. CYP genes are target genes of PXR TF that have PXR recognition sites. the hypothesis here is that GR might also bind to PXR sites directly or probably is regulating CYP genes via interaction with PXR TF on these genes. For example, based on our RNA-seq dex triggers many CYP genes and the inhibitors too, but protac doesn’t.</p>
<p>The PXR motif looks like this (from Hocomoco):</p>
<p><img src="assets/Q7_1_motif.png" /></p>
<pre class="r"><code>gr_peaks &lt;- readRDS(&quot;../data/EOS_files/Q2_annotated_peaks.rds&quot;)
moi &lt;- readRDS(file=&quot;../data/EOS_files/pxr_peaks.rds&quot;)

cat(paste0(&quot;We have &quot;, length(gr_peaks), &quot; GR peaks, of which &quot;, length(moi), &quot; (&quot;, round(length(moi)/length(gr_peaks),4)*100, &quot;%) contain a PRX motif&quot;))</code></pre>
<pre><code>We have 7920 GR peaks, of which 238 (3.01%) contain a PRX motif</code></pre>
<div id="we-have-7920-gr-peaks-of-which-238-3.01-contain-a-prx-motif" class="section level3">
<h3>We have 7920 GR peaks, of which 238 (3.01%) contain a PRX motif</h3>
<p>We can also look at the signals around the PXR motifs found in the accessible chromatin (identified using the 12h ATAC seq data from the Reddy lab)</p>
<p><img src="assets/Q7_2_signal.png" /></p>
</div>
</div>
<div id="section" class="section level1">
<h1></h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] epiwraps_0.99.29            rtracklayer_1.54.0         
 [3] stringdist_0.9.8            openxlsx_4.2.5             
 [5] hrbrthemes_0.8.6            VennDiagram_1.7.3          
 [7] futile.logger_1.4.3         SEtools_1.8.0              
 [9] edgeR_3.36.0                limma_3.50.3               
[11] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[13] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[15] IRanges_2.28.0              S4Vectors_0.32.4           
[17] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[19] matrixStats_0.62.0          dplyr_1.0.9                
[21] biomaRt_2.50.3              ggplot2_3.3.6              

loaded via a namespace (and not attached):
  [1] utf8_1.2.2               tidyselect_1.1.2         htmlwidgets_1.5.4       
  [4] RSQLite_2.2.14           AnnotationDbi_1.56.2     TSP_1.2-0               
  [7] BiocParallel_1.28.3      Rtsne_0.16               munsell_0.5.0           
 [10] codetools_0.2-18         withr_2.5.0              colorspace_2.0-3        
 [13] filelock_1.0.2           highr_0.9                knitr_1.39              
 [16] rstudioapi_0.13          Rttf2pt1_1.3.10          labeling_0.4.2          
 [19] git2r_0.30.1             GenomeInfoDbData_1.2.7   farver_2.1.0            
 [22] bit64_4.0.5              rprojroot_2.0.3          vctrs_0.4.1             
 [25] generics_0.1.2           lambda.r_1.2.4           xfun_0.31               
 [28] biovizBase_1.42.0        BiocFileCache_2.2.1      R6_2.5.1                
 [31] doParallel_1.0.17        clue_0.3-60              seriation_1.3.5         
 [34] locfit_1.5-9.5           AnnotationFilter_1.18.0  bitops_1.0-7            
 [37] cachem_1.0.6             DelayedArray_0.20.0      assertthat_0.2.1        
 [40] promises_1.2.0.1         BiocIO_1.4.0             scales_1.2.0            
 [43] nnet_7.3-17              gtable_0.3.0             sva_3.42.0              
 [46] ensembldb_2.18.4         workflowr_1.7.0          rlang_1.0.2             
 [49] genefilter_1.76.0        systemfonts_1.0.4        GlobalOptions_0.1.2     
 [52] splines_4.1.0            extrafontdb_1.0          lazyeval_0.2.2          
 [55] dichromat_2.0-0.1        checkmate_2.1.0          yaml_2.3.5              
 [58] backports_1.4.1          GenomicFeatures_1.46.5   httpuv_1.6.5            
 [61] Hmisc_4.7-0              extrafont_0.18           EnrichedHeatmap_1.24.0  
 [64] tools_4.1.0              kableExtra_1.3.4         ellipsis_0.3.2          
 [67] jquerylib_0.1.4          RColorBrewer_1.1-3       plyr_1.8.7              
 [70] Rcpp_1.0.8.3             base64enc_0.1-3          progress_1.2.2          
 [73] zlibbioc_1.40.0          purrr_0.3.4              RCurl_1.98-1.6          
 [76] prettyunits_1.1.1        rpart_4.1.16             viridis_0.6.2           
 [79] GetoptLong_1.0.5         cowplot_1.1.1            GenomicFiles_1.30.0     
 [82] ggrepel_0.9.1            cluster_2.1.3            fs_1.5.2                
 [85] magrittr_2.0.3           magick_2.7.3             data.table_1.14.2       
 [88] futile.options_1.0.1     circlize_0.4.15          whisker_0.4             
 [91] ProtGenerics_1.26.0      randomcoloR_1.1.0.1      hms_1.1.1               
 [94] evaluate_0.15            xtable_1.8-4             XML_3.99-0.9            
 [97] jpeg_0.1-9               gridExtra_2.3            shape_1.4.6             
[100] compiler_4.1.0           tibble_3.1.7             V8_4.1.0                
[103] crayon_1.5.1             htmltools_0.5.2          mgcv_1.8-40             
[106] later_1.3.0              Formula_1.2-4            geneplotter_1.72.0      
[109] DBI_1.1.2                formatR_1.12             dbplyr_2.1.1            
[112] ComplexHeatmap_2.10.0    rappdirs_0.3.3           Matrix_1.4-1            
[115] cli_3.3.0                parallel_4.1.0           Gviz_1.38.4             
[118] pkgconfig_2.0.3          GenomicAlignments_1.30.0 registry_0.5-1          
[121] foreign_0.8-82           xml2_1.3.3               foreach_1.5.2           
[124] svglite_2.1.0            annotate_1.72.0          bslib_0.3.1             
[127] webshot_0.5.3            XVector_0.34.0           rvest_1.0.2             
[130] stringr_1.4.0            VariantAnnotation_1.40.0 digest_0.6.29           
[133] Biostrings_2.62.0        rmarkdown_2.14           htmlTable_2.4.0         
[136] gdtools_0.2.4            restfulr_0.0.13          curl_4.3.2              
[139] Rsamtools_2.10.0         rjson_0.2.21             lifecycle_1.0.1         
[142] nlme_3.1-157             jsonlite_1.8.0           viridisLite_0.4.0       
[145] BSgenome_1.62.0          fansi_1.0.3              pillar_1.7.0            
[148] lattice_0.20-45          KEGGREST_1.34.0          fastmap_1.1.0           
[151] httr_1.4.3               survival_3.3-1           glue_1.6.2              
[154] UpSetR_1.4.0             zip_2.2.0                png_0.1-7               
[157] iterators_1.0.14         bit_4.0.4                stringi_1.7.6           
[160] sass_0.4.1               blob_1.2.3               DESeq2_1.34.0           
[163] latticeExtra_0.6-29      memoise_2.0.1           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
