<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>About</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree207afa653f1bc6b51cf5f07acfe8c498be77c57btargetblank207afa6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/207afa653f1bc6b51cf5f07acfe8c498be77c57b" target="_blank">207afa6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree207afa653f1bc6b51cf5f07acfe8c498be77c57btargetblank207afa6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/207afa653f1bc6b51cf5f07acfe8c498be77c57b" target="_blank">207afa6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/

Untracked files:
    Untracked:  code/helper_functions.R
    Untracked:  data/reddy_bed_ATAC_12h_DEX.bed.gz
    Untracked:  data/reddy_bed_ATAC_control.bed.gz
    Untracked:  data/reddy_bed_DEX12h_r123.bed
    Untracked:  data/reddy_bed_DEX2h_r123.bed.gz
    Untracked:  data/reddy_bed_JUNB_DEX12h_r123.bed
    Untracked:  data/reddy_bigwig_DEX12h_r123.bigWig
    Untracked:  data/reddy_bigwig_DEX2h_r123.bigWig
    Untracked:  data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Untracked:  data/reddy_rna_DEX12h.tsv
    Untracked:  data/reddy_rna_DEX12h_r1.tsv
    Untracked:  data/reddy_rna_DEX12h_r2.tsv
    Untracked:  data/reddy_rna_DEX12h_r3.tsv
    Untracked:  data/reddy_rna_DEX12h_r4.tsv
    Untracked:  data/reddy_rna_DEX12h_transcript.tsv
    Untracked:  data/reddy_rna_control.tsv
    Untracked:  data/reddy_rna_control_r1.tsv
    Untracked:  data/reddy_rna_control_r2.tsv
    Untracked:  data/reddy_rna_control_r3.tsv
    Untracked:  data/reddy_rna_control_r4.tsv
    Untracked:  data/results/
    Untracked:  output/Q2_GR_bound_DE_genes.xlsx
    Untracked:  output/Q4_KH_DE_GR_bound.xlsx

Unstaged changes:
    Modified:   .gitignore
    Modified:   _workflowr.yml
    Modified:   analysis/01_download_files.Rmd
    Modified:   analysis/02_exploratory_analysis.Rmd
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q2-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/Q4-GR-binding-PROTAC-genes.Rmd
    Deleted:    analysis/Q6-JUNB-GR-overlap.Rmd
    Modified:   analysis/Q7-PRX-sequences.Rmd
    Modified:   analysis/about.Rmd
    Modified:   analysis/admin.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/about.Rmd</code>) and HTML (<code>docs/about.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/analysis/about.Rmd" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/about.html" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/59004f167341286081d621dc8a624e7fdc7ab182/analysis/about.Rmd" target="_blank">59004f1</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-05-15
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="todo-lecture-presentation" class="section level1">
<h1>todo lecture presentation</h1>
<p>Q7: Scan open areas (ATAC-seq) for PXR binding (NR1I2). How many of those are bound by GR?</p>
<p>Create boxplots: x: number of peaks y: distribution of logFCs</p>
<p>Plot signals for DE genes with BED data</p>
<p>16 dmso + dex -&gt; could look at that</p>
</div>
<div id="analysing-the-use-of-protacs-to-target-glucorticoid-receptors" class="section level1">
<h1>Analysing the use of PROTACs to target glucorticoid receptors</h1>
<div id="background" class="section level2">
<h2>Background</h2>
<div id="glucorticoids" class="section level3">
<h3>Glucorticoids</h3>
<p>Glucorticoids (a portmanteau of glucose + cortex + steroids; abbreviated GC) are a form of steroids which play an important role as a feedback mechanism in the human immune system. They are generated in the adrenal cortex and are transported throughout the body where they diffuse through the cell membrane of the cell and bind to glucorticoid receptors (GR) which reside in the cytosol. These activated GC-GR complexes can either indirectly or directly influence gene expression. Direct regulation involves homo-dimerisation of the receptor which then acts as a transcription factor (TF) in the nucleus to regulate gene expression (transactivation). In the absence of the GC-GR complex other TFs are able to function normally However, when GR is activated (in form of a GC-GR complex) it can interact with these TFs and hinder them from binding to DNA (transrepression). The GC-GR complex is particularly known for up-regulating the expression of anti-inflammatory proteins and repression of pro-inflammatory genes.</p>
<p>Cortisol, a stress hormone, is the best-known glucorticoid and also the most important one. It is essential for many cardiovascular, metabolic, immunologic and homeostatic functions.</p>
</div>
<div id="protacs" class="section level3">
<h3>PROTACs</h3>
<p>Proteolysis targeting chimeras (PROTACs) are small molecules with two active domains bound together by a so-called linker domain. They are able of catalysing the targeted degradation of proteins. One of the two active domains binds the protein of interest. While inhibitors block the functioning of proteins by binding to the active domain, PROTACs can instead bind to any part of the protein. PROTACs are thus easier to construct in the sense that the active domain can have many forms and need fit the active domain of the target. The other active domain of the PROTAC binds to a protein (ER ubiquitin ligase) which is responsible for adding ubiquitin molecules to proteins, which are read by the cell as marks for destruction. Proteins marked for destruction eventually bind to the proteasome which subsequently destroys said proteins.</p>
<p>Often, to block the functioning of proteins we use inhibitors in biology. However, inhibitors have to bind to the active site of a protein to work, while PROTACs only need to bind to the targeted protein with high selectivity rather than fully block the active domain, as mentioned above. Furthermore, one inhibitor has to stay bound to the targeted protein to block its function. To fully block the protein from functioning, we hence would need at least as many inhibitors as there are proteins that we want to inhibit. PROTACs on the other hand can direct the destruction of multiple proteins before they themselves are degraded. PROTACs can thus be administered at lower dosages compared to inhibitors to achieve similar effects.</p>
<p>Being administerable at a lower dosage, being more selective and not necessarily having to bind to the active domain makes PROTACs attractive drug candidates.</p>
</div>
</div>
<div id="what-is-the-overall-research-goal-of-the-project" class="section level2">
<h2>What is the overall research goal of the project?</h2>
<p>The goal of the project is to understand (1) the contribution of GR to the transcriptional stress response in the hippocampus in vivo and (2) show that PROTACs can be used in-vivo as a tool in neuroscience to normalise stress response.</p>
</div>
<div id="what-has-been-done-so-far" class="section level2">
<h2>What has been done so far?</h2>
<p>Katharina and Mahshid have performed experiments on mice where they treated mice with combinations of (1) a synthetic form of a GC called Dexamethasone (DEX) (2) a PROTAC inhibitor (KH-103) and (3) GR inhibitors. They have collected RNA-seq data to understand what the effects of treatment are - which was analysed by Pierre-Luc - and have determined differentially expressed genes.</p>
<p>[TODO: Part on PL’s analysis]</p>
</div>
<div id="what-is-my-contribution" class="section level2">
<h2>What is my contribution?</h2>
<p>The <a href="http://reddylab.org/">Tim Reddy lab</a> has done related research and has performed RNA-seq, ChIP-seq, and ATAC-seq experiments on the same cell line and under similar conditions. While we only have RNA-seq data, the Reddy lab has ChIP-seq experiments for two proteins, one of which being the GR, and DNA accessibility data.</p>
<p>This subanalysis of the project will analyse the data from the Reddy lab to complement and extend findings from our own experiments.</p>
</div>
<div id="what-data-will-i-be-working-with" class="section level2">
<h2>What data will I be working with?</h2>
<div id="our-data" class="section level3">
<h3>Our data</h3>
<p>Mahshid and Katharina have collected RNA-seq data for the following treatment conditions of</p>
<p><strong>Control treatments: 18h treatment with inhibitor or PROTAC only</strong></p>
<ol style="list-style-type: decimal">
<li>Inhibitor 1: “18h Cort113”</li>
<li>Inhibitor 2: “18h MIF”</li>
<li>PROTAC only: “18h KH-103”“16h DMSO &gt; 2h DEX+DMSO”</li>
</ol>
<p><strong>Switched treatments: 16h+2h</strong></p>
<p>We first treat with the inhibitor, PROTAC or DMSO as control and the add DEX for the last two hours</p>
<ol style="list-style-type: decimal">
<li>Inhibitor 1: “16h Cort113 &gt; 2h DEX+Cort113”</li>
<li>Inhibitor 2: “16h MIF &gt; 2h DEX+MIF”</li>
<li>PROTAC: “16h KH-103 &gt; 2h DEX+KH-103”</li>
<li>Control with DMSO: 16h DMSO &gt; 2h DEX+DMSO"</li>
</ol>
<p><strong>Switched Treatment: 2h+16h</strong></p>
<p>We first treat with DEX for 2 hours and then add the inhibitor PROTAC for another 16 hours</p>
<ol style="list-style-type: decimal">
<li>Inhibitor 1 “2h DEX &gt; 16h DEX+Cort113”</li>
<li>Inhibitor 2: “2h DEX &gt; 16h DEX+MIF”</li>
<li>PROTAC: “2h DEX &gt; 16h DEX+KH-103”</li>
<li>DEX only: “2h DEX &gt; 16h DEX+DMSO”</li>
</ol>
</div>
<div id="reddy-lab-data" class="section level3">
<h3>Reddy lab data</h3>
<p>The experiments from the Reddy lab include:</p>
<ol style="list-style-type: decimal">
<li><strong>RNA - DEX vs control:</strong> One RNA-seq experiment with a control group and dexamethasone treatment for multiple time intervals, up to 12 hours</li>
<li><strong>ChIP - NR3C1 &amp; JUNB:</strong> Two ChIP-seq experiments. One is for NR3C1 - which is as I understand is the only relevant (?) GR in the cell. The second ChIP-seq experiment is for JUNB. JUNB appears to be a transcription factor subunit.</li>
<li><strong>ATAC-seq - DEX exposure:</strong> experiment with a control as well as data for exposure of the cell culture to DEX for multiple intervals.</li>
</ol>
</div>
</div>
<div id="what-will-the-first-questions-be-that-i-will-be-analysing" class="section level2">
<h2>What will the first questions be that I will be analysing?</h2>
<div id="question-1-similarity-of-our-rna-data-with-the-reddy-data" class="section level3">
<h3>Question 1: Similarity of our RNA data with the Reddy data</h3>
<p>(DONE) Q: Which gene expressions have similar patterns of up-regulation (control vs. Dex) in our data compared to the Reddy data?</p>
<p><em>Planned approach:</em> Check which genes are differentially expressed in the Reddy data and create a Venn-diagram for comparable experiments with the differentially expressed genes from our analysis. Also, create a plot</p>
</div>
<div id="question-2-binding-of-gr-gc-complex-to-de-genes-from-our-analysis" class="section level3">
<h3>Question 2: Binding of GR-GC complex to DE genes from our analysis</h3>
<p>(Discuss in call) Q: For the genes that our RNA-seq data shows to be DE after treatment with DEX as well as the inhibitors and control: do we see any direct TF binding to these sites? For which of these is there a difference compared to the control ChIP-seq experiment?</p>
</div>
<div id="question-3-for-our-rna-are-there-any-genes-that-treated-by-the-protac-beforeafter-dex-that-could-be-reversed-but-which-could-not-by-the-inhibitors-or-vice-versa" class="section level3">
<h3>Question 3: For our RNA: are there any genes that treated by the PROTAC before/after DEX that could be reversed but which could not by the inhibitors or vice versa?</h3>
<p>(DONE) We have performed experiments where we first exposed cells to either (1) DEX (2) inhibitors or (3) KH PROTAC for a substantial time and then added inhibitors or the KH PROTAC in case (1) or DEX in cases (2) and (3). We would like to see whether the genes which were activated by DEX but the returned to a normal expression by the KH PROTAC, where a similar treatment by the inhibitors could not lead to such an effect and vice versa.</p>
</div>
<div id="question-4-binding-of-gr-to-kh-protac-treated-de-genes" class="section level3">
<h3>Question 4: Binding of GR to KH PROTAC treated DE genes?</h3>
<p>(DONE) Q: Is there a direct binding of GR to the 13 genes shown to be affected by the KH PROTAC?</p>
</div>
<div id="question-5-binding-to-selected-genes" class="section level3">
<h3>Question 5: Binding to selected genes</h3>
<p>(DONE)Q: Does GR bind to BRCA1, Hsd11b1 and CH25H?</p>
</div>
<div id="question-6-common-binding-sites-of-junb-and-gr" class="section level3">
<h3>Question 6: Common binding sites of JUNB and GR</h3>
<p>Q: What are common binding sites between JUNB and GR? -&gt; Generally not a priority though</p>
</div>
<div id="question-7-pxr-sequences" class="section level3">
<h3>Question 7: PXR Sequences</h3>
<p>Q: Does GC-GR bind to PXR sequences</p>
<p>Scan for PXR sequences. See if GR binds there</p>
</div>
</div>
<div id="abbreviations" class="section level2">
<h2>Abbreviations</h2>
<p>GC = Glucorticoid</p>
<p>GR = Glucorticoid receptor</p>
<p>PROTAC = Proteolysis Targeting Chimera</p>
<p>DEX = Dexamethasone</p>
<p>TF = Transcription Factor</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     rstudioapi_0.13  whisker_0.4      knitr_1.39      
 [5] magrittr_2.0.3   workflowr_1.7.0  R6_2.5.1         rlang_1.0.2     
 [9] fastmap_1.1.0    fansi_1.0.3      stringr_1.4.0    tools_4.1.0     
[13] xfun_0.31        utf8_1.2.2       cli_3.3.0        git2r_0.30.1    
[17] jquerylib_0.1.4  htmltools_0.5.2  ellipsis_0.3.2   rprojroot_2.0.3 
[21] yaml_2.3.5       digest_0.6.29    tibble_3.1.7     lifecycle_1.0.1 
[25] crayon_1.5.1     later_1.3.0      sass_0.4.1       vctrs_0.4.1     
[29] fs_1.5.2         promises_1.2.0.1 glue_1.6.2       evaluate_0.15   
[33] rmarkdown_2.14   stringi_1.7.6    bslib_0.3.1      compiler_4.1.0  
[37] pillar_1.7.0     jsonlite_1.8.0   httpuv_1.6.5     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
