<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree42559356a81583e0cc4d967a24e31f222f9d05detargetblank4255935a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/42559356a81583e0cc4d967a24e31f222f9d05de" target="_blank">4255935</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree42559356a81583e0cc4d967a24e31f222f9d05detargetblank4255935a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/42559356a81583e0cc4d967a24e31f222f9d05de" target="_blank">4255935</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/Formatted_Candidate_ETG_20200305_TAD10Kb.tsv
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Ignored:    data/reddy_rna_DEX2h_r1.tsv
    Ignored:    data/reddy_rna_DEX2h_r2.tsv
    Ignored:    data/reddy_rna_DEX2h_r3.tsv
    Ignored:    data/reddy_rna_DEX2h_r4.tsv
    Ignored:    data/single_replicate_peaks/.DS_Store
    Ignored:    data/single_replicate_peaks/control/
    Ignored:    data/single_replicate_peaks/dex12h/
    Ignored:    data/single_replicate_peaks/dex2h/
    Ignored:    data/single_replicate_peaks/peaks/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/Files/

Untracked files:
    Untracked:  analysis/Q2b-TF-Binding-to-DE-genes-2h.Rmd
    Untracked:  data/00_paper_figures/

Unstaged changes:
    Modified:   analysis/01_Results.Rmd
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q11-MR_paper.Rmd
    Modified:   analysis/Q1b-Gene-overlap-2h.Rmd
    Deleted:    analysis/Q2b-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/admin.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/42559356a81583e0cc4d967a24e31f222f9d05de/analysis/index.Rmd" target="_blank">4255935</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-21
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/42559356a81583e0cc4d967a24e31f222f9d05de/docs/index.html" target="_blank">4255935</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-21
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/748f409e28a1149f7546c4b1cdffb66aea132fcc/analysis/index.Rmd" target="_blank">748f409</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-19
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/d9ccb20416d5cd9a2aa6dee82f3a6c3dd01bec2d/docs/index.html" target="_blank">d9ccb20</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/index.html" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/54fa9d53fcf21807e7b992f7bc59e68f4f5173a7/docs/index.html" target="_blank">54fa9d5</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/analysis/index.Rmd" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/index.html" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1/docs/index.html" target="_blank">779ceb2</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/analysis/index.Rmd" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/index.html" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/analysis/index.Rmd" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/index.html" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/59004f167341286081d621dc8a624e7fdc7ab182/analysis/index.Rmd" target="_blank">59004f1</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-05-15
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level1" number="1">
<h1 number="1"><span class="header-section-number">1</span> Overview</h1>
<p>In this analysis I’ve combined previous work on glucocorticoids and PROTACs (RNA-seq) with external data (RNA-seq, ATAC-seq, ChIP-seq) to better understand the effects of glucocorticoids on DNA regulation. A short introduction is provided in the following.</p>
<p>The <strong>results</strong> are summarised in a separate markdown file <a href="01_Results.html">here</a>, which is also linked in the corresponding section <a href="#results">further below</a>.</p>
<p>For some results only the relevant plots are shown. More verbose analyses were run in separate files. You can find these <strong>methods / individual analyses</strong>in the <a href="#methods">methods section</a> below.</p>
<p>Some resulting tables are accessible as <strong>downloadable files</strong> and can be found at the bottom of this page in the <a href="#downloads">downloads section</a>.</p>
</div>
<div id="introduction" class="section level1" number="2">
<h1 number="2"><span class="header-section-number">2</span> Introduction</h1>
<div id="background" class="section level2" number="2.1">
<h2 number="2.1"><span class="header-section-number">2.1</span> Background</h2>
<div id="glucocorticoids" class="section level3" number="2.1.1">
<h3 number="2.1.1"><span class="header-section-number">2.1.1</span> Glucocorticoids</h3>
<p>Glucocorticoids (a portmanteau of glucose + cortex + steroids; abbreviated GC) are a form of steroids which play an important role as a feedback mechanism in the human immune system. They are generated in the adrenal cortex and are transported throughout the body where they diffuse through the cell membrane and bind to glucocorticoid receptors (GR) which reside in the cytosol.</p>
<p>These activated GC-GR complexes can either indirectly or directly influence gene expression. Direct regulation involves homo-dimerization of the receptor which then acts as a transcription factor (TF) in the nucleus to regulate gene expression (transactivation).</p>
<p>In the absence of the GC-GR complex other TFs are able to function normally, however, when GR is activated (in form of a GC-GR complex) it can interact with these TFs and hinder them from binding to DNA (transrepression). The GC-GR complex is particularly known for up-regulating the expression of anti-inflammatory proteins and repression of pro-inflammatory genes.</p>
<p>Cortisol, a stress hormone, is the best-known glucocorticoid and also the most important one. It is essential for many cardiovascular, metabolic, immunologic and homeostatic functions.</p>
</div>
<div id="protacs" class="section level3" number="2.1.2">
<h3 number="2.1.2"><span class="header-section-number">2.1.2</span> PROTACs</h3>
<p>Proteolysis targeting chimeras (PROTACs) are small molecules with two active domains bound together by a so-called linker domain. PROTAC catalyse the targeted degradation of proteins. One of the two active domains binds the protein of interest. While inhibitors block the functioning of proteins by binding to the active domain, PROTACs can instead bind to any part of the protein. Hence, there are less constraints on synthesising a PROTAC compared to an inhibitor: The active domain of the PROTAC can have many forms and need not fit the active domain of the target, but any part of the molecule. Inhibitors need to bind to the active domain of the target. The other active domain of the PROTAC binds to a protein (ER ubiquitin ligase) which is responsible for adding ubiquitin molecules to proteins, which are read by the cell as marks for destruction. Proteins marked for destruction eventually bind to the proteasome which subsequently destroys said proteins.</p>
<p>Often, to block the functioning of proteins we use inhibitors in biology. However, inhibitors have to bind to the active site of a protein to work, while PROTACs only need to bind to the targeted protein with high selectivity rather than fully block the active domain, as mentioned above. Furthermore, a given inhibitor has to stay bound to the target protein to block its function. To fully block a certain protein in a cell, we would need at least as many inhibitors as there are proteins that we want to inhibit. PROTACs on the other hand can direct the destruction of multiple proteins before the PROTACs themselves are degraded. PROTACs can thus be administered at lower doses compared to inhibitors to achieve similar effects.</p>
<p>PROTACs are attractive drug candidates because they are administrable at a lower dose, can be more selective, and do not necessarily have to bind to the active domain of a target.</p>
</div>
</div>
<div id="overall-research-goal-and-contribution-of-this-analysis" class="section level2" number="2.2">
<h2 number="2.2"><span class="header-section-number">2.2</span> Overall research goal and contribution of this analysis</h2>
<p>The goal of the project is to understand (1) the contribution of GR to the transcriptional stress response in the hippocampus in vivo and (2) show that PROTACs can be used in-vivo as a tool in neuroscience to normalise stress response.</p>
<p>The <a href="http://reddylab.org/">Tim Reddy lab</a> has done related research and has performed RNA-seq, ChIP-seq, and ATAC-seq experiments on the same cell line and under similar conditions. While we only have RNA-seq data, the Reddy lab has ChIP-seq experiments for two proteins, one of which being the GR, and DNA accessibility data.</p>
<p>This analysis contributes to the project by analysing the Reddy data to complement and extend findings from our own experiments. Specifically, it aims to better understand the effect of glucocorticoids on the DNA regulation using the ChIP-seq data and relating it to the RNA-seq data from experiments already performed. Moreover, some components of the analysis answer questions only related to our own RNA-seq data that had not been investigated yet.</p>
</div>
<div id="prior-work" class="section level2" number="2.3">
<h2 number="2.3"><span class="header-section-number">2.3</span> Prior work</h2>
<p>Katharina and Mahshid have performed experiments on mice where they treated mice with combinations of (1) a synthetic form of a GC called Dexamethasone (DEX) (2) a PROTAC inhibitor (KH-103) and (3) GR inhibitors. They have collected RNA-seq data to understand what the effects of treatment are - which was analysed by Pierre-Luc - and have determined differentially expressed genes.</p>
</div>
<div id="data-used" class="section level2" number="2.4">
<h2 number="2.4"><span class="header-section-number">2.4</span> Data used</h2>
<div id="our-data" class="section level3" number="2.4.1">
<h3 number="2.4.1"><span class="header-section-number">2.4.1</span> Our data</h3>
<p>Mahshid and Katharina have collected RNA-seq data for the following treatment conditions of</p>
<p><strong>Control treatments: 18h treatment with inhibitor or PROTAC only</strong></p>
<ol style="list-style-type: decimal">
<li>Inhibitor 1: “18h Cort113”</li>
<li>Inhibitor 2: “18h MIF”</li>
<li>PROTAC only: “18h KH-103”“16h DMSO &gt; 2h DEX+DMSO”</li>
</ol>
<p><strong>Switched treatments: 16h+2h</strong></p>
<p>We first treat with the inhibitor, PROTAC or DMSO as control and the add DEX for the last two hours</p>
<ol style="list-style-type: decimal">
<li>Inhibitor 1: “16h Cort113 &gt; 2h DEX+Cort113”</li>
<li>Inhibitor 2: “16h MIF &gt; 2h DEX+MIF”</li>
<li>PROTAC: “16h KH-103 &gt; 2h DEX+KH-103”</li>
<li>Control with DMSO: 16h DMSO &gt; 2h DEX+DMSO"</li>
</ol>
<p><strong>Switched Treatment: 2h+16h</strong></p>
<p>We first treat with DEX for 2 hours and then add the inhibitor PROTAC for another 16 hours</p>
<ol style="list-style-type: decimal">
<li>Inhibitor 1 “2h DEX &gt; 16h DEX+Cort113”</li>
<li>Inhibitor 2: “2h DEX &gt; 16h DEX+MIF”</li>
<li>PROTAC: “2h DEX &gt; 16h DEX+KH-103”</li>
<li>DEX only: “2h DEX &gt; 16h DEX+DMSO”</li>
</ol>
</div>
<div id="reddy-lab-data" class="section level3" number="2.4.2">
<h3 number="2.4.2"><span class="header-section-number">2.4.2</span> Reddy lab data</h3>
<p>The experiments from the Reddy lab include:</p>
<ol style="list-style-type: decimal">
<li><strong>RNA - DEX vs control:</strong> One RNA-seq experiment with a control group and dexamethasone treatment for multiple time intervals, up to 12 hours</li>
<li><strong>ChIP - NR3C1 &amp; JUNB:</strong> Two ChIP-seq experiments. One is for NR3C1 - which is as I understand is the only relevant (?) GR in the cell. The second ChIP-seq experiment is for JUNB. JUNB appears to be a transcription factor subunit.</li>
<li><strong>ATAC-seq - DEX exposure:</strong> experiment with a control as well as data for exposure of the cell culture to DEX for multiple intervals.</li>
</ol>
</div>
</div>
</div>
<div id="results" class="section level1" number="3">
<h1 number="3"><span class="header-section-number">3</span> Results</h1>
<p>A summary of the results can be found here: <a href="01_Results.html">Results overview</a> <em><font size="1">(Last modified: 2022-09-05 )</font></em></p>
</div>
<div id="methods" class="section level1" number="4">
<h1 number="4"><span class="header-section-number">4</span> Methods / Individual analyses</h1>
<p>For an overview of how the files were downloaded, please see this file:</p>
<ol style="list-style-type: decimal">
<li><a href="./00_Download_files.html">Data download</a> <em><font size="1">(Last modified: 2022-07-08 )</font></em></li>
</ol>
<p>The rendered markdown files for the individual research questions can be found here:</p>
<ol style="list-style-type: decimal">
<li>Q1: <a href="./Q1-Gene-overlap.html">Do the Reddy RNA-seq experiments show the same DE pattern as our experiments? (12h data)</a> <em><font size="1">(Last modified: 2022-09-21 )</font></em></li>
<li>Q1: <a href="./Q1b-Gene-overlap-2h.html">Do the Reddy RNA-seq experiments show the same DE pattern as our experiments? (2h data)</a> <em><font size="1">(Last modified: 2022-09-21 )</font></em></li>
<li>Q2: <a href="./Q2-TF-Binding-to-DE-genes.html">Does the Reddy data show GR binding to DE genes from our analysis? (12h data)</a> <em><font size="1">(Last modified: 2022-09-21 )</font></em></li>
<li>Q2: <a href="./Q2b-TF-Binding-to-DE-genes-2h.html">Does the Reddy data show GR binding to DE genes from our analysis? (2h data)</a> <em><font size="1">(Last modified: 2022-09-21 )</font></em></li>
<li>Q3: <a href="./Q3-PROTAC-reversal.html">Is this the PROTAC able to reverse the effect of DEX on genes where inhibitors don’t work?</a> <em><font size="1">(Last modified: 2022-09-21 )</font></em></li>
<li>Q4: <a href="./Q4-GR-binding-PROTAC-genes.html">Is there a direct binding of GR to the 13 genes shown to be affected by the KH PROTAC?</a> <em><font size="1">(Last modified: 2022-09-21 )</font></em></li>
<li>Q7: <a href="./Q7-PRX-sequences.html">Does GC-GR bind to PRX sequences?</a> <em><font size="1">(Last modified: 2022-07-07 )</font></em></li>
<li>Q9: <a href="./Q9-Differential_binding.html">How does differential binding relate to RNA DE results?</a> <em><font size="1">(Last modified: never )</font></em></li>
<li>Q11: <a href="./Q11-MR_data">MR peaks</a> <em><font size="1">(Last modified: never )</font></em></li>
</ol>
</div>
<div id="downloads" class="section level1" number="5">
<h1 number="5"><span class="header-section-number">5</span> Files available to download</h1>
<p><strong>Q2: Table of 18h DE results augmented with 12h DEX ChIP data:</strong> <a href="./downloads/Q2_GR_bound_DE_genes.xlsx">download</a></p>
<p><strong>Q3: DE results of the 2h&gt;16h and 16h&gt;2h experiments on effect reversal</strong>. It includes data on (1) DMSO control (2) Protac (3) MIF and (4) Cort treatment with columns indicating whether the inhibitor/protac as able to reverse the effect of DEX compared to the DMSO control. Tabs available for the 2h &gt; 16h as well as 16h &gt; 2h treatment <a href="./downloads/Q3_protac_flipped_DE.xlsx">download</a></p>
<p><strong>Q4: 18h KH-103 augmented with 12h GR ChIP data:</strong> <a href="./downloads/Q4_KH_DE_GR_bound.xlsx">download</a></p>
<p><strong>Q9: Differential binding results:</strong> <a href="./downloads/Q9_differential_binding.xlsx">download</a></p>
<p><strong>Q10: DE results from 18h DEX treatment augmented with long-range interactions:</strong> <a href="./downloads/Q10_DE_genes_long_range_interactions.xlsx">download</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] magrittr_2.0.3

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     rstudioapi_0.13  whisker_0.4      knitr_1.39      
 [5] workflowr_1.7.0  here_1.0.1       R6_2.5.1         rlang_1.0.3     
 [9] fastmap_1.1.0    fansi_1.0.3      stringr_1.4.0    tools_4.1.0     
[13] xfun_0.31        utf8_1.2.2       cli_3.3.0        git2r_0.30.1    
[17] jquerylib_0.1.4  htmltools_0.5.2  ellipsis_0.3.2   rprojroot_2.0.3 
[21] yaml_2.3.5       digest_0.6.29    tibble_3.1.7     lifecycle_1.0.1 
[25] crayon_1.5.1     later_1.3.0      sass_0.4.1       vctrs_0.4.1     
[29] fs_1.5.2         promises_1.2.0.1 glue_1.6.2       evaluate_0.15   
[33] rmarkdown_2.14   stringi_1.7.6    bslib_0.3.1      compiler_4.1.0  
[37] pillar_1.7.0     jsonlite_1.8.0   httpuv_1.6.5     pkgconfig_2.0.3 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
