<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />


<title>minimal error</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">minimal error</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">9/29/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>GR-PROTAC-analysis/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree15c3919a15b917c0e8231c668f097e969f7c2b69targetblank15c3919a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/15c3919a15b917c0e8231c668f097e969f7c2b69" target="_blank">15c3919</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree15c3919a15b917c0e8231c668f097e969f7c2b69targetblank15c3919a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/15c3919a15b917c0e8231c668f097e969f7c2b69" target="_blank">15c3919</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store
    Ignored:    data/00_paper_figures/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/Formatted_Candidate_ETG_20200305_TAD10Kb.tsv
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Ignored:    data/reddy_rna_DEX2h_r1.tsv
    Ignored:    data/reddy_rna_DEX2h_r2.tsv
    Ignored:    data/reddy_rna_DEX2h_r3.tsv
    Ignored:    data/reddy_rna_DEX2h_r4.tsv
    Ignored:    data/single_replicate_peaks/.DS_Store
    Ignored:    data/single_replicate_peaks/control/
    Ignored:    data/single_replicate_peaks/dex12h/
    Ignored:    data/single_replicate_peaks/dex2h/
    Ignored:    data/single_replicate_peaks/peaks/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/Files/

Untracked files:
    Untracked:  analysis/logs/
    Untracked:  analysis/minimal error.Rmd
    Untracked:  analysis/minimal-error.Rmd
    Untracked:  analysis/myEnvironment.RData
    Untracked:  data/00_paper_figures/6c1b.pdf
    Untracked:  data/00_paper_figures/6d1b.pdf

Unstaged changes:
    Modified:   analysis/01_Results.Rmd
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q1b-Gene-overlap-2h.Rmd
    Modified:   analysis/Q2-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q2b-TF-Binding-to-DE-genes-2h.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/admin.Rmd
    Deleted:    analysis/docs/assets/#1_venn_diagram.png
    Deleted:    analysis/docs/assets/#2_logFC_Mahshid_Reddy.png
    Modified:   data/00_paper_figures/6c1.pdf
    Modified:   data/00_paper_figures/6c2a.pdf
    Modified:   data/00_paper_figures/6c2b.pdf
    Modified:   data/00_paper_figures/6c2c.pdf
    Modified:   data/00_paper_figures/6c2d.pdf
    Modified:   data/00_paper_figures/6d1.pdf
    Modified:   data/00_paper_figures/6d2a.pdf
    Modified:   data/00_paper_figures/6d2b.pdf
    Modified:   data/00_paper_figures/6d2c.pdf
    Modified:   data/00_paper_figures/6d2d.pdf
    Modified:   data/00_paper_figures/7b1.pdf
    Modified:   data/00_paper_figures/fig7b2.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code># load summarised experiment
se &lt;- readRDS(file=&quot;../data/results/SE.processed.rds&quot;)
# removes the row without name
se &lt;- se[2:dim(se)[1]]</code></pre>
<pre class="r"><code># a function to cluster signal tracks and rna at the same time

cluster_tracks_and_rna &lt;- function(signal_tracks, summ_exp, k, assay_name=&quot;log2FC&quot;){
  # head(unclass(m[[1]]))
  tracks.enrichment.score &lt;- lapply(signal_tracks, enriched_score)
  # playground heatmap clustering
  mat.tracks &lt;- matrix(unlist(tracks.enrichment.score), ncol=length(signal_tracks))
  mat.rna &lt;- as.matrix(assay(summ_exp[,], assay_name))
  # concat horizontally
  mat &lt;- cbind(mat.tracks, mat.rna)
  
  # standardise by column
  standardise &lt;- function(x){
    original_values &lt;- x
    # x is a vector of a column
    trim &lt;- c(0.05, 0.95)
    q &lt;- quantile(x, trim)
    x[x&gt;q[2]] &lt;- q[2]
    x[x&lt;q[1]] &lt;- q[1]
    
    # standardise column
    x.mean &lt;- mean(x)
    std.dev &lt;- sd(x)
    values.standardised &lt;-  (x - x.mean) / std.dev
    
    return(values.standardised)
  }
  
  # standarise all rows
  mat.standardised &lt;- apply(mat, 2, standardise)
  
  # cluster rows
  cl &lt;- kmeans(mat.standardised, centers=k)  
  
  # returns a vector cluster ids
  return(cl$cluster)
}</code></pre>
<pre class="r"><code>tracks &lt;-  list(&quot;2h-DEX&quot;=&quot;../data/reddy_bigwig_DEX2h_r123.bigWig&quot;,
                &quot;6h-DEX&quot;=&quot;../data/reddy_bigwig_DEX6h_r123.bigWig&quot;,
                &quot;12h-DEX&quot;=&quot;../data/reddy_bigwig_DEX12h_r123.bigWig&quot;)

ah &lt;- AnnotationHub()</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre class="r"><code>ensdb &lt;- ah[[&quot;AH98047&quot;]] # AH89211</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>require(&quot;ensembldb&quot;)</code></pre>
<pre><code>Warning: package &#39;GenomicFeatures&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;AnnotationFilter&#39; was built under R version 4.1.1</code></pre>
<pre class="r"><code># all transcripts from the genome. Don&#39;t want to download this for each function call
all.transcripts &lt;-  transcripts(ensdb, columns=c(&quot;tx_id&quot;, &quot;tx_name&quot;, &quot;tx_biotype&quot;,
                                                                &quot;tx_support_level&quot;,
                                                                &quot;gene_id&quot;,&quot;gene_name&quot;))</code></pre>
<pre class="r"><code># peaks &lt;- gr_peaks
genes_subset &lt;- NULL
plot_around_gene_tss &lt;- TRUE
tracks_subset &lt;- c(&quot;12h-DEX&quot;)
subset_columns &lt;- c(&quot;18h untreated&quot;, &quot;18h DMSO&quot;, &quot;2h DEX &gt;\n16h DEX+DMSO&quot;, &quot;2h DEX &gt;\n16h DEX+KH-103&quot;, &quot;2h DEX &gt;\n16h DEX+MIF&quot;, &quot;2h DEX DEX+MIF&quot;, &quot;16h Cort113 &gt;\n2h DEX+Cort113&quot;) 
show_rownames &lt;- FALSE
assayName &lt;- &quot;log2FC&quot;
nclusters &lt;- 3
height &lt;- 13</code></pre>
<pre class="r"><code># first subset for a given selection of genes
if (is.null(genes_subset) == FALSE &amp; plot_around_gene_tss==FALSE) {
  peaks &lt;- peaks[peaks$nearestTSS.gene_name %in% genes_subset]
  if (length(peaks) == 0) return(paste0(&quot;No peaks found for the genes:&quot;, paste(genes_subset, collapse = &#39;, &#39;)))
}

se0 &lt;- se
if (!is.null(subset_columns)){
    se0 &lt;- se0[,se0$condition %in% subset_columns]
    se0$condition &lt;- factor(se0$condition, levels=subset_columns)
  }

if (is.null(tracks_subset)){
  tracks_used &lt;- tracks
} else {
  tracks_used &lt;- tracks[tracks_subset]
}</code></pre>
<pre class="r"><code>mygenes.transcripts &lt;- all.transcripts
if (!is.null(genes_subset)){
  mygenes.transcripts &lt;- subset(mygenes.transcripts, gene_name %in% genes_subset)
} else {
  mygenes.transcripts &lt;- subset(mygenes.transcripts, gene_name %in% rownames(se0))
}</code></pre>
<pre class="r"><code># mygenes.transcripts &lt;- subset(mygenes.transcripts, gene_name %in% peaks$nearestTSS.gene_name)
mygenes.tss &lt;-  resize(mygenes.transcripts, width=1, fix=&#39;start&#39;)
seqlevelsStyle(mygenes.tss) &lt;- &quot;UCSC&quot;</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre class="r"><code>mygenes_unique &lt;- mygenes.tss[!duplicated(mygenes.tss$gene_name),]</code></pre>
<pre class="r"><code># we need to deduplicate again, maybe some gene names describe the same region
mygenes_unique &lt;- mygenes_unique[!duplicated(mygenes_unique)] # 13827
# sum(seqlengths(mygenes_unique) &lt; 0)

# idx &lt;- GenomicRanges:::get_out_of_bound_index(mygenes_unique)
# if (length(idx) != 0L)
#     mygenes_unique &lt;- mygenes_unique[-idx]

# which(end(mygenes_unique) &gt; seqlengths(ensdb)[as.character(seqnames(mygenes_unique))])
# sometimes we might have sequences which extend beyond the genome
mask1 &lt;- which(end(mygenes_unique) &gt; seqlengths(ensdb)[as.character(seqnames(mygenes_unique))])
mask2 &lt;- which(start(mygenes_unique) &lt; seqlengths(ensdb)[as.character(seqnames(mygenes_unique))])
mask &lt;- c(mask1, mask2)
mygenes_unique &lt;- mygenes_unique[-mask]

#which(mygr.original != mygr.trimmed)
# which(mygenes_unique != trim(mygenes_unique))</code></pre>
<pre class="r"><code># get the signals
signals &lt;- signal2Matrix(tracks_used, mygenes_unique, w=5, extend=2000) # 13820</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 2 out-of-bound ranges located on sequence chrM.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.

Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 2 out-of-bound ranges located on sequence chrM.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.

Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 2 out-of-bound ranges located on sequence chrM.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code># make sure that the rows are the same. Sometimes we can lose genes when running signal2matrix
mygenes_unique &lt;- mygenes_unique[mygenes_unique$tx_id %in% rownames(signals[[1]]),]</code></pre>
<pre class="r"><code># adjust the se2 genes to only show the ones for which we have the TSS
genes &lt;- mygenes_unique$gene_name
se2 &lt;- se0[genes,] # 13827</code></pre>
<pre class="r"><code># cluster 
if (!is.null(nclusters)){
  cluster_ids &lt;- cluster_tracks_and_rna(signals, se2, k=nclusters)
  h1 &lt;- plotEnrichedHeatmaps(signals, width=unit(8, &quot;cm&quot;), row_split=cluster_ids)
} else {
  h1 &lt;- plotEnrichedHeatmaps(signals, width=unit(8, &quot;cm&quot;))  
}
set.seed(100) # for colours
h2 &lt;- sechm::sechm(se2, genes=row.names(se2), assayName=assayName, sortRowsOn=NULL, 
                   cluster_rows=FALSE, width=unit(12, &quot;cm&quot;), top_annotation=c(&quot;condition&quot;,&quot;batch&quot;), 
                   gaps_at=&quot;condition&quot;, column_title_gp=gpar(fontsize=10), column_title_rot=90, height=height)</code></pre>
<pre><code>`use_raster` is automatically set to TRUE for a matrix with more than
2000 rows. You can control `use_raster` argument by explicitly setting
TRUE/FALSE to it.

Set `ht_opt$message = FALSE` to turn off this message.</code></pre>
<pre class="r"><code># (For this second one it&#39;s important that rows are in the input order)
h1 + h2</code></pre>
<img src="figure/minimal%20error.Rmd/5-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.18.4            AnnotationFilter_1.18.0    
 [3] GenomicFeatures_1.46.5      AnnotationDbi_1.56.2       
 [5] AnnotationHub_3.2.2         BiocFileCache_2.2.1        
 [7] dbplyr_2.2.1                kableExtra_1.3.4           
 [9] epiwraps_0.99.42            EnrichedHeatmap_1.24.0     
[11] ComplexHeatmap_2.10.0       rtracklayer_1.54.0         
[13] stringdist_0.9.8            openxlsx_4.2.5             
[15] hrbrthemes_0.8.6            VennDiagram_1.7.3          
[17] futile.logger_1.4.3         SEtools_1.8.0              
[19] edgeR_3.36.0                limma_3.50.3               
[21] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[23] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[25] IRanges_2.28.0              S4Vectors_0.32.4           
[27] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[29] matrixStats_0.62.0          dplyr_1.0.9                
[31] biomaRt_2.50.3              ggplot2_3.3.6              

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    tidyselect_1.1.2             
  [3] htmlwidgets_1.5.4             RSQLite_2.2.14               
  [5] TSP_1.2-0                     BiocParallel_1.28.3          
  [7] Rtsne_0.16                    munsell_0.5.0                
  [9] codetools_0.2-18              interp_1.1-2                 
 [11] withr_2.5.0                   colorspace_2.0-3             
 [13] filelock_1.0.2                highr_0.9                    
 [15] knitr_1.39                    rstudioapi_0.13              
 [17] Rttf2pt1_1.3.10               git2r_0.30.1                 
 [19] GenomeInfoDbData_1.2.7        bit64_4.0.5                  
 [21] rprojroot_2.0.3               vctrs_0.4.1                  
 [23] generics_0.1.3                lambda.r_1.2.4               
 [25] xfun_0.31                     biovizBase_1.42.0            
 [27] R6_2.5.1                      doParallel_1.0.17            
 [29] clue_0.3-61                   seriation_1.3.5              
 [31] locfit_1.5-9.5                bitops_1.0-7                 
 [33] cachem_1.0.6                  DelayedArray_0.20.0          
 [35] assertthat_0.2.1              promises_1.2.0.1             
 [37] BiocIO_1.4.0                  scales_1.2.0                 
 [39] nnet_7.3-17                   gtable_0.3.0                 
 [41] sva_3.42.0                    workflowr_1.7.0              
 [43] rlang_1.0.3                   genefilter_1.76.0            
 [45] systemfonts_1.0.4             GlobalOptions_0.1.2          
 [47] splines_4.1.0                 extrafontdb_1.0              
 [49] lazyeval_0.2.2                dichromat_2.0-0.1            
 [51] checkmate_2.1.0               BiocManager_1.30.18          
 [53] yaml_2.3.5                    backports_1.4.1              
 [55] httpuv_1.6.5                  Hmisc_4.7-0                  
 [57] extrafont_0.18                tools_4.1.0                  
 [59] ellipsis_0.3.2                jquerylib_0.1.4              
 [61] RColorBrewer_1.1-3            plyr_1.8.7                   
 [63] Rcpp_1.0.8.3                  base64enc_0.1-3              
 [65] progress_1.2.2                zlibbioc_1.40.0              
 [67] purrr_0.3.4                   RCurl_1.98-1.7               
 [69] prettyunits_1.1.1             rpart_4.1.16                 
 [71] deldir_1.0-6                  GetoptLong_1.0.5             
 [73] cowplot_1.1.1                 GenomicFiles_1.30.0          
 [75] cluster_2.1.3                 fs_1.5.2                     
 [77] magrittr_2.0.3                magick_2.7.3                 
 [79] data.table_1.14.2             futile.options_1.0.1         
 [81] circlize_0.4.15               ProtGenerics_1.26.0          
 [83] randomcoloR_1.1.0.1           mime_0.12                    
 [85] hms_1.1.1                     evaluate_0.15                
 [87] xtable_1.8-4                  XML_3.99-0.10                
 [89] jpeg_0.1-9                    gridExtra_2.3                
 [91] shape_1.4.6                   compiler_4.1.0               
 [93] tibble_3.1.7                  V8_4.2.0                     
 [95] crayon_1.5.1                  htmltools_0.5.2              
 [97] mgcv_1.8-40                   later_1.3.0                  
 [99] Formula_1.2-4                 geneplotter_1.72.0           
[101] DBI_1.1.3                     formatR_1.12                 
[103] rappdirs_0.3.3                Matrix_1.4-1                 
[105] cli_3.3.0                     parallel_4.1.0               
[107] Gviz_1.38.4                   pkgconfig_2.0.3              
[109] GenomicAlignments_1.30.0      registry_0.5-1               
[111] foreign_0.8-82                xml2_1.3.3                   
[113] foreach_1.5.2                 svglite_2.1.0                
[115] annotate_1.72.0               bslib_0.3.1                  
[117] webshot_0.5.3                 XVector_0.34.0               
[119] rvest_1.0.2                   stringr_1.4.0                
[121] VariantAnnotation_1.40.0      digest_0.6.29                
[123] Biostrings_2.62.0             rmarkdown_2.14               
[125] htmlTable_2.4.0               gdtools_0.2.4                
[127] restfulr_0.0.15               curl_4.3.2                   
[129] shiny_1.7.1                   Rsamtools_2.10.0             
[131] rjson_0.2.21                  lifecycle_1.0.1              
[133] nlme_3.1-158                  jsonlite_1.8.0               
[135] viridisLite_0.4.0             BSgenome_1.62.0              
[137] fansi_1.0.3                   pillar_1.7.0                 
[139] lattice_0.20-45               KEGGREST_1.34.0              
[141] fastmap_1.1.0                 httr_1.4.3                   
[143] survival_3.3-1                interactiveDisplayBase_1.32.0
[145] glue_1.6.2                    UpSetR_1.4.0                 
[147] zip_2.2.0                     sechm_1.2.0                  
[149] png_0.1-7                     iterators_1.0.14             
[151] BiocVersion_3.14.0            bit_4.0.4                    
[153] stringi_1.7.6                 sass_0.4.1                   
[155] blob_1.2.3                    DESeq2_1.34.0                
[157] latticeExtra_0.6-30           memoise_2.0.1                </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
