<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-09-22" />

<title>Analysis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-09-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree9d91c8d7b54856de5a773a3a2fc6f6047bdb01f7targetblank9d91c8da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/9d91c8d7b54856de5a773a3a2fc6f6047bdb01f7" target="_blank">9d91c8d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree9d91c8d7b54856de5a773a3a2fc6f6047bdb01f7targetblank9d91c8da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/9d91c8d7b54856de5a773a3a2fc6f6047bdb01f7" target="_blank">9d91c8d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/Formatted_Candidate_ETG_20200305_TAD10Kb.tsv
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Ignored:    data/reddy_rna_DEX2h_r1.tsv
    Ignored:    data/reddy_rna_DEX2h_r2.tsv
    Ignored:    data/reddy_rna_DEX2h_r3.tsv
    Ignored:    data/reddy_rna_DEX2h_r4.tsv
    Ignored:    data/single_replicate_peaks/.DS_Store
    Ignored:    data/single_replicate_peaks/control/
    Ignored:    data/single_replicate_peaks/dex12h/
    Ignored:    data/single_replicate_peaks/dex2h/
    Ignored:    data/single_replicate_peaks/peaks/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/Files/

Untracked files:
    Untracked:  data/00_paper_figures/6c2a.pdf
    Untracked:  data/00_paper_figures/6c2b.pdf
    Untracked:  data/00_paper_figures/6c2c.pdf
    Untracked:  data/00_paper_figures/6c2d.pdf
    Untracked:  data/00_paper_figures/6d2a.pdf
    Untracked:  data/00_paper_figures/6d2b.pdf
    Untracked:  data/00_paper_figures/6d2c.pdf
    Untracked:  data/00_paper_figures/6d2d.pdf

Unstaged changes:
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q11-MR_paper.Rmd
    Modified:   analysis/Q1b-Gene-overlap-2h.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/Q4-GR-binding-PROTAC-genes.Rmd
    Modified:   analysis/admin.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Q11-MR_paper.Rmd</code>) and HTML (<code>docs/Q11-MR_paper.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/9d91c8d7b54856de5a773a3a2fc6f6047bdb01f7/analysis/Q11-MR_paper.Rmd" target="_blank">9d91c8d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-21
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/42559356a81583e0cc4d967a24e31f222f9d05de/analysis/Q11-MR_paper.Rmd" target="_blank">4255935</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-21
</td>
<td>
updates september
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-chip-data" class="section level1">
<h1>Load chip data</h1>
<p>Fig 7b.3. Looking into MR data if for example MR would be bound to most of the triggered genes in conditions (inhibitor &gt; 2h Dex)</p>
<p><strong>Message:</strong> In the absence of GR, MR is triggering most of those genes). For example if in 16h KH-103 &gt; 2h dex condition DE genes would have an MR peak on their promoter region or on their enhancer region</p>
<pre class="r"><code># Needed to annotate peaks
ah &lt;- AnnotationHub()</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre class="r"><code># using the rat reference genome
ensdb &lt;- ah[[&quot;AH95846&quot;]] # AH89211 (mouse), AH98047 (human), AH95846 (rat)</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>require(&quot;ensembldb&quot;)</code></pre>
<pre><code>Warning: package &#39;GenomicFeatures&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.1.1</code></pre>
<pre><code>Warning: package &#39;AnnotationFilter&#39; was built under R version 4.1.1</code></pre>
<pre class="r"><code>folder.path &lt;- &quot;../data/MR-data&quot;
file.names &lt;- dir(folder.path)
file.names &lt;- file.names[sapply(file.names, function(x){stringr::str_detect(x, &quot;.gz&quot;)})]
mr.chip &lt;- lapply(file.names, function(filename){
  file.path = paste0(folder.path, &quot;/&quot;, filename)
  peaks &lt;- import.bed15(file.path, format=&quot;narrowPeak&quot;)
})

# Change the names of the list to the filenames
names(mr.chip) &lt;- file.names

# merge the peaks
peaks.merged &lt;- reduce(unlist(GRangesList(mr.chip)), with.revmap=TRUE)

# annotate peaks
annotated_peaks &lt;- annotateRegions(peaks.merged, ensdb)</code></pre>
<pre><code>Warning in .normarg_seqlevelsStyle(value): more than one seqlevels style
supplied, using the 1st one only</code></pre>
<pre class="r"><code># save the annnotated peaks
saveRDS(annotated_peaks, file=&quot;../data/MR-data/00_peaks_annotated.rds&quot;)
annotated_peaks &lt;- readRDS(file=&quot;../data/MR-data/00_peaks_annotated.rds&quot;)</code></pre>
<div id="peak-genes-translated-into-human-genes" class="section level3">
<h3>Peak genes translated into human genes</h3>
<pre class="r"><code>human = useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, 
                dataset = &quot;hsapiens_gene_ensembl&quot;,
                host = &quot;https://dec2021.archive.ensembl.org/&quot;)
rat = useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, 
              dataset = &quot;rnorvegicus_gene_ensembl&quot;,
              host = &quot;https://dec2021.archive.ensembl.org/&quot;)

# only keep peaks within 2500bp range
annotated_peaks &lt;-  annotated_peaks[abs(annotated_peaks$distance2nearestTSS) &lt; 2500,]

mr.genes = getLDS(attributes = c(&quot;rgd_symbol&quot;), 
                 filters = &quot;rgd_symbol&quot;, 
                 values = annotated_peaks$nearestTSS.gene_name , 
                 mart = rat, 
                 attributesL = c(&quot;hgnc_symbol&quot;), 
                 martL = human, 
                 uniqueRows=T)</code></pre>
<p>Triggered genes in the conditions inhibitor &gt; 2h DEX</p>
<pre class="r"><code>res_int_all &lt;- readRDS(file=&quot;../data/results/DE_our_data.rds&quot;)
names(res_int_all)</code></pre>
<pre><code> [1] &quot;16h Cort113 &gt; 2h DEX+Cort113&quot; &quot;16h DMSO &gt; 2h DEX+DMSO&quot;      
 [3] &quot;16h KH-103 &gt; 2h DEX+KH-103&quot;   &quot;16h MIF  &gt; 2h DEX+MIF&quot;       
 [5] &quot;18h Cort113&quot;                  &quot;18h KH-103&quot;                  
 [7] &quot;18h MIF&quot;                      &quot;2h DEX &gt; 16h DEX+Cort113&quot;    
 [9] &quot;2h DEX &gt; 16h DEX+DMSO&quot;        &quot;2h DEX &gt; 16h DEX+KH-103&quot;     
[11] &quot;2h DEX &gt; 16h DEX+MIF&quot;        </code></pre>
<pre class="r"><code># Relevant tables are:
#   &quot;16h DMSO &gt; 2h DEX+DMSO&quot;
#   &quot;16h Cort113 &gt; 2h DEX+Cort113&quot; 
#   &quot;16h KH-103 &gt; 2h DEX+KH-103&quot;  
#   &quot;16h MIF  &gt; 2h DEX+MIF&quot; 

gene_list_names &lt;- c(&quot;16h DMSO &gt; 2h DEX+DMSO&quot;, &quot;16h Cort113 &gt; 2h DEX+Cort113&quot;, &quot;16h KH-103 &gt; 2h DEX+KH-103&quot;, &quot;16h MIF  &gt; 2h DEX+MIF&quot;)

results &lt;- data.frame(matrix(ncol = 4, nrow = 0))

for (gene_list_name in gene_list_names){
  df &lt;- res_int_all[[gene_list_name]]
  df &lt;- df[df$FDR &lt; 0.05,] # filter for significant genes
  df$mr.activated &lt;- rownames(df) %in% mr.genes$HGNC.symbol
  
  res &lt;- t(data.frame(c(dim(res_int_all[[gene_list_name]])[1],
               dim(df)[1],
               sum(df$mr.activated),
               round(sum(df$mr.activated) / dim(df)[1],3)
               )))
  results &lt;- rbind(results, res)
}

colnames(results) &lt;- c(&quot;Total_genes&quot;, &quot;Significant_genes&quot;, &quot;Num_Sig_Genes_MR_peak&quot;, &quot;as.perc&quot;)
rownames(results) &lt;- gene_list_names


results %&gt;% 
  kable() %&gt;% 
  kable_styling()</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Total_genes
</th>
<th style="text-align:right;">
Significant_genes
</th>
<th style="text-align:right;">
Num_Sig_Genes_MR_peak
</th>
<th style="text-align:right;">
as.perc
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
16h DMSO &gt; 2h DEX+DMSO
</td>
<td style="text-align:right;">
13936
</td>
<td style="text-align:right;">
1505
</td>
<td style="text-align:right;">
113
</td>
<td style="text-align:right;">
0.075
</td>
</tr>
<tr>
<td style="text-align:left;">
16h Cort113 &gt; 2h DEX+Cort113
</td>
<td style="text-align:right;">
13936
</td>
<td style="text-align:right;">
952
</td>
<td style="text-align:right;">
62
</td>
<td style="text-align:right;">
0.065
</td>
</tr>
<tr>
<td style="text-align:left;">
16h KH-103 &gt; 2h DEX+KH-103
</td>
<td style="text-align:right;">
13936
</td>
<td style="text-align:right;">
67
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
0.134
</td>
</tr>
<tr>
<td style="text-align:left;">
16h MIF &gt; 2h DEX+MIF
</td>
<td style="text-align:right;">
13936
</td>
<td style="text-align:right;">
666
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:right;">
0.068
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>plBiomart &lt;- function(glist,martdataset,fromK,toK,returnTable=F){
  library(biomaRt)
  ensembl = useMart(&quot;ensembl&quot;, dataset = martdataset)
  g2 &lt;- getBM(attributes=c(toK,fromK), filters = fromK,values = glist, mart = ensembl)
  if(returnTable){
    g2 &lt;- g2[!duplicated(g2),]
    tmp &lt;- unique(g2[duplicated(g2[,2]),2])
    g2 &lt;- g2[which( !(g2[,1] == &quot;&quot; &amp; g2[,2] %in% tmp) ),]
    return(g2)
  }
  names(g2) &lt;- c(&quot;toK&quot;,&quot;fromK&quot;)
  allg &lt;- unique(g2$fromK)
  g2 &lt;- g2[g2$toK!=&quot;&quot; &amp; !is.na(g2$toK),]
  lostg &lt;- sum(!(allg %in% g2$fromK))
  if(lostg &gt; 0){
    print(paste(&quot;Warning: &quot;,lostg,&quot;genes were lost during the conversion.&quot;))
  }
  return(unique(g2$toK))
}

#&#39; @export
bm.availableSpecies &lt;- function(){
  c(&quot;human&quot;=&quot;hsa&quot;,&quot;mouse&quot;=&quot;mmu&quot;)
}
.bm.getEnsembl &lt;- function(sp){
  switch(match.arg(tolower(sp), bm.availableSpecies()),
         hsa=&quot;hsapiens_gene_ensembl&quot;,
         mmu=&quot;mmusculus_gene_ensembl&quot;
  )
}
.bm.getSymbolKey &lt;- function(sp){
  switch(match.arg(tolower(sp), bm.availableSpecies()),
         
  )
}
.bm.getKey &lt;- function(type, sp){
  type &lt;- match.arg(tolower(type), c(&quot;entrez&quot;,&quot;symbol&quot;,&quot;ensembl&quot;,&quot;refseq&quot;,&quot;uniprot&quot;))
  sp &lt;- match.arg(tolower(sp), bm.availableSpecies())
  switch( type,
          entrez=&quot;entrezgene&quot;,
          symbol=switch(sp, hsa=&quot;hgnc_symbol&quot;, mmu=&quot;mgi_symbol&quot;),
          ensembl=&quot;ensembl_gene_id&quot;,
          refseq=&quot;refseq_mrna&quot;,
          uniprot=&quot;uniprot_swissprot&quot;
  )
}


#&#39; bm.idMap
#&#39;
#&#39; @param ids A vector of ids to be converted.
#&#39; @param from The ID type of `ids`. Possible types are &quot;entrez&quot;,&quot;symbol&quot;,&quot;ensembl&quot;,&quot;refseq&quot;,&quot;uniprot&quot;
#&#39; @param to The desired ID type. Possible types are &quot;entrez&quot;,&quot;symbol&quot;,&quot;ensembl&quot;,&quot;refseq&quot;,&quot;uniprot&quot;
#&#39; @param species The species (see `bm.availableSpecies()`). Default &#39;hsa&#39;.
#&#39; @param returnTable Logical; whether to return a conversion table (default). If FALSE, the converted IDs are returned.
#&#39;
#&#39; @return Either a table or a vector.
#&#39; @export
bm.idMap &lt;- function(ids, from, to, species=&quot;hsa&quot;, returnTable=TRUE){
  from &lt;- .bm.getKey(from, species)
  to &lt;- .bm.getKey(to, species)
  plBiomart(ids,.bm.getEnsembl(species),from, to, returnTable=returnTable)
}




#&#39; bm.symbol2desc
#&#39;
#&#39; Return descriptions of gene symbols. Requires an internet connection.
#&#39;
#&#39; @param glist a character vector containing the official gene symbols
#&#39; @param species a character specifying the species (see `bm.availableSpecies()`). Default &#39;hsa&#39;.
#&#39;
#&#39; @return A table.
#&#39;
#&#39; @export
bm.symbol2desc &lt;- function(glist, species=&quot;hsa&quot;){
  plBiomart(glist,.bm.getEnsembl(species),.bm.getSymbolKey(species),c(&quot;wikigene_description&quot;,&quot;gene_biotype&quot;),returnTable=T)
}



#&#39; Human to mouse gene symbols
#&#39;
#&#39; Returns the mouse orthologs of human genes. Requires an internet connection.
#&#39;
#&#39; @param glist a character vector containing the HGNC gene symbols
#&#39; @param returnTable logical; whether to return the whole conversion table, rather than just the converted IDs (default).
#&#39;
#&#39; @return Either a character vector containing the converted IDs, or the conversion table if returnTable=TRUE.
#&#39;
#&#39; @export
hs2mm &lt;- function(glist, returnTable=F){
    library(biomaRt)
    human = useMart(&quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)
    mouse = useMart(&quot;ensembl&quot;, dataset = &quot;mmusculus_gene_ensembl&quot;) 
    g2 &lt;- getLDS(attributes = c(&quot;hgnc_symbol&quot;), filters = &quot;hgnc_symbol&quot;, values = glist, mart = human, attributesL = c(&quot;mgi_symbol&quot;), martL = mouse)
    if(returnTable){
        return(g2)
    }
    names(g2) &lt;- c(&quot;fromK&quot;,&quot;toK&quot;)
    allg &lt;- unique(g2$fromK)
    g2 &lt;- g2[g2$toK!=&quot;&quot; &amp; !is.na(g2$toK),]
    lostg &lt;- sum(!(allg %in% g2$fromK))
    if(lostg &gt; 0){
        print(paste(&quot;Warning: &quot;,lostg,&quot;genes were lost during the conversion.&quot;))
    }
    return(unique(g2$toK))    
}


#&#39; Yeast to human gene symbols
#&#39;
#&#39; Returns the human orthologs of yeast genes. Requires an internet connection.
#&#39;
#&#39; @param glist a character vector containing the yeast gene symbols
#&#39; @param returnTable logical; whether to return the whole conversion table, rather than just the converted IDs (default).
#&#39;
#&#39; @return Either a character vector containing the converted IDs, or the conversion table if returnTable=TRUE.
#&#39;
#&#39; @export
yeast2human &lt;- function(glist, returnTable=F){
    library(biomaRt)
    human = useMart(&quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)
    yeast = useMart(&quot;ensembl&quot;, dataset = &quot;scerevisiae_gene_ensembl&quot;) 
    g2 &lt;- getLDS(attributes = c(&quot;external_gene_name&quot;), filters = &quot;external_gene_name&quot;, values = glist, mart = yeast, attributesL = c(&quot;hgnc_symbol&quot;), martL = human)
    if(returnTable){
        return(g2)
    }
    names(g2) &lt;- c(&quot;fromK&quot;,&quot;toK&quot;)
    allg &lt;- unique(g2$fromK)
    g2 &lt;- g2[g2$toK!=&quot;&quot; &amp; !is.na(g2$toK),]
    lostg &lt;- sum(!(allg %in% g2$fromK))
    if(lostg &gt; 0){
        print(paste(&quot;Warning: &quot;,lostg,&quot;genes were lost during the conversion.&quot;))
    }
    return(unique(g2$toK))    
}

#&#39; Mouse to human gene symbols
#&#39;
#&#39; Returns the human orthologs of mouse genes. Requires an internet connection.
#&#39;
#&#39; @param glist a character vector containing the MGI gene symbols
#&#39; @param returnTable logical; whether to return the whole conversion table, rather than just the converted IDs (default).
#&#39;
#&#39; @return Either a character vector containing the converted IDs, or the conversion table if returnTable=TRUE.
#&#39;
#&#39; @export
rat2hs &lt;- function(glist, returnTable=F){
    library(biomaRt)
    human = useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, 
                    dataset = &quot;hsapiens_gene_ensembl&quot;,
                    host=&quot;uswest.ensembl.org&quot;, 
                    ensemblRedirect = FALSE)
    #mouse = useMart(&quot;ensembl&quot;, dataset = &quot;mmusculus_gene_ensembl&quot;)
    rat = useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, 
                  dataset = &quot;rnorvegicus_gene_ensembl&quot;,
                  host=&quot;uswest.ensembl.org&quot;, 
                  ensemblRedirect = FALSE)
    
    # change this to rat and mgi_symbol to rat gene symbol (rgd_symbol)
    g2 &lt;- getLDS(attributes = c(&quot;rgd_symbol&quot;), filters = &quot;rgd_symbol&quot;, values = glist, mart = rat, attributesL = c(&quot;hgnc_symbol&quot;), martL = human)
    if(returnTable){
        return(g2)
    }
    names(g2) &lt;- c(&quot;fromK&quot;,&quot;toK&quot;)
    allg &lt;- unique(g2$fromK)
    g2 &lt;- g2[g2$toK!=&quot;&quot; &amp; !is.na(g2$toK),]
    lostg &lt;- sum(!(allg %in% g2$fromK))
    if(lostg &gt; 0){
        print(paste(&quot;Warning: &quot;,lostg, &quot;genes were lost during the conversion.&quot;))
    }
    return(unique(g2$toK))    
}</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.18.4        AnnotationFilter_1.18.0 GenomicFeatures_1.46.5 
 [4] AnnotationDbi_1.56.2    Biobase_2.54.0          epiwraps_0.99.42       
 [7] EnrichedHeatmap_1.24.0  ComplexHeatmap_2.10.0   biomaRt_2.50.3         
[10] kableExtra_1.3.4        AnnotationHub_3.2.2     BiocFileCache_2.2.1    
[13] dbplyr_2.2.1            rtracklayer_1.54.0      GenomicRanges_1.46.1   
[16] GenomeInfoDb_1.30.1     IRanges_2.28.0          S4Vectors_0.32.4       
[19] BiocGenerics_0.40.0    

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    tidyselect_1.1.2             
  [3] RSQLite_2.2.14                htmlwidgets_1.5.4            
  [5] BiocParallel_1.28.3           munsell_0.5.0                
  [7] codetools_0.2-18              interp_1.1-2                 
  [9] colorspace_2.0-3              filelock_1.0.2               
 [11] highr_0.9                     knitr_1.39                   
 [13] rstudioapi_0.13               Rttf2pt1_1.3.10              
 [15] MatrixGenerics_1.6.0          git2r_0.30.1                 
 [17] GenomeInfoDbData_1.2.7        bit64_4.0.5                  
 [19] rprojroot_2.0.3               vctrs_0.4.1                  
 [21] generics_0.1.3                xfun_0.31                    
 [23] biovizBase_1.42.0             R6_2.5.1                     
 [25] doParallel_1.0.17             clue_0.3-61                  
 [27] locfit_1.5-9.5                bitops_1.0-7                 
 [29] cachem_1.0.6                  DelayedArray_0.20.0          
 [31] assertthat_0.2.1              promises_1.2.0.1             
 [33] BiocIO_1.4.0                  scales_1.2.0                 
 [35] nnet_7.3-17                   gtable_0.3.0                 
 [37] workflowr_1.7.0               rlang_1.0.3                  
 [39] systemfonts_1.0.4             GlobalOptions_0.1.2          
 [41] splines_4.1.0                 extrafontdb_1.0              
 [43] lazyeval_0.2.2                dichromat_2.0-0.1            
 [45] checkmate_2.1.0               BiocManager_1.30.18          
 [47] yaml_2.3.5                    backports_1.4.1              
 [49] httpuv_1.6.5                  Hmisc_4.7-0                  
 [51] extrafont_0.18                tools_4.1.0                  
 [53] ggplot2_3.3.6                 ellipsis_0.3.2               
 [55] jquerylib_0.1.4               RColorBrewer_1.1-3           
 [57] Rcpp_1.0.8.3                  plyr_1.8.7                   
 [59] base64enc_0.1-3               progress_1.2.2               
 [61] zlibbioc_1.40.0               purrr_0.3.4                  
 [63] RCurl_1.98-1.7                prettyunits_1.1.1            
 [65] rpart_4.1.16                  deldir_1.0-6                 
 [67] GetoptLong_1.0.5              cowplot_1.1.1                
 [69] GenomicFiles_1.30.0           SummarizedExperiment_1.24.0  
 [71] cluster_2.1.3                 hrbrthemes_0.8.6             
 [73] fs_1.5.2                      magrittr_2.0.3               
 [75] data.table_1.14.2             circlize_0.4.15              
 [77] whisker_0.4                   ProtGenerics_1.26.0          
 [79] matrixStats_0.62.0            hms_1.1.1                    
 [81] mime_0.12                     evaluate_0.15                
 [83] xtable_1.8-4                  XML_3.99-0.10                
 [85] jpeg_0.1-9                    gridExtra_2.3                
 [87] shape_1.4.6                   compiler_4.1.0               
 [89] tibble_3.1.7                  crayon_1.5.1                 
 [91] htmltools_0.5.2               later_1.3.0                  
 [93] Formula_1.2-4                 DBI_1.1.3                    
 [95] rappdirs_0.3.3                Matrix_1.4-1                 
 [97] cli_3.3.0                     parallel_4.1.0               
 [99] Gviz_1.38.4                   pkgconfig_2.0.3              
[101] GenomicAlignments_1.30.0      foreign_0.8-82               
[103] xml2_1.3.3                    foreach_1.5.2                
[105] svglite_2.1.0                 bslib_0.3.1                  
[107] webshot_0.5.3                 XVector_0.34.0               
[109] rvest_1.0.2                   stringr_1.4.0                
[111] VariantAnnotation_1.40.0      digest_0.6.29                
[113] Biostrings_2.62.0             rmarkdown_2.14               
[115] htmlTable_2.4.0               edgeR_3.36.0                 
[117] gdtools_0.2.4                 restfulr_0.0.15              
[119] curl_4.3.2                    shiny_1.7.1                  
[121] Rsamtools_2.10.0              rjson_0.2.21                 
[123] lifecycle_1.0.1               jsonlite_1.8.0               
[125] viridisLite_0.4.0             limma_3.50.3                 
[127] BSgenome_1.62.0               fansi_1.0.3                  
[129] pillar_1.7.0                  lattice_0.20-45              
[131] KEGGREST_1.34.0               fastmap_1.1.0                
[133] httr_1.4.3                    survival_3.3-1               
[135] interactiveDisplayBase_1.32.0 glue_1.6.2                   
[137] UpSetR_1.4.0                  png_0.1-7                    
[139] iterators_1.0.14              BiocVersion_3.14.0           
[141] bit_4.0.4                     stringi_1.7.6                
[143] sass_0.4.1                    blob_1.2.3                   
[145] latticeExtra_0.6-30           memoise_2.0.1                
[147] dplyr_1.0.9                  </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
