<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-06-22" />

<title>Q2: TF Binding to DE genes</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Q2: TF Binding to DE genes</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-06-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree207afa653f1bc6b51cf5f07acfe8c498be77c57btargetblank207afa6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/207afa653f1bc6b51cf5f07acfe8c498be77c57b" target="_blank">207afa6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree207afa653f1bc6b51cf5f07acfe8c498be77c57btargetblank207afa6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/207afa653f1bc6b51cf5f07acfe8c498be77c57b" target="_blank">207afa6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/

Untracked files:
    Untracked:  code/helper_functions.R
    Untracked:  data/reddy_bed_ATAC_12h_DEX.bed.gz
    Untracked:  data/reddy_bed_ATAC_control.bed.gz
    Untracked:  data/reddy_bed_DEX12h_r123.bed
    Untracked:  data/reddy_bed_DEX2h_r123.bed.gz
    Untracked:  data/reddy_bed_JUNB_DEX12h_r123.bed
    Untracked:  data/reddy_bigwig_DEX12h_r123.bigWig
    Untracked:  data/reddy_bigwig_DEX2h_r123.bigWig
    Untracked:  data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Untracked:  data/reddy_rna_DEX12h.tsv
    Untracked:  data/reddy_rna_DEX12h_r1.tsv
    Untracked:  data/reddy_rna_DEX12h_r2.tsv
    Untracked:  data/reddy_rna_DEX12h_r3.tsv
    Untracked:  data/reddy_rna_DEX12h_r4.tsv
    Untracked:  data/reddy_rna_DEX12h_transcript.tsv
    Untracked:  data/reddy_rna_control.tsv
    Untracked:  data/reddy_rna_control_r1.tsv
    Untracked:  data/reddy_rna_control_r2.tsv
    Untracked:  data/reddy_rna_control_r3.tsv
    Untracked:  data/reddy_rna_control_r4.tsv
    Untracked:  data/results/
    Untracked:  output/Q2_GR_bound_DE_genes.xlsx
    Untracked:  output/Q4_KH_DE_GR_bound.xlsx

Unstaged changes:
    Modified:   .gitignore
    Modified:   _workflowr.yml
    Modified:   analysis/01_download_files.Rmd
    Modified:   analysis/02_exploratory_analysis.Rmd
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q2-TF-Binding-to-DE-genes.Rmd
    Deleted:    analysis/Q6-JUNB-GR-overlap.Rmd
    Modified:   analysis/Q7-PRX-sequences.Rmd
    Modified:   analysis/about.Rmd
    Modified:   analysis/admin.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Q2-TF-Binding-to-DE-genes.Rmd</code>) and HTML (<code>docs/Q2-TF-Binding-to-DE-genes.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/analysis/Q2-TF-Binding-to-DE-genes.Rmd" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/Q2-TF-Binding-to-DE-genes.html" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="question-2-does-the-reddy-data-show-gr-binding-to-de-genes-from-our-analysis" class="section level3">
<h3>Question 2: Does the Reddy data show GR binding to DE genes from our analysis</h3>
<p>(Binding of GR-GC complex to DE genes from our analysis)</p>
<p>Q: For the genes that our RNA-seq data shows to be DE after treatment with DEX as well as the inhibitors and control: do we see any direct TF binding to these sites? For which of these is there a difference compared to the control ChIP-seq experiment?</p>
<p><em>Planned approach:</em> We first identify which genes the GC-GR complex binds to and then whether these genes are in our list of DE genes.</p>
<p><em>Steps:</em></p>
<ol style="list-style-type: decimal">
<li>Choose data to download –&gt; BED files for Reddy DEX12h experiment</li>
<li>Do we need the control tracks? –&gt; No, BED tracks have already been constructed</li>
<li>Identify binding sites of TF</li>
<li>Load Mahshid’s DE genes</li>
</ol>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>peaks &lt;- import.bed15(&quot;../data/reddy_bed_DEX12h_r123.bed&quot;, format=&quot;narrowPeak&quot;)</code></pre>
<p>Peak annotation</p>
<pre class="r"><code>ah &lt;- AnnotationHub()</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre class="r"><code>ensdb &lt;- ah[[&quot;AH98047&quot;]] # AH89211</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>require(&quot;ensembldb&quot;)</code></pre>
<pre><code>Warning: package &#39;GenomicFeatures&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.1.1</code></pre>
<pre><code>Warning: package &#39;AnnotationFilter&#39; was built under R version 4.1.1</code></pre>
<pre class="r"><code>peakAnno &lt;- annotateRegions(peaks, ensdb)</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre class="r"><code>saveRDS(peakAnno, &quot;../data/EOS_files/Q2_annotated_peaks.rds&quot;)
table(peakAnno$class)</code></pre>
<pre><code>
                 exonic              intergenic                intronic 
                    348                    2111                    3615 
      proximal &lt;=1000bp proximal &gt;1000&amp;&lt;=2500bp                     TSS 
                    669                     738                     439 </code></pre>
<pre class="r"><code>peakfreq &lt;- as.data.frame(table(peakAnno$nearestTSS.gene_name)) %&gt;% mutate_all(na_if,&quot;&quot;)
colnames(peakfreq) &lt;- c(&quot;Gene&quot;, &quot;Freq&quot;)

saveRDS(file=&quot;../data/EOS_files/Q2_gene_peaks.rds&quot;, peakfreq)</code></pre>
</div>
<div id="load-mahshids-de-genes" class="section level1">
<h1>Load Mahshid’s DE genes</h1>
<pre class="r"><code>res_int_all &lt;- readRDS(file=&quot;../data/results/DE_our_data.rds&quot;)
# names(res_int_all)
res_int &lt;- res_int_all$`2h DEX &gt; 16h DEX+DMSO`
res_int_significant &lt;- res_int[res_int$FDR &lt; 0.05,]</code></pre>
<pre class="r"><code># Merge the frequency count onto the DE table
res_int &lt;- left_join(res_int %&gt;% mutate(Gene = rownames(res_int)), 
          peakfreq[2:nrow(peakfreq),],
          by=&quot;Gene&quot;) %&gt;%
  tidyr::replace_na(list(Freq=0)) %&gt;%
  mutate(gc_bound = Freq &gt; 0) %&gt;%
  rename(chip_peaks=Freq) %&gt;%
  relocate(Gene)

# add a binary tag for significance
res_int$is_significant &lt;- res_int$FDR &lt; 0.05

# What is the fraction of significant DE genes that have GC-GR bound to them?
table(res_int$is_significant, res_int$gc_bound)</code></pre>
<pre><code>       
        FALSE TRUE
  FALSE  6856  492
  TRUE   5486 1102</code></pre>
<pre class="r"><code># Of all the genes where GC-GR is bound, what is the distribution of (a) p.values and (b) logFC. How is this different for genes where GC-GR did not bind?

plt_value &lt;- ggplot(res_int, aes(x=logFC)) +
  # geom_histogram(aes(y=..density.., fill=gc_bound)) +
  geom_density(aes(y=..density.., fill=gc_bound), alpha=0.5) +
  xlim(c(-3,3)) +
  geom_vline(xintercept=0) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of logFC values&quot;, subtitle=&quot;DE genes from Mahshid&#39;s experiments; \nbinding from Reddy data&quot;,
       y=&quot;Density&quot;)
plt_value</code></pre>
<pre><code>Warning: Removed 79 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/Q2-TF-Binding-to-DE-genes.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/figure/Q2-TF-Binding-to-DE-genes.Rmd/unnamed-chunk-3-1.png" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plt_logfc &lt;- ggplot(res_int, aes(x=FDR)) +
  # geom_histogram(aes(y=..density.., fill=gc_bound)) +
  geom_density(aes(y=..density.., group=gc_bound, fill=gc_bound), alpha=0.5) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of p.values&quot;, subtitle=&quot;DE genes from Mahshid&#39;s experiments; \nbinding from Reddy data&quot;,
       y=&quot;Density&quot;)
plt_logfc</code></pre>
<p><img src="figure/Q2-TF-Binding-to-DE-genes.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/figure/Q2-TF-Binding-to-DE-genes.Rmd/unnamed-chunk-3-2.png" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="same-plots-for-reddy-data" class="section level3">
<h3>Same plots for Reddy data</h3>
<p>Load Reddy DE genes</p>
<pre class="r"><code>EOS1 &lt;- readRDS(file=&quot;../data/EOS_files/Q1_EOS.rds&quot;)
res_reddy &lt;- EOS1[[&quot;Reddy_all&quot;]]
# res_reddy_significant &lt;- res_reddy[res_reddy$FDR &lt; 0.05,]

# Merge the frequency count onto the DE table and create binary variable
res_reddy &lt;- left_join(res_reddy %&gt;% mutate(Gene = rownames(res_reddy)), 
          peakfreq[2:nrow(peakfreq),],
          by=&quot;Gene&quot;) %&gt;%
  tidyr::replace_na(list(Freq=0)) %&gt;%
  mutate(gc_bound = Freq &gt; 0) %&gt;%
  rename(chip_peaks=Freq) %&gt;% 
  dplyr::select(-Gene)

# add a binary tag for significance
res_reddy$is_significant &lt;- res_reddy$FDR &lt; 0.05

# What is the fraction significantly DE genes have GC-GR bound to them?
table(res_reddy$is_significant, res_reddy$gc_bound)</code></pre>
<pre><code>       
        FALSE TRUE
  FALSE  7848  579
  TRUE   4164 1000</code></pre>
<pre class="r"><code># Of all the genes where GC-GR is bound, what is the distribution of (a) p.values and (b) logFC. How is this different for genes where GC-GR did not bind?

plt_value_reddy &lt;- ggplot(res_reddy, aes(x=logFC)) +
  # geom_histogram(aes(y=..density.., fill=gc_bound)) +
  geom_density(aes(y=..density.., fill=gc_bound), alpha=0.5) +
  xlim(c(-3,3)) +
  geom_vline(xintercept=0) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of logFC values&quot;, subtitle=&quot;DE genes from Reddy experiments; \nbinding from Reddy data&quot;,
       y=&quot;Density&quot;)

plt_logfc_reddy &lt;- ggplot(res_reddy, aes(x=FDR)) +
  # geom_histogram(aes(y=..density.., fill=gc_bound)) +
  geom_density(aes(y=..density.., group=gc_bound, fill=gc_bound), alpha=0.5) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of p.values&quot;, subtitle=&quot;DE genes from Reddy experiments; \nbinding from Reddy data&quot;,
       y=&quot;Density&quot;)

combined_plot &lt;- ggpubr::ggarrange(plt_value, plt_logfc, plt_value_reddy, plt_logfc_reddy, ncol=2, nrow=2)</code></pre>
<pre><code>Warning: Removed 79 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database</code></pre>
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database</code></pre>
<pre><code>Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database</code></pre>
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database</code></pre>
<pre><code>Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed&#39; not found in PostScript font database</code></pre>
<pre><code>Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database

Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
&#39;Roboto Condensed Light&#39; not found in PostScript font database</code></pre>
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database</code></pre>
<pre><code>Warning: Removed 76 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database

Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family &#39;Roboto Condensed&#39; not found in PostScript font database</code></pre>
<pre class="r"><code>combined_plot</code></pre>
<p><img src="figure/Q2-TF-Binding-to-DE-genes.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/figure/Q2-TF-Binding-to-DE-genes.Rmd/unnamed-chunk-4-1.png" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>peaks &lt;- import.bed15(&quot;../data/reddy_bed_DEX12h_r123.bed&quot;, format=&quot;narrowPeak&quot;)

ensdb &lt;- ah[[&quot;AH98047&quot;]]</code></pre>
<pre><code>loading from cache</code></pre>
<pre class="r"><code>peakAnno &lt;- annotateRegions(peaks, ensdb)</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre class="r"><code>table(peakAnno$class)</code></pre>
<pre><code>
                 exonic              intergenic                intronic 
                    348                    2111                    3615 
      proximal &lt;=1000bp proximal &gt;1000&amp;&lt;=2500bp                     TSS 
                    669                     738                     439 </code></pre>
<pre class="r"><code>peakfreq &lt;- as.data.frame(table(peakAnno$nearestTSS.gene_name)) %&gt;% mutate_all(na_if,&quot;&quot;)
colnames(peakfreq) &lt;- c(&quot;Gene&quot;, &quot;Freq&quot;)

saveRDS(file=&quot;../data/EOS_files/Q2_gene_peaks.rds&quot;, peakfreq)</code></pre>
<p>Save plots and annotated dataframe</p>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q2_1_pvalue_dist.png&quot;, plt_value)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre><code>Warning: Removed 79 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q2_2_logfc_dist.png&quot;, plt_logfc)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q2_3_mahshid_reddy.png&quot;, combined_plot, width=12, height=8)

openxlsx::write.xlsx(file=&quot;../output/Q2_GR_bound_DE_genes.xlsx&quot;, res_int)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ensembldb_2.18.4        AnnotationFilter_1.18.0 GenomicFeatures_1.46.5 
 [4] AnnotationDbi_1.56.2    Biobase_2.54.0          hrbrthemes_0.8.6       
 [7] ggplot2_3.3.6           dplyr_1.0.9             AnnotationHub_3.2.2    
[10] BiocFileCache_2.2.1     dbplyr_2.1.1            rtracklayer_1.54.0     
[13] GenomicRanges_1.46.1    GenomeInfoDb_1.30.1     IRanges_2.28.0         
[16] S4Vectors_0.32.4        BiocGenerics_0.40.0     epiwraps_0.99.29       

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    tidyselect_1.1.2             
  [3] RSQLite_2.2.14                htmlwidgets_1.5.4            
  [5] grid_4.1.0                    BiocParallel_1.28.3          
  [7] munsell_0.5.0                 ragg_1.2.2                   
  [9] codetools_0.2-18              withr_2.5.0                  
 [11] colorspace_2.0-3              filelock_1.0.2               
 [13] highr_0.9                     knitr_1.39                   
 [15] rstudioapi_0.13               ggsignif_0.6.3               
 [17] Rttf2pt1_1.3.10               MatrixGenerics_1.6.0         
 [19] labeling_0.4.2                git2r_0.30.1                 
 [21] GenomeInfoDbData_1.2.7        bit64_4.0.5                  
 [23] farver_2.1.0                  rprojroot_2.0.3              
 [25] vctrs_0.4.1                   generics_0.1.2               
 [27] xfun_0.31                     biovizBase_1.42.0            
 [29] R6_2.5.1                      doParallel_1.0.17            
 [31] clue_0.3-60                   locfit_1.5-9.5               
 [33] bitops_1.0-7                  cachem_1.0.6                 
 [35] DelayedArray_0.20.0           assertthat_0.2.1             
 [37] promises_1.2.0.1              BiocIO_1.4.0                 
 [39] scales_1.2.0                  nnet_7.3-17                  
 [41] gtable_0.3.0                  workflowr_1.7.0              
 [43] rlang_1.0.2                   systemfonts_1.0.4            
 [45] GlobalOptions_0.1.2           splines_4.1.0                
 [47] rstatix_0.7.0                 extrafontdb_1.0              
 [49] lazyeval_0.2.2                dichromat_2.0-0.1            
 [51] broom_0.8.0                   checkmate_2.1.0              
 [53] abind_1.4-5                   BiocManager_1.30.17          
 [55] yaml_2.3.5                    backports_1.4.1              
 [57] httpuv_1.6.5                  Hmisc_4.7-0                  
 [59] extrafont_0.18                EnrichedHeatmap_1.24.0       
 [61] tools_4.1.0                   ellipsis_0.3.2               
 [63] jquerylib_0.1.4               RColorBrewer_1.1-3           
 [65] Rcpp_1.0.8.3                  plyr_1.8.7                   
 [67] base64enc_0.1-3               progress_1.2.2               
 [69] zlibbioc_1.40.0               purrr_0.3.4                  
 [71] RCurl_1.98-1.6                prettyunits_1.1.1            
 [73] ggpubr_0.4.0                  rpart_4.1.16                 
 [75] GetoptLong_1.0.5              cowplot_1.1.1                
 [77] GenomicFiles_1.30.0           SummarizedExperiment_1.24.0  
 [79] cluster_2.1.3                 fs_1.5.2                     
 [81] magrittr_2.0.3                data.table_1.14.2            
 [83] openxlsx_4.2.5                circlize_0.4.15              
 [85] whisker_0.4                   ProtGenerics_1.26.0          
 [87] matrixStats_0.62.0            hms_1.1.1                    
 [89] mime_0.12                     evaluate_0.15                
 [91] xtable_1.8-4                  XML_3.99-0.9                 
 [93] jpeg_0.1-9                    gridExtra_2.3                
 [95] shape_1.4.6                   compiler_4.1.0               
 [97] biomaRt_2.50.3                tibble_3.1.7                 
 [99] crayon_1.5.1                  htmltools_0.5.2              
[101] later_1.3.0                   Formula_1.2-4                
[103] tidyr_1.2.0                   DBI_1.1.2                    
[105] ComplexHeatmap_2.10.0         rappdirs_0.3.3               
[107] car_3.0-13                    Matrix_1.4-1                 
[109] cli_3.3.0                     parallel_4.1.0               
[111] Gviz_1.38.4                   pkgconfig_2.0.3              
[113] GenomicAlignments_1.30.0      foreign_0.8-82               
[115] xml2_1.3.3                    foreach_1.5.2                
[117] bslib_0.3.1                   XVector_0.34.0               
[119] stringr_1.4.0                 VariantAnnotation_1.40.0     
[121] digest_0.6.29                 Biostrings_2.62.0            
[123] rmarkdown_2.14                htmlTable_2.4.0              
[125] edgeR_3.36.0                  gdtools_0.2.4                
[127] restfulr_0.0.13               curl_4.3.2                   
[129] shiny_1.7.1                   Rsamtools_2.10.0             
[131] rjson_0.2.21                  lifecycle_1.0.1              
[133] jsonlite_1.8.0                carData_3.0-5                
[135] viridisLite_0.4.0             limma_3.50.3                 
[137] BSgenome_1.62.0               fansi_1.0.3                  
[139] pillar_1.7.0                  lattice_0.20-45              
[141] KEGGREST_1.34.0               fastmap_1.1.0                
[143] httr_1.4.3                    survival_3.3-1               
[145] interactiveDisplayBase_1.32.0 glue_1.6.2                   
[147] zip_2.2.0                     UpSetR_1.4.0                 
[149] png_0.1-7                     iterators_1.0.14             
[151] BiocVersion_3.14.0            bit_4.0.4                    
[153] stringi_1.7.6                 sass_0.4.1                   
[155] blob_1.2.3                    textshaping_0.3.6            
[157] latticeExtra_0.6-29           memoise_2.0.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
