<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-09-21" />

<title>Q2: TF Binding to DE genes</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Q2: TF Binding to DE genes</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-09-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree42559356a81583e0cc4d967a24e31f222f9d05detargetblank4255935a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/42559356a81583e0cc4d967a24e31f222f9d05de" target="_blank">4255935</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree42559356a81583e0cc4d967a24e31f222f9d05detargetblank4255935a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/42559356a81583e0cc4d967a24e31f222f9d05de" target="_blank">4255935</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/Formatted_Candidate_ETG_20200305_TAD10Kb.tsv
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Ignored:    data/reddy_rna_DEX2h_r1.tsv
    Ignored:    data/reddy_rna_DEX2h_r2.tsv
    Ignored:    data/reddy_rna_DEX2h_r3.tsv
    Ignored:    data/reddy_rna_DEX2h_r4.tsv
    Ignored:    data/single_replicate_peaks/.DS_Store
    Ignored:    data/single_replicate_peaks/control/
    Ignored:    data/single_replicate_peaks/dex12h/
    Ignored:    data/single_replicate_peaks/dex2h/
    Ignored:    data/single_replicate_peaks/peaks/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/Files/

Untracked files:
    Untracked:  analysis/Q2b-TF-Binding-to-DE-genes-2h.Rmd

Unstaged changes:
    Modified:   analysis/01_Results.Rmd
    Deleted:    analysis/Q2b-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/admin.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="question-2-does-the-reddy-data-show-gr-binding-to-de-genes-from-our-analysis" class="section level3">
<h3>Question 2: Does the Reddy data show GR binding to DE genes from our analysis</h3>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>peaks &lt;- import.bed15(&quot;../data/reddy_bed_DEX2h_r123.bed.gz&quot;, format=&quot;narrowPeak&quot;)</code></pre>
<p>Peak annotation</p>
<pre class="r"><code>ah &lt;- AnnotationHub()</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre class="r"><code>ensdb &lt;- ah[[&quot;AH98047&quot;]] # AH89211</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>require(&quot;ensembldb&quot;)</code></pre>
<pre><code>Warning: package &#39;GenomicFeatures&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.1.1</code></pre>
<pre><code>Warning: package &#39;AnnotationFilter&#39; was built under R version 4.1.1</code></pre>
<pre class="r"><code>peakAnno &lt;- annotateRegions(peaks, ensdb)</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre class="r"><code># column only for peaks with 2500np range
peakAnno$nearestTSS.gene_name.2500r &lt;- peakAnno$nearestTSS.gene_name
peakAnno[abs(peakAnno$distance2nearestTSS) &gt;= 2500, ]$nearestTSS.gene_name.2500r &lt;- &quot;&quot;
saveRDS(peakAnno, &quot;../data/EOS_files/Q2_annotated_peaks.rds&quot;)
table(peakAnno$class)</code></pre>
<pre><code>
                 exonic              intergenic                intronic 
                    751                    4416                    7898 
      proximal &lt;=1000bp proximal &gt;1000&amp;&lt;=2500bp                     TSS 
                   1472                    1698                     967 </code></pre>
<pre class="r"><code>peakfreq &lt;- as.data.frame(table(peakAnno$nearestTSS.gene_name)) %&gt;% mutate_all(na_if,&quot;&quot;) %&gt;%
  left_join(as.data.frame(table(peakAnno$nearestTSS.gene_name.2500r)) %&gt;% mutate_all(na_if,&quot;&quot;), by=&quot;Var1&quot;)
colnames(peakfreq) &lt;- c(&quot;Gene&quot;, &quot;Freq&quot;, &quot;Freq_2500peaks&quot;)

saveRDS(file=&quot;../data/EOS_files/Q2_gene_peaks_2h.rds&quot;, peakfreq)</code></pre>
</div>
<div id="load-mahshids-de-genes" class="section level1">
<h1>Load Mahshid’s DE genes</h1>
<pre class="r"><code>res_int_all &lt;- readRDS(file=&quot;../data/results/DE_our_data.rds&quot;)
# names(res_int_all)
res_int &lt;- res_int_all$`16h DMSO &gt; 2h DEX+DMSO`
res_int_significant &lt;- res_int[res_int$FDR &lt; 0.05,]

# Merge the frequency count onto the DE table
res_int &lt;- left_join(res_int %&gt;% mutate(Gene = rownames(res_int)), 
          peakfreq[2:nrow(peakfreq),],
          by=&quot;Gene&quot;) %&gt;%
  tidyr::replace_na(list(Freq=0, Freq_2500peaks=0)) %&gt;%
  mutate(gc_bound = Freq &gt; 0) %&gt;%
  dplyr::rename(chip_peaks=Freq, chip_peaks.2500r=Freq_2500peaks) %&gt;%
  relocate(Gene)

# add a binary tag for significance
res_int$is_significant &lt;- res_int$FDR &lt; 0.05

# What is the fraction of significant DE genes that have GC-GR bound to them?
table(res_int$is_significant, res_int$gc_bound)</code></pre>
<pre><code>       
        FALSE  TRUE
  FALSE 10176  2255
  TRUE    843   662</code></pre>
<pre class="r"><code># Of all the genes where GC-GR is bound, what is the distribution of (a) p.values and (b) logFC. How is this different for genes where GC-GR did not bind?
plt_value &lt;- ggplot(res_int, aes(x=logFC)) +
  # geom_histogram(aes(y=..density.., fill=gc_bound)) +
  geom_density(aes(y=..density.., fill=gc_bound), alpha=0.5) +
  xlim(c(-3,3)) +
  geom_vline(xintercept=0) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of logFC values&quot;, subtitle=&quot;DE genes from Mahshid&#39;s experiments; \nbinding from Reddy data&quot;,
       y=&quot;Density&quot;)
plt_value</code></pre>
<pre><code>Warning: Removed 31 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/Q2b-TF-Binding-to-DE-genes-2h.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plt_logfc &lt;- ggplot(res_int, aes(x=FDR)) +
  # geom_histogram(aes(y=..density.., fill=gc_bound)) +
  geom_density(aes(y=..density.., group=gc_bound, fill=gc_bound), alpha=0.5) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of p.values&quot;, subtitle=&quot;DE genes from Mahshid&#39;s experiments; \nbinding from Reddy data&quot;,
       y=&quot;Density&quot;)
plt_logfc</code></pre>
<p><img src="figure/Q2b-TF-Binding-to-DE-genes-2h.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="same-plots-for-reddy-data" class="section level3">
<h3>Same plots for Reddy data</h3>
<p>Load Reddy DE genes</p>
<pre class="r"><code>EOS1 &lt;- readRDS(file=&quot;../data/EOS_files/Q1_EOS.rds&quot;)
res_reddy &lt;- EOS1[[&quot;Reddy_all&quot;]]
# res_reddy_significant &lt;- res_reddy[res_reddy$FDR &lt; 0.05,]

# Merge the frequency count onto the DE table and create binary variable
res_reddy &lt;- left_join(res_reddy %&gt;% mutate(Gene = rownames(res_reddy)), 
          peakfreq[2:nrow(peakfreq),],
          by=&quot;Gene&quot;) %&gt;%
  tidyr::replace_na(list(Freq=0, Freq_2500peaks=0)) %&gt;%
  mutate(gc_bound = Freq &gt; 0) %&gt;%
  dplyr::rename(chip_peaks=Freq, chip_peaks.2500r=Freq_2500peaks) %&gt;% 
  tibble::column_to_rownames(&quot;Gene&quot;)

# add a binary tag for significance
res_reddy$is_significant &lt;- res_reddy$FDR &lt; 0.05

# What is the fraction significantly DE genes have GC-GR bound to them?
table(res_reddy$is_significant, res_reddy$gc_bound)</code></pre>
<pre><code>       
        FALSE TRUE
  FALSE  7217 1210
  TRUE   3551 1613</code></pre>
<pre class="r"><code># Of all the genes where GC-GR is bound, what is the distribution of (a) p.values and (b) logFC. How is this different for genes where GC-GR did not bind?

plt_value_reddy &lt;- ggplot(res_reddy, aes(x=logFC)) +
  # geom_histogram(aes(y=..density.., fill=gc_bound)) +
  geom_density(aes(y=..density.., fill=gc_bound), alpha=0.5) +
  xlim(c(-3,3)) +
  geom_vline(xintercept=0) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of logFC values&quot;, subtitle=&quot;DE genes from Reddy experiments; \nbinding from Reddy data&quot;,
       y=&quot;Density&quot;)

plt_logfc_reddy &lt;- ggplot(res_reddy, aes(x=FDR)) +
  # geom_histogram(aes(y=..density.., fill=gc_bound)) +
  geom_density(aes(y=..density.., group=gc_bound, fill=gc_bound), alpha=0.5) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of p.values&quot;, subtitle=&quot;DE genes from Reddy experiments; \nbinding from Reddy data&quot;,
       y=&quot;Density&quot;)

combined_plot &lt;- ggpubr::ggarrange(plt_value, plt_logfc, plt_value_reddy, plt_logfc_reddy, ncol=2, nrow=2)</code></pre>
<pre><code>Warning: Removed 31 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 76 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>combined_plot</code></pre>
<p><img src="figure/Q2b-TF-Binding-to-DE-genes-2h.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Save plots</p>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q2_1_pvalue_dist_2h.png&quot;, plt_value)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre><code>Warning: Removed 31 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q2_2_logfc_dist_2h.png&quot;, plt_logfc)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q2_3_mahshid_reddy_2h.png&quot;, combined_plot, width=12, height=8)</code></pre>
</div>
</div>
<div id="boxplot-distribution-of-logfc-by-number-of-nearby-gr-peaks" class="section level1">
<h1>Boxplot: Distribution of logFC by number of nearby GR-peaks</h1>
<p>Each peaks was attributed to the gene of the TSS closest to the peak</p>
<pre class="r"><code>mixed_data &lt;- rbind(as.data.frame(res_reddy) %&gt;% 
        tibble::rownames_to_column(&quot;Gene&quot;) %&gt;%
        dplyr::select(&quot;Gene&quot;, &quot;logFC&quot;, &quot;chip_peaks&quot;, &quot;chip_peaks.2500r&quot;) %&gt;%
        mutate(Source=&quot;Reddy&quot;),
      as.data.frame(res_int) %&gt;%
        dplyr::select(&quot;Gene&quot;, &quot;logFC&quot;, &quot;chip_peaks&quot;, &quot;chip_peaks.2500r&quot;) %&gt;%
        mutate(Source=&quot;Mahshid&quot;)
        ) %&gt;%
  mutate(chip_peaks=factor(chip_peaks, levels=sort(as.numeric(unique(chip_peaks)))),
         chip_peaks.2500r=factor(chip_peaks.2500r, levels=sort(as.numeric(unique(chip_peaks.2500r)))),
         abs_logFC=abs(logFC))


boxplot &lt;- ggplot(mixed_data, aes(x=chip_peaks, y=logFC, colour=Source)) +
  geom_boxplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of logFC by number of nearby GR-peaks&quot;, subtitle=&quot;Closest peak to gene no matter distance&quot;, x=&quot;# ChIP peaks&quot;) +
  ylim(c(-5,5))

boxplot_2500 &lt;- ggplot(mixed_data, aes(x=chip_peaks.2500r, y=logFC, colour=Source)) +
  geom_boxplot() +
  theme_ipsum_rc() +
  labs(title=&quot;&quot;, subtitle=&quot;Peaks within 2500bp range&quot;, x=&quot;# ChIP peaks&quot;) +
  ylim(c(-5,5))

boxplot_abs &lt;- ggplot(mixed_data, aes(x=chip_peaks, y=abs_logFC, colour=Source)) +
  geom_boxplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of ABSOLUTE logFCs by number of nearby GR-peaks&quot;, subtitle=&quot;Closest peak to gene no matter distances&quot;, x=&quot;# ChIP peaks&quot;) +
  ylim(c(0,5))

boxplot_abs_2500 &lt;- ggplot(mixed_data, aes(x=chip_peaks.2500r, y=abs_logFC, colour=Source)) +
  geom_boxplot() +
  theme_ipsum_rc() +
  labs(title=&quot;&quot;, subtitle=&quot;Peaks within 2500bp range&quot;, x=&quot;# ChIP peaks&quot;) +
  ylim(c(0,5))

boxplot &lt;- ggarrange(
  ggarrange(boxplot, boxplot_2500, ncol=2),
  ggarrange(boxplot_abs, boxplot_abs_2500, ncol=2),
  ncol=1)</code></pre>
<pre><code>Warning: Removed 27 rows containing non-finite values (stat_boxplot).
Removed 27 rows containing non-finite values (stat_boxplot).
Removed 27 rows containing non-finite values (stat_boxplot).
Removed 27 rows containing non-finite values (stat_boxplot).</code></pre>
<pre class="r"><code>boxplot</code></pre>
<p><img src="figure/Q2b-TF-Binding-to-DE-genes-2h.Rmd/distribution%20by%20numbers%20of%20peaks-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../docs/assets/Q2_4_boxplots_2h.png&quot;, boxplot, width=14, height=10, unit=&quot;in&quot;)</code></pre>
</div>
<div id="relationship-between-a-logfc-and-the-closest-gr-peak" class="section level1">
<h1>Relationship between a logFC and the closest GR peak</h1>
<pre class="r"><code>data &lt;- as.data.frame(peakAnno) %&gt;%
  dplyr::select(c(&quot;nearestTSS.gene_name&quot;, &quot;score&quot;, &quot;signalValue&quot;, &quot;peak&quot;, &quot;distance2nearestTSS&quot;, &quot;class&quot;)) %&gt;% 
  mutate_all(na_if,&quot;&quot;) %&gt;% 
  dplyr::filter(!is.na(nearestTSS.gene_name)) %&gt;%
  dplyr::rename(Gene = nearestTSS.gene_name)

(plt_data &lt;- data %&gt;% dplyr::left_join(as.data.frame(res_int) %&gt;%
                            dplyr::select(&quot;Gene&quot;, &quot;logFC&quot;, &quot;chip_peaks&quot;, &quot;FDR&quot;, &quot;is_significant&quot;) %&gt;%
                            mutate(Source=&quot;Mahshid&quot;), 
                          by=&quot;Gene&quot;) %&gt;%
  dplyr::group_by(Gene) %&gt;%
  summarise(min_dist2tss=min(abs(distance2nearestTSS)),
            max_dist2tss=max(abs(distance2nearestTSS)),
            mean_dist2tss=mean(abs(distance2nearestTSS)),
            minSignal=min(signalValue),
            maxSignal=max(signalValue),
            n_peaks=n(),
            logFC=logFC,
            is_significant=is_significant) %&gt;%
    distinct() %&gt;%
    dplyr::filter(!is.na(logFC))
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Gene&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 2,917 × 9
# Groups:   Gene [2,917]
   Gene    min_dist2tss max_dist2tss mean_dist2tss minSignal maxSignal n_peaks
   &lt;chr&gt;          &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;   &lt;int&gt;
 1 AACS           10616        10616         10616      98.9      98.9       1
 2 AADAC          22918        22918         22918     110.      110.        1
 3 AADACP1            0         3756          1878      43.4     147.        2
 4 AADAT           7456         7456          7456      35.5      35.5       1
 5 ABCA1          42111        42111         42111      69.2      69.2       1
 6 ABCA12           189          189           189      31.7      31.7       1
 7 ABCA13         30997        30997         30997      28.4      28.4       1
 8 ABCA5            318         2146          1232      24.2      36.3       2
 9 ABCA7           1021         1021          1021      45.9      45.9       1
10 ABCA8            347          347           347      22.8      22.8       1
# … with 2,907 more rows, and 2 more variables: logFC &lt;dbl&gt;,
#   is_significant &lt;lgl&gt;</code></pre>
<pre class="r"><code># Min distance to TSS
min_dist &lt;- ggplot(plt_data, aes(min_dist2tss, logFC, size=n_peaks, colour=is_significant)) +
  geom_point(alpha=0.8) +
  geom_density_2d(bins = 10, colour=&quot;green&quot;) +
  labs(title=&quot;LogFC vs minimum distance peak to TSS&quot;, subtitle=&quot;Size indicates the number of peaks for a given gene&quot;)

# Mean distance to TSS
min_dist2 &lt;- plt_data %&gt;%
  ggplot(aes(log(min_dist2tss), logFC, size=n_peaks, colour=is_significant)) +
  geom_point(alpha=0.8) +
  geom_density_2d(bins = 10, colour=&quot;green&quot;) +
  labs(title=&quot;LogFC vs log(min distance peak to TSS)&quot;, subtitle=&quot;Size indicates the number of peaks for a given gene&quot;) +
  ylim(c(-3,3))

TSS_distances &lt;- ggarrange(min_dist, min_dist2, ncol=1)</code></pre>
<pre><code>Warning: Removed 448 rows containing non-finite values (stat_density2d).</code></pre>
<pre><code>Warning: Removed 17 rows containing missing values (geom_point).</code></pre>
<pre class="r"><code>TSS_distances</code></pre>
<p><img src="figure/Q2b-TF-Binding-to-DE-genes-2h.Rmd/logFC%20vs%20GR%20peak-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../docs/assets/Q2_5_TSS_distances_2h.png&quot;, TSS_distances)</code></pre>
<pre><code>Saving 14 x 18 in image</code></pre>
<pre class="r"><code># Min distance to TSS
min_dist_xlim &lt;- ggplot(plt_data, aes(min_dist2tss, logFC, size=n_peaks, colour=is_significant)) +
  geom_point(alpha=0.8) +
  geom_density_2d(bins = 10, colour=&quot;green&quot;) +
  labs(title=&quot;LogFC vs minimum distance peak to TSS&quot;, subtitle=&quot;Size indicates the number of peaks for a given gene&quot;) +
  xlim(c(0,2500))

# Mean distance to TSS
min_dist2_xlim &lt;- plt_data %&gt;%
  ggplot(aes(log(min_dist2tss), logFC, size=n_peaks, colour=is_significant)) +
  geom_point(alpha=0.8) +
  geom_density_2d(bins = 10, colour=&quot;green&quot;) +
  labs(title=&quot;LogFC vs log(min distance peak to TSS)&quot;, subtitle=&quot;Size indicates the number of peaks for a given gene&quot;) +
  ylim(c(-3,3)) +
  xlim(c(0,log(2500)))

TSS_distances_xlim &lt;- ggarrange(min_dist_xlim, min_dist2_xlim, ncol=1)</code></pre>
<pre><code>Warning: Removed 1597 rows containing non-finite values (stat_density2d).</code></pre>
<pre><code>Warning: Removed 1597 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 2038 rows containing non-finite values (stat_density2d).</code></pre>
<pre><code>Warning: Removed 1607 rows containing missing values (geom_point).</code></pre>
<pre class="r"><code>TSS_distances_xlim</code></pre>
<p><img src="figure/Q2b-TF-Binding-to-DE-genes-2h.Rmd/unnamed-chunk-5-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../docs/assets/Q2_5_TSS_distances_xlim2500_2h.png&quot;, TSS_distances_xlim)</code></pre>
<pre><code>Saving 14 x 18 in image</code></pre>
<pre class="r"><code>ggsave(filename=&quot;../docs/00_paper_figures/6b_2h.png&quot;, TSS_distances_xlim)</code></pre>
<pre><code>Saving 14 x 18 in image</code></pre>
<pre class="r"><code>mixed_data2 &lt;- mixed_data %&gt;% left_join(data %&gt;%
                                         group_by(Gene) %&gt;%
                                         dplyr::slice(which.min(abs(distance2nearestTSS))) %&gt;% 
                                         dplyr::select(&quot;Gene&quot;, &quot;class&quot;) %&gt;%
                                         dplyr::rename(closest_peak_class=class) %&gt;%
                                         mutate(closest_peak_class = factor(closest_peak_class, levels=c(&quot;NA&quot;, &quot;intergenic&quot;, &quot;exonic&quot;, &quot;intronic&quot;, &quot;proximal &gt;1000&amp;&lt;=2500bp&quot;, &quot;proximal &lt;=1000bp&quot;, &quot;TSS&quot;))), 
                           by=&quot;Gene&quot;)

boxplot2 &lt;- ggplot(mixed_data2, aes(x=closest_peak_class, y=logFC, colour=Source)) +
  geom_boxplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of logFC by type of closest GR-peak&quot;, subtitle=&quot;Some outliers cutoff by y-axis&quot;, x=&quot;&quot;) +
  ylim(c(-5,5)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=9),
        plot.margin = unit(c(0.2,0.2,0.2,0.2), &quot;cm&quot;))

violin2 &lt;- ggplot(mixed_data2, aes(x=closest_peak_class, y=logFC, colour=Source)) +
  geom_violin() +
  theme_ipsum_rc() +
  labs(title=&quot;&quot;, subtitle=&quot;&quot;, x=&quot;&quot;) +
  ylim(c(-5,5)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=9),
        plot.margin = unit(c(0.2,0.2,0.2,0.2), &quot;cm&quot;))

boxplot_abs2 &lt;- ggplot(mixed_data2, aes(x=closest_peak_class, y=abs_logFC, colour=Source)) +
  geom_boxplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of ABSOLUTE logFCs by type of closest GR-peak&quot;, subtitle=&quot;Some outliers cutoff by y-axis&quot;, x=&quot;&quot;) +
  ylim(c(0,5)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=9),
        plot.margin = unit(c(0.2,0.2,0.2,0.2), &quot;cm&quot;))

violin_abs2 &lt;- ggplot(mixed_data2, aes(x=closest_peak_class, y=abs_logFC, colour=Source)) +
  geom_violin() +
  theme_ipsum_rc() +
  labs(title=&quot;&quot;, subtitle=&quot;&quot;, x=&quot;&quot;) +
  ylim(c(0,5)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=9),
        plot.margin = unit(c(0.2,0.2,0.2,0.2), &quot;cm&quot;))

group_image2 &lt;- ggarrange(
  ggarrange(boxplot2, violin2, ncol=2),
  ggarrange(boxplot_abs2, violin_abs2, ncol=2),
  ncol=1)</code></pre>
<pre><code>Warning: Removed 27 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 27 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 27 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 27 rows containing non-finite values (stat_ydensity).</code></pre>
<pre class="r"><code>group_image2</code></pre>
<p><img src="figure/Q2b-TF-Binding-to-DE-genes-2h.Rmd/boxplots%20based%20on%20class-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../docs/assets/Q2_6_boxplots_class_2h.png&quot;, group_image2, width=14, height=10, unit=&quot;in&quot;)</code></pre>
</div>
<div id="save-excel" class="section level1">
<h1>Save excel</h1>
<pre class="r"><code>openxlsx::write.xlsx(file=&quot;../docs/downloads/Q2_GR_bound_DE_genes.xlsx&quot;, res_int)

# sessionInfo()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.18.4        AnnotationFilter_1.18.0 GenomicFeatures_1.46.5 
 [4] AnnotationDbi_1.56.2    Biobase_2.54.0          ggpubr_0.4.0           
 [7] hrbrthemes_0.8.6        ggplot2_3.3.6           dplyr_1.0.9            
[10] AnnotationHub_3.2.2     BiocFileCache_2.2.1     dbplyr_2.2.1           
[13] rtracklayer_1.54.0      epiwraps_0.99.42        EnrichedHeatmap_1.24.0 
[16] ComplexHeatmap_2.10.0   GenomicRanges_1.46.1    GenomeInfoDb_1.30.1    
[19] IRanges_2.28.0          S4Vectors_0.32.4        BiocGenerics_0.40.0    

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    tidyselect_1.1.2             
  [3] RSQLite_2.2.14                htmlwidgets_1.5.4            
  [5] BiocParallel_1.28.3           munsell_0.5.0                
  [7] ragg_1.2.2                    codetools_0.2-18             
  [9] interp_1.1-2                  withr_2.5.0                  
 [11] colorspace_2.0-3              filelock_1.0.2               
 [13] highr_0.9                     knitr_1.39                   
 [15] rstudioapi_0.13               ggsignif_0.6.3               
 [17] Rttf2pt1_1.3.10               labeling_0.4.2               
 [19] MatrixGenerics_1.6.0          git2r_0.30.1                 
 [21] GenomeInfoDbData_1.2.7        farver_2.1.1                 
 [23] bit64_4.0.5                   rprojroot_2.0.3              
 [25] vctrs_0.4.1                   generics_0.1.3               
 [27] xfun_0.31                     biovizBase_1.42.0            
 [29] R6_2.5.1                      doParallel_1.0.17            
 [31] clue_0.3-61                   isoband_0.2.5                
 [33] locfit_1.5-9.5                bitops_1.0-7                 
 [35] cachem_1.0.6                  DelayedArray_0.20.0          
 [37] assertthat_0.2.1              promises_1.2.0.1             
 [39] BiocIO_1.4.0                  scales_1.2.0                 
 [41] nnet_7.3-17                   gtable_0.3.0                 
 [43] workflowr_1.7.0               rlang_1.0.3                  
 [45] systemfonts_1.0.4             GlobalOptions_0.1.2          
 [47] splines_4.1.0                 rstatix_0.7.0                
 [49] extrafontdb_1.0               lazyeval_0.2.2               
 [51] dichromat_2.0-0.1             broom_1.0.0                  
 [53] checkmate_2.1.0               BiocManager_1.30.18          
 [55] yaml_2.3.5                    abind_1.4-5                  
 [57] backports_1.4.1               httpuv_1.6.5                 
 [59] Hmisc_4.7-0                   extrafont_0.18               
 [61] tools_4.1.0                   ellipsis_0.3.2               
 [63] jquerylib_0.1.4               RColorBrewer_1.1-3           
 [65] Rcpp_1.0.8.3                  plyr_1.8.7                   
 [67] base64enc_0.1-3               progress_1.2.2               
 [69] zlibbioc_1.40.0               purrr_0.3.4                  
 [71] RCurl_1.98-1.7                prettyunits_1.1.1            
 [73] rpart_4.1.16                  deldir_1.0-6                 
 [75] GetoptLong_1.0.5              cowplot_1.1.1                
 [77] GenomicFiles_1.30.0           SummarizedExperiment_1.24.0  
 [79] cluster_2.1.3                 fs_1.5.2                     
 [81] magrittr_2.0.3                data.table_1.14.2            
 [83] openxlsx_4.2.5                circlize_0.4.15              
 [85] ProtGenerics_1.26.0           matrixStats_0.62.0           
 [87] hms_1.1.1                     mime_0.12                    
 [89] evaluate_0.15                 xtable_1.8-4                 
 [91] XML_3.99-0.10                 jpeg_0.1-9                   
 [93] gridExtra_2.3                 shape_1.4.6                  
 [95] compiler_4.1.0                biomaRt_2.50.3               
 [97] tibble_3.1.7                  crayon_1.5.1                 
 [99] htmltools_0.5.2               later_1.3.0                  
[101] Formula_1.2-4                 tidyr_1.2.0                  
[103] DBI_1.1.3                     MASS_7.3-57                  
[105] rappdirs_0.3.3                Matrix_1.4-1                 
[107] car_3.1-0                     cli_3.3.0                    
[109] parallel_4.1.0                Gviz_1.38.4                  
[111] pkgconfig_2.0.3               GenomicAlignments_1.30.0     
[113] foreign_0.8-82                xml2_1.3.3                   
[115] foreach_1.5.2                 bslib_0.3.1                  
[117] XVector_0.34.0                stringr_1.4.0                
[119] VariantAnnotation_1.40.0      digest_0.6.29                
[121] Biostrings_2.62.0             rmarkdown_2.14               
[123] htmlTable_2.4.0               edgeR_3.36.0                 
[125] gdtools_0.2.4                 restfulr_0.0.15              
[127] curl_4.3.2                    shiny_1.7.1                  
[129] Rsamtools_2.10.0              rjson_0.2.21                 
[131] lifecycle_1.0.1               jsonlite_1.8.0               
[133] carData_3.0-5                 viridisLite_0.4.0            
[135] limma_3.50.3                  BSgenome_1.62.0              
[137] fansi_1.0.3                   pillar_1.7.0                 
[139] lattice_0.20-45               KEGGREST_1.34.0              
[141] fastmap_1.1.0                 httr_1.4.3                   
[143] survival_3.3-1                interactiveDisplayBase_1.32.0
[145] glue_1.6.2                    zip_2.2.0                    
[147] UpSetR_1.4.0                  png_0.1-7                    
[149] iterators_1.0.14              BiocVersion_3.14.0           
[151] bit_4.0.4                     stringi_1.7.6                
[153] sass_0.4.1                    blob_1.2.3                   
[155] textshaping_0.3.6             latticeExtra_0.6-30          
[157] memoise_2.0.1                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
