<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-09-21" />

<title>Q3: PROTAC reversal</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Q3: PROTAC reversal</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-09-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree42559356a81583e0cc4d967a24e31f222f9d05detargetblank4255935a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/42559356a81583e0cc4d967a24e31f222f9d05de" target="_blank">4255935</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree42559356a81583e0cc4d967a24e31f222f9d05detargetblank4255935a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/42559356a81583e0cc4d967a24e31f222f9d05de" target="_blank">4255935</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/docs/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/Formatted_Candidate_ETG_20200305_TAD10Kb.tsv
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Ignored:    data/reddy_rna_DEX2h_r1.tsv
    Ignored:    data/reddy_rna_DEX2h_r2.tsv
    Ignored:    data/reddy_rna_DEX2h_r3.tsv
    Ignored:    data/reddy_rna_DEX2h_r4.tsv
    Ignored:    data/single_replicate_peaks/.DS_Store
    Ignored:    data/single_replicate_peaks/control/
    Ignored:    data/single_replicate_peaks/dex12h/
    Ignored:    data/single_replicate_peaks/dex2h/
    Ignored:    data/single_replicate_peaks/peaks/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/Files/

Untracked files:
    Untracked:  analysis/Q2b-TF-Binding-to-DE-genes-2h.Rmd
    Untracked:  data/00_paper_figures/

Unstaged changes:
    Modified:   analysis/01_Results.Rmd
    Modified:   analysis/Q11-MR_paper.Rmd
    Deleted:    analysis/Q2b-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/admin.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Q3-PROTAC-reversal.Rmd</code>) and HTML (<code>docs/Q3-PROTAC-reversal.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/748f409e28a1149f7546c4b1cdffb66aea132fcc/analysis/Q3-PROTAC-reversal.Rmd" target="_blank">748f409</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-19
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/analysis/Q3-PROTAC-reversal.Rmd" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/Q3-PROTAC-reversal.html" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
<td>
updates september
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/4a4e1e6dc540ada645ce8cc331f7608414ed569a/analysis/Q3-PROTAC-reversal.Rmd" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/4a4e1e6dc540ada645ce8cc331f7608414ed569a/docs/Q3-PROTAC-reversal.html" target="_blank">4a4e1e6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-07-09
</td>
<td>
final changes for submission in main files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/analysis/Q3-PROTAC-reversal.Rmd" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/Q3-PROTAC-reversal.html" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/analysis/Q3-PROTAC-reversal.Rmd" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="question-3" class="section level1">
<h1>Question 3:</h1>
<p>Our RNA: are there any genes that treated by the PROTAC before/after DEX that could be reversed but which could not by the inhibitors or vice versa?</p>
<p>We have performed experiments where we first exposed cells to either (1) DEX, (2) inhibitors, or (3) KH PROTAC for a substantial time and then added inhibitors or the KH PROTAC in case (1) or DEX in cases (2) and (3). We would like to see whether the genes which were activated by DEX but then returned to a normal expression by the KH PROTAC, where a similar treatment by the inhibitors could not lead to such an effect and vice versa.</p>
<p>In our RNAseq, are there any genes that Protac after dex or before dex could reverse or partially reverse that cort113 or mifepristone could not? do these gene show GR binding according to Reddy’s chip data?</p>
<pre class="r"><code>res_int_all &lt;- readRDS(file=&quot;../data/results/DE_our_data.rds&quot;)
names(res_int_all)</code></pre>
<pre><code> [1] &quot;16h Cort113 &gt; 2h DEX+Cort113&quot; &quot;16h DMSO &gt; 2h DEX+DMSO&quot;      
 [3] &quot;16h KH-103 &gt; 2h DEX+KH-103&quot;   &quot;16h MIF  &gt; 2h DEX+MIF&quot;       
 [5] &quot;18h Cort113&quot;                  &quot;18h KH-103&quot;                  
 [7] &quot;18h MIF&quot;                      &quot;2h DEX &gt; 16h DEX+Cort113&quot;    
 [9] &quot;2h DEX &gt; 16h DEX+DMSO&quot;        &quot;2h DEX &gt; 16h DEX+KH-103&quot;     
[11] &quot;2h DEX &gt; 16h DEX+MIF&quot;        </code></pre>
<pre class="r"><code># The threshold we choose for which values count as DE or not
fdr_thresh &lt;- 0.05

# Load the control data for the 2+16 experiment
control216 &lt;- res_int_all[[&quot;2h DEX &gt; 16h DEX+DMSO&quot;]]
control162 &lt;- res_int_all[[&quot;16h DMSO &gt; 2h DEX+DMSO&quot;]]</code></pre>
<div id="h-16h" class="section level2">
<h2>2h &gt; 16h</h2>
<p><em>Def. 1:</em> We consider “reversed by inhibitor” to mean: “Is DE in ‘2h DEX &gt; 16h DEX+DMSO’ but not in ‘2h DEX &gt; 16h DEX+Cort113’”. We look at both inhibitors and the PROTAC</p>
<pre class="r"><code># Load inhibitors/Protac data
cort216 &lt;- res_int_all[[&quot;2h DEX &gt; 16h DEX+Cort113&quot;]]
mif216 &lt;- res_int_all[[&quot;2h DEX &gt; 16h DEX+MIF&quot;]]
protac216 &lt;- res_int_all[[&quot;2h DEX &gt; 16h DEX+KH-103&quot;]]

# merge the respective FDRs onto DF
merged216 &lt;- control216 %&gt;% 
  tibble::rownames_to_column() %&gt;% 
  left_join(cort216 %&gt;% 
              tibble::rownames_to_column() %&gt;%
              dplyr::select(rowname, logFC, FDR), suffix=c(&quot;&quot;,&quot;_cort&quot;), by=&quot;rowname&quot;) %&gt;%
  left_join(mif216 %&gt;% 
              tibble::rownames_to_column() %&gt;%
              dplyr::select(rowname, logFC, FDR), suffix=c(&quot;&quot;,&quot;_MIF&quot;), by=&quot;rowname&quot;) %&gt;%
  left_join(protac216 %&gt;% 
              tibble::rownames_to_column() %&gt;%
              dplyr::select(rowname, logFC, FDR), suffix=c(&quot;&quot;,&quot;_KH&quot;), by=&quot;rowname&quot;)

# Using a FDR of 0.05, how many clear reversals do we have?
merged216$cort_reversed &lt;- ifelse(merged216$FDR &lt; 0.05 &amp; merged216$FDR_cort &gt;= 0.05, TRUE, FALSE)
merged216$MIF_reversed &lt;- ifelse(merged216$FDR &lt; 0.05 &amp; merged216$FDR_MIF &gt;= 0.05, TRUE, FALSE)
merged216$KH_reversed &lt;- ifelse(merged216$FDR &lt; 0.05 &amp; merged216$FDR_KH &gt;= 0.05, TRUE, FALSE)

# transform from wide to long
merged216_long &lt;- reshape2::melt(merged216, 
                id.vars=&quot;rowname&quot;,
                  measure.vars=c(&quot;logFC&quot;, &quot;logFC_cort&quot;, &quot;logFC_MIF&quot;, &quot;logFC_KH&quot;)) %&gt;%
  mutate(Group=factor(variable))
levels(merged216_long$Group) &lt;- c(&quot;Control&quot;, &quot;Cort&quot;, &quot;MIF&quot;, &quot;KH&quot;)

# sort rownames in descending order of control logFCs
rnames &lt;- merged216_long[merged216_long$Group == &quot;Control&quot;,] %&gt;% arrange(desc(value)) %&gt;% pull(rowname)
merged216_long$rowname &lt;- factor(merged216_long$rowname, levels=rnames)

# unique protac reversals
merged216 &lt;- merged216 %&gt;% arrange(logFC)
rnames_kh_reversal &lt;- merged216[merged216$KH_reversed == TRUE &amp; 
                                  merged216$cort_reversed == F &amp; 
                                  merged216$MIF_reversed == F,&quot;rowname&quot;]

merged216[merged216$rowname %in% rnames_kh_reversal,&quot;KH_reversed&quot;] &lt;- TRUE</code></pre>
<pre class="r"><code>plt1 &lt;- ggplot(merged216_long[merged216_long$rowname %in% rnames_kh_reversal,]) +
  geom_point(aes(x=rowname, y=value, color=Group)) +
  theme_ipsum_rc() +
  labs(title=&quot;Genes reversed by Protac but not by inhibitors&quot;,
       subtitle=&quot;2h DEX &gt; 16h DEX + Inhibitor/Protac&quot;,
       x=&quot;&quot;, y=&quot;&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
    scale_color_manual(values=c(Control=&quot;Black&quot;, Cort=&quot;#7ef2b4&quot;, MIF=&quot;#dda6ff&quot;, KH=&quot;#ff000d&quot;))
plt1</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-1-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-1-1.png" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q3_1_reversal216.png&quot;, plt1, width=12, height=7)</code></pre>
<div id="sign-flips-reversals" class="section level3">
<h3>Sign flips / reversals</h3>
<p>We search for cases where an inhibitor or the protac has flipped the sign of a logFC. We filter for genes where the gene was DE in both the comparison with the control as well as inhibitor/protac. We apply the additional filter than the sign of the logFC has changed with reference to the control experiment.</p>
<pre class="r"><code># Check for flips in sign
merged216$cort_flipped &lt;- ifelse(merged216$FDR &lt; 0.05 &amp; merged216$FDR_cort &lt; 0.05 &amp; 
                                   ((merged216$logFC * merged216$logFC_cort) &lt; 0), TRUE, FALSE)
merged216$MIF_flipped &lt;- ifelse(merged216$FDR &lt; 0.05 &amp; merged216$FDR_MIF &lt; 0.05 &amp; 
                                   ((merged216$logFC * merged216$logFC_MIF) &lt; 0), TRUE, FALSE)
merged216$KH_flipped &lt;- ifelse(merged216$FDR &lt; 0.05 &amp; merged216$FDR_KH &lt; 0.05 &amp; 
                                   ((merged216$logFC * merged216$logFC_KH) &lt; 0), TRUE, FALSE)

plot_logFCs_216 &lt;- function(rownames_subset, title){
  # cort
  plt &lt;- ggplot(merged216_long[merged216_long$rowname %in% rownames_subset,]) +
    geom_point(aes(x=rowname, y=value, color=Group)) +
    theme_ipsum_rc() +
    labs(title=title,
         subtitle=&quot;2h DEX &gt; 16h DEX + Inhibitor/Protac&quot;,
         x=&quot;&quot;, y=&quot;&quot;) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
      scale_color_manual(values=c(Control=&quot;Black&quot;, Cort=&quot;#7ef2b4&quot;, MIF=&quot;#dda6ff&quot;, KH=&quot;#ff000d&quot;))
  plt
}

rnames_flipped_cort &lt;- merged216 %&gt;% dplyr::filter(cort_flipped == TRUE) %&gt;% pull(rowname)
rnames_flipped_mif &lt;- merged216 %&gt;% dplyr::filter(MIF_flipped == TRUE) %&gt;% pull(rowname)
rnames_flipped_kh &lt;- merged216 %&gt;% dplyr::filter(KH_flipped == TRUE) %&gt;% pull(rowname)

plt_flipped_cort &lt;- plot_logFCs_216(rnames_flipped_cort, &quot;Genes flipped by cort&quot;)
plt_flipped_mif &lt;- plot_logFCs_216(rnames_flipped_mif, &quot;Genes flipped by MIF&quot;)
plt_flipped_kh &lt;- plot_logFCs_216(rnames_flipped_kh, &quot;Genes flipped by KH&quot;)

# Save all plots
ggsave(file=&quot;../docs/assets/Q3_1_flipped_cort_216.png&quot;, plt_flipped_cort, width=12, height=7)
ggsave(file=&quot;../docs/assets/Q3_1_flipped_mif_216.png&quot;, plt_flipped_mif, width=12, height=7)
ggsave(file=&quot;../docs/assets/Q3_1_flipped_kh_216.png&quot;, plt_flipped_kh, width=12, height=7)

# Show all plots
cat(&quot;\n\n### Flipped effects {.tabset} \n\n&quot;)</code></pre>
</div>
<div id="flipped-effects" class="section level3 tabset">
<h3 class="tabset">Flipped effects</h3>
<pre class="r"><code>cat(&quot;\n\n#### Cort \n\n&quot;)</code></pre>
<div id="cort" class="section level4">
<h4>Cort</h4>
<pre class="r"><code>plt_flipped_cort</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n#### MIF  \n\n&quot;)</code></pre>
</div>
<div id="mif" class="section level4">
<h4>MIF</h4>
<pre class="r"><code>plt_flipped_mif</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-2.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n#### KH \n\n&quot;)</code></pre>
</div>
<div id="kh" class="section level4">
<h4>KH</h4>
<pre class="r"><code>plt_flipped_kh</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-3">
Past versions of unnamed-chunk-2-3.png
</button>
</p>
<div id="fig-unnamed-chunk-2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-3.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># plot genes flipped by the respective markers only
rnames_flipped_cort_only &lt;- rnames_flipped_cort[!(rnames_flipped_cort %in% c(rnames_flipped_mif, rnames_flipped_kh))]
rnames_flipped_mif_only &lt;- rnames_flipped_mif[!(rnames_flipped_mif %in% c(plt_flipped_kh, rnames_flipped_kh))]
rnames_flipped_kh_only &lt;- rnames_flipped_mif[!(rnames_flipped_mif %in% c(rnames_flipped_mif, plt_flipped_cort))]

plt_flipped_cort_only &lt;- plot_logFCs_216(rnames_flipped_cort_only, &quot;Genes flipped by cort only&quot;)
plt_flipped_mif_only &lt;- plot_logFCs_216(rnames_flipped_mif_only, &quot;Genes flipped by MIF only&quot;)
plt_flipped_kh_only &lt;- plot_logFCs_216(rnames_flipped_kh_only, &quot;Genes flipped by KH only&quot;)

# Save all plots
ggsave(file=&quot;../docs/assets/Q3_1_flipped_cort_only_216.png&quot;, plt_flipped_cort_only, width=12, height=7)
ggsave(file=&quot;../docs/assets/Q3_1_flipped_mif_only_216.png&quot;, plt_flipped_mif_only, width=12, height=7)
ggsave(file=&quot;../docs/assets/Q3_1_flipped_kh_only_216.png&quot;, plt_flipped_kh_only, width=12, height=7)

# Show all plots
cat(&quot;\n\n### Flipped effects with unique genes per treatment {.tabset} \n\n&quot;)</code></pre>
</div>
</div>
<div id="flipped-effects-with-unique-genes-per-treatment" class="section level3 tabset">
<h3 class="tabset">Flipped effects with unique genes per treatment</h3>
<pre class="r"><code>cat(&quot;\n\n#### Cort \n\n&quot;)</code></pre>
<div id="cort-1" class="section level4">
<h4>Cort</h4>
<pre class="r"><code>plt_flipped_cort_only</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-4">
Past versions of unnamed-chunk-2-4.png
</button>
</p>
<div id="fig-unnamed-chunk-2-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-4.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n#### MIF  \n\n&quot;)</code></pre>
</div>
<div id="mif-1" class="section level4">
<h4>MIF</h4>
<pre class="r"><code>plt_flipped_mif_only</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-5.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-5">
Past versions of unnamed-chunk-2-5.png
</button>
</p>
<div id="fig-unnamed-chunk-2-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-5.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n#### KH \n\n&quot;)</code></pre>
</div>
<div id="kh-1" class="section level4">
<h4>KH</h4>
<pre class="r"><code>plt_flipped_kh_only</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-6.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-6">
Past versions of unnamed-chunk-2-6.png
</button>
</p>
<div id="fig-unnamed-chunk-2-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-2-6.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="plots-from-mahshids-summary" class="section level2">
<h2>Plots from Mahshids summary</h2>
</div>
<div id="only-those-genes-which-are-significant-in-dex-18h-ns-in-2hkh103-but-are-significant-in-the-same-direction---as-dex-18h-in-2h-dexmif-and-2h-dexcort113" class="section level2">
<h2>1. Only those genes which are significant in Dex 18h, ns in 2h&gt;KH103 but are significant in the same direction (+/-) as Dex 18h in 2h dex&gt;mif and 2h dex&gt;cort113</h2>
<pre class="r"><code>merged216$category1 &lt;- (merged216$FDR &lt; 0.05) &amp; 
  (merged216$FDR_KH &gt; 0.05) &amp; 
  (merged216$FDR_MIF &lt; 0.05) &amp; (merged216$FDR_cort &lt; 0.05) &amp; 
  (merged216$logFC_MIF * merged216$logFC_cort &gt; 0) &amp; # checks that mif and cort have same sign
  (merged216$logFC * merged216$logFC_cort &gt; 0)  # checks that mif and 2h dex have the same sign

rnames_flipped_cat1 &lt;- merged216 %&gt;% dplyr::filter(category1 == TRUE) %&gt;% pull(rowname)

# fig 6d.2a
plot_logFCs_216(rnames_flipped_cat1, &quot;Category 1&quot;)</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-3-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-3-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-3-1.png" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(filename=&quot;../data/00_paper_figures/6d2a.png&quot;, width = 14, height=5)</code></pre>
</div>
<div id="only-those-genes-which-are-significant-in-dex-18h-ns-in-2h-dex-kh103-but-are-significant-in-the-opposite-direction-as-dex-18h-in-2h-dexmif-and-2h-dex-cort113" class="section level2">
<h2>2. Only those genes which are significant in Dex 18h, ns in 2h dex&gt; KH103 but are significant in the opposite direction as Dex 18h in 2h dex&gt;mif and 2h dex &gt; cort113</h2>
<pre class="r"><code>merged216$category2 &lt;- (merged216$FDR &lt; 0.05) &amp; 
  (merged216$FDR_KH &gt; 0.05) &amp; 
  (merged216$FDR_MIF &lt; 0.05) &amp; (merged216$FDR_cort &lt; 0.05) &amp; 
  (merged216$logFC_MIF * merged216$logFC_cort &gt; 0) &amp; # checks that mif and cort have same sign
  (merged216$logFC * merged216$logFC_cort &lt; 0)  # checks that mif and 2h dex have different sign

rnames_flipped_cat2 &lt;- merged216 %&gt;% dplyr::filter(category2 == TRUE) %&gt;% pull(rowname)

# fig 6d.2b
plot_logFCs_216(rnames_flipped_cat2, &quot;Category 2&quot;)</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../data/00_paper_figures/6d2b.png&quot;, width = 7, height=5)</code></pre>
</div>
<div id="only-those-genes-which-are-significant-in-dex-18h-and-are-ns-or-significant-in-the-same-direction-as-dex-18h-for-2h-dexmif-and-2h-dexcort113-but-are-significant-in-the-opposite-direction-as-dex-18h-in-2h-dexkh-103" class="section level2">
<h2>3. Only those genes which are significant in Dex 18h, and are ns or significant in the same direction as Dex 18h for 2h dex&gt;mif and 2h dex&gt;cort113 but are significant in the opposite direction as Dex 18h in 2h dex&gt;KH-103</h2>
<pre class="r"><code>merged216$category3 &lt;- (merged216$FDR &lt; 0.05) &amp; 
  (merged216$logFC * merged216$logFC_MIF &gt; 0) &amp; # 2h DEX and MIF point in same direction
  (merged216$logFC * merged216$logFC_cort &gt; 0) &amp; # 2h DEX and cort point in same direction
  (merged216$logFC * merged216$logFC_KH &lt; 0) &amp; # 2h DEX and KH have different logFC signs
  (merged216$FDR_KH &gt; 0) # KH is significant

rnames_flipped_cat3 &lt;- merged216 %&gt;% dplyr::filter(category3 == TRUE) %&gt;% pull(rowname)

# fig 6d.2c
plot_logFCs_216(rnames_flipped_cat3, &quot;Category 3&quot;)</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-5-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-5-1.png" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(filename=&quot;../data/00_paper_figures/6d2c.png&quot;, width = 14, height=5)</code></pre>
</div>
<div id="only-those-genes-which-are-significant-in-dex-18h-and-are-significant-in-the-opposite-direction-as-dex-18h-in-all-three-conditions-of-2h-dexkh-103-2h-dexmif-2h-dexcort113" class="section level2">
<h2>4. Only those genes which are significant in Dex 18h, and are significant in the opposite direction as Dex 18h in all three conditions of 2h dex&gt;KH-103, 2h dex&gt;MIF, 2h dex&gt;Cort113</h2>
<pre class="r"><code>merged216$category4 &lt;- (merged216$FDR &lt; 0.05) &amp; 
  (merged216$logFC * merged216$logFC_KH &lt; 0) &amp; (merged216$FDR_KH &lt; 0.05) &amp; # KH is significant in other direction as 2h DEX
  (merged216$logFC * merged216$logFC_MIF &lt; 0) &amp; (merged216$FDR_MIF &lt; 0.05) &amp; # MIF is significant in other direction as 2h DEX
  (merged216$logFC * merged216$logFC_cort &lt; 0) &amp; (merged216$FDR_cort &lt; 0.05) # cort is significant in other direction as 2h DEX

rnames_flipped_cat4 &lt;- merged216 %&gt;% dplyr::filter(category4 == TRUE) %&gt;% pull(rowname)

# fig 6d.2d
plot_logFCs_216(rnames_flipped_cat4, &quot;Category 4&quot;)</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-6-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../data/00_paper_figures/6d2d.png&quot;, width = 14, height=5)</code></pre>
</div>
</div>
<div id="h-2h" class="section level1">
<h1>16h &gt; 2h</h1>
<p><em>Def. 2:</em> “Is DE from 16h DEX &gt; 2h DEX+DMSO but is not DE in 16h DEX &gt; 2h DEX+Cort113 experiment”</p>
<pre class="r"><code># Load inhibitors/Protac data
cort162 &lt;- res_int_all[[&quot;16h Cort113 &gt; 2h DEX+Cort113&quot;]]
mif162 &lt;- res_int_all[[ &quot;16h MIF  &gt; 2h DEX+MIF&quot;]]
protac162 &lt;- res_int_all[[&quot;16h KH-103 &gt; 2h DEX+KH-103&quot;]]

# merge the respective FDRs onto DF
merged162 &lt;- control162 %&gt;% 
  tibble::rownames_to_column() %&gt;% 
  left_join(cort162 %&gt;% 
              tibble::rownames_to_column() %&gt;%
              dplyr::select(rowname, logFC, FDR), suffix=c(&quot;&quot;,&quot;_cort&quot;), by=&quot;rowname&quot;) %&gt;%
  left_join(mif162 %&gt;% 
              tibble::rownames_to_column() %&gt;%
              dplyr::select(rowname, logFC, FDR), suffix=c(&quot;&quot;,&quot;_MIF&quot;), by=&quot;rowname&quot;) %&gt;%
  left_join(protac162 %&gt;% 
              tibble::rownames_to_column() %&gt;%
              dplyr::select(rowname, logFC, FDR), suffix=c(&quot;&quot;,&quot;_KH&quot;), by=&quot;rowname&quot;)

# Using a FDR of 0.05, how many clear reversals do we have?
merged162$cort_reversed &lt;- ifelse(merged162$FDR &lt; 0.05 &amp; merged162$FDR_cort &gt;= 0.05, TRUE, FALSE)
merged162$MIF_reversed &lt;- ifelse(merged162$FDR &lt; 0.05 &amp; merged162$FDR_MIF &gt;= 0.05, TRUE, FALSE)
merged162$KH_reversed &lt;- ifelse(merged162$FDR &lt; 0.05 &amp; merged162$FDR_KH &gt;= 0.05, TRUE, FALSE)

# transform from wide to long
merged162_long &lt;- reshape2::melt(merged162,
                id.vars=&quot;rowname&quot;,
                  measure.vars=c(&quot;logFC&quot;, &quot;logFC_cort&quot;, &quot;logFC_MIF&quot;, &quot;logFC_KH&quot;)) %&gt;%
  mutate(Group=factor(variable))
levels(merged162_long$Group) &lt;- c(&quot;Control&quot;, &quot;Cort&quot;, &quot;MIF&quot;, &quot;KH&quot;)

# sort rownames in descending order of control logFCs
rnames &lt;- merged162_long[merged162_long$Group == &quot;Control&quot;,] %&gt;% arrange(desc(value)) %&gt;% pull(rowname)
merged162_long$rowname &lt;- factor(merged162_long$rowname, levels=rnames)

# unique protac reversals
merged162 &lt;- merged162 %&gt;% arrange(logFC)
rnames_kh_reversal &lt;- merged216[merged162$KH_reversed == TRUE &amp; 
                                  merged162$cort_reversed == F &amp; 
                                  merged162$MIF_reversed == F,&quot;rowname&quot;]

merged162[merged162$rowname %in% rnames_kh_reversal,&quot;KH_reversed&quot;] &lt;- TRUE</code></pre>
<p>Because we have a lot of genes to show in one plot for the 16h &gt; 2h experiment, we split them into two groups. One plot shows the genes for which the control logFC is &gt;0 and the other &lt;0.</p>
<pre class="r"><code># split the long formatted dataset into two groups
data_plt &lt;- merged162_long[merged162_long$rowname %in% rnames_kh_reversal,]
data &lt;- data_plt[data_plt$Group == &quot;Control&quot;,]
rnames_plt1 &lt;- data[data$value &gt; 0,&quot;rowname&quot;]
rnames_plt2 &lt;- data[data$value &lt;= 0,&quot;rowname&quot;]
data_plt1 &lt;- data_plt[data_plt$rowname %in% rnames_plt1,]
data_plt2 &lt;- data_plt[data_plt$rowname %in% rnames_plt2,]

plt2_1 &lt;- ggplot(data_plt1) +
  geom_point(aes(x=rowname, y=value, color=Group)) +
  theme_ipsum_rc() +
  labs(title=&quot;Genes reversed by Protac but not by inhibitors&quot;,
       subtitle=&quot;16h Inhibitor/Protac &gt; 2h Inhibitor/Protac + DEX&quot;,
       x=&quot;&quot;, y=&quot;&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
    scale_color_manual(values=c(Control=&quot;Black&quot;, Cort=&quot;#7ef2b4&quot;, MIF=&quot;#dda6ff&quot;, KH=&quot;#ff000d&quot;))

plt2_2 &lt;- ggplot(data_plt2) +
  geom_point(aes(x=rowname, y=value, color=Group)) +
  theme_ipsum_rc() +
  labs(title=&quot;Genes reversed by Protac but not by inhibitors&quot;,
       subtitle=&quot;16h Inhibitor/Protac &gt; 2h Inhibitor/Protac + DEX&quot;,
       x=&quot;&quot;, y=&quot;&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
    scale_color_manual(values=c(Control=&quot;Black&quot;, Cort=&quot;#7ef2b4&quot;, MIF=&quot;#dda6ff&quot;, KH=&quot;#ff000d&quot;))

# show the plots
plt2_1</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/plots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-1">
Past versions of plots-1.png
</button>
</p>
<div id="fig-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/plots-1.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/figure/Q3-PROTAC-reversal.Rmd/plots-1.png" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plt2_2</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/plots-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plots-2">
Past versions of plots-2.png
</button>
</p>
<div id="fig-plots-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/fd8956d244119afeedefa8cf31c9eddca56d6014/docs/figure/Q3-PROTAC-reversal.Rmd/plots-2.png" target="_blank">fd8956d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/figure/Q3-PROTAC-reversal.Rmd/plots-2.png" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q3_2_reversal162_p1.png&quot;, plt2_1, width=24, height=8)
ggsave(file=&quot;../docs/assets/Q3_3_reversal162_p2.png&quot;, plt2_2, width=24, height=8)</code></pre>
<div id="sign-flips-reversals-1" class="section level3">
<h3>Sign flips / reversals</h3>
<p>We search for cases where an inhibitor or the protac has flipped the sign of a logFC. We filter for genes where the gene was DE in both the comparison with the control as well as inhibitor/protac. We apply the additional filter than the sign of the logFC has changed with reference to the control experiment.</p>
<pre class="r"><code># Check for flips in sign
merged162$cort_flipped &lt;- ifelse(merged162$FDR &lt; 0.05 &amp; merged162$FDR_cort &lt; 0.05 &amp; 
                                   ((merged162$logFC * merged162$logFC_cort) &lt; 0), TRUE, FALSE)
merged162$MIF_flipped &lt;- ifelse(merged162$FDR &lt; 0.05 &amp; merged162$FDR_MIF &lt; 0.05 &amp; 
                                   ((merged162$logFC * merged162$logFC_MIF) &lt; 0), TRUE, FALSE)
merged162$KH_flipped &lt;- ifelse(merged162$FDR &lt; 0.05 &amp; merged162$FDR_KH &lt; 0.05 &amp; 
                                   ((merged162$logFC * merged162$logFC_KH) &lt; 0), TRUE, FALSE)

plot_logFCs_162 &lt;- function(rownames_subset, title){
  # cort
  plt &lt;- ggplot(merged162_long[merged162_long$rowname %in% rownames_subset,]) +
    geom_point(aes(x=rowname, y=value, color=Group)) +
    theme_ipsum_rc() +
    labs(title=title,
         subtitle=&quot;2h DEX &gt; 16h DEX + Inhibitor/Protac&quot;,
         x=&quot;&quot;, y=&quot;&quot;) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
      scale_color_manual(values=c(Control=&quot;Black&quot;, Cort=&quot;#7ef2b4&quot;, MIF=&quot;#dda6ff&quot;, KH=&quot;#ff000d&quot;))
  plt
}

rnames_flipped_cort &lt;- merged162 %&gt;% dplyr::filter(cort_flipped == TRUE) %&gt;% pull(rowname)
rnames_flipped_mif &lt;- merged162 %&gt;% dplyr::filter(MIF_flipped == TRUE) %&gt;% pull(rowname)
rnames_flipped_kh &lt;- merged162 %&gt;% dplyr::filter(KH_flipped == TRUE) %&gt;% pull(rowname)

print(kableExtra::kable(data.frame(&quot;Inhibitor&quot;=c(&quot;Cort only&quot;, &quot;MIF only&quot;, &quot;KH only&quot;), 
                                   &quot;length&quot;=c(length(rnames_flipped_cort), 
                                              length(rnames_flipped_mif), 
                                              length(rnames_flipped_kh)))))</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
Inhibitor
</th>
<th style="text-align:right;">
length
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Cort only
</td>
<td style="text-align:right;">
34
</td>
</tr>
<tr>
<td style="text-align:left;">
MIF only
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
KH only
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>plt_flipped_cort &lt;- plot_logFCs_162(rnames_flipped_cort, &quot;Genes flipped by cort&quot;)
plt_flipped_mif &lt;- plot_logFCs_162(rnames_flipped_mif, &quot;Genes flipped by MIF&quot;)
plt_flipped_kh &lt;- plot_logFCs_162(rnames_flipped_kh, &quot;Genes flipped by KH&quot;)

# Save all plots
ggsave(file=&quot;../docs/assets/Q3_1_flipped_cort_216.png&quot;, plt_flipped_cort, width=12, height=7)
ggsave(file=&quot;../docs/assets/Q3_1_flipped_mif_216.png&quot;, plt_flipped_mif, width=12, height=7)
ggsave(file=&quot;../docs/assets/Q3_1_flipped_kh_216.png&quot;, plt_flipped_kh, width=12, height=7)

# Show all plots
cat(&quot;\n\n### Flipped effects {.tabset} \n\n&quot;)</code></pre>
</div>
<div id="flipped-effects-1" class="section level3 tabset">
<h3 class="tabset">Flipped effects</h3>
<pre class="r"><code>cat(&quot;\n\n#### Cort \n\n&quot;)</code></pre>
<div id="cort-2" class="section level4">
<h4>Cort</h4>
<pre class="r"><code>plt_flipped_cort</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n#### MIF  \n\n&quot;)</code></pre>
</div>
<div id="mif-2" class="section level4">
<h4>MIF</h4>
<pre class="r"><code>plt_flipped_mif</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-7-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n#### KH \n\n&quot;)</code></pre>
</div>
<div id="kh-2" class="section level4">
<h4>KH</h4>
<pre class="r"><code>plt_flipped_kh</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-7-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot genes flipped by the respective markers only
rnames_flipped_cort_only &lt;- rnames_flipped_cort[!(rnames_flipped_cort %in% c(rnames_flipped_mif, rnames_flipped_kh))]
rnames_flipped_mif_only &lt;- rnames_flipped_mif[!(rnames_flipped_mif %in% c(plt_flipped_kh, rnames_flipped_kh))]
rnames_flipped_kh_only &lt;- rnames_flipped_mif[!(rnames_flipped_mif %in% c(rnames_flipped_mif, plt_flipped_cort))]

print(kableExtra::kable(data.frame(&quot;Inhibitor&quot;=c(&quot;Cort only&quot;, &quot;MIF only&quot;, &quot;KH only&quot;), 
                                   &quot;length&quot;=c(length(rnames_flipped_cort_only), 
                                              length(rnames_flipped_mif_only), 
                                              length(rnames_flipped_kh_only)))))</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
Inhibitor
</th>
<th style="text-align:right;">
length
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Cort only
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
MIF only
</td>
<td style="text-align:right;">
26
</td>
</tr>
<tr>
<td style="text-align:left;">
KH only
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>plt_flipped_cort_only &lt;- plot_logFCs_162(rnames_flipped_cort_only, &quot;Genes flipped by cort only&quot;)
plt_flipped_mif_only &lt;- plot_logFCs_162(rnames_flipped_mif_only, &quot;Genes flipped by MIF only&quot;)
plt_flipped_kh_only &lt;- plot_logFCs_162(rnames_flipped_kh_only, &quot;Genes flipped by KH only&quot;)

# Save all plots
ggsave(file=&quot;../docs/assets/Q3_1_flipped_cort_only_216.png&quot;, plt_flipped_cort_only, width=12, height=7)
ggsave(file=&quot;../docs/assets/Q3_1_flipped_mif_only_216.png&quot;, plt_flipped_mif_only, width=12, height=7)
ggsave(file=&quot;../docs/assets/Q3_1_flipped_kh_only_216.png&quot;, plt_flipped_kh_only, width=12, height=7)

# Show all plots
cat(&quot;\n\n### Flipped effects with unique genes per treatment {.tabset} \n\n&quot;)</code></pre>
</div>
</div>
<div id="flipped-effects-with-unique-genes-per-treatment-1" class="section level3 tabset">
<h3 class="tabset">Flipped effects with unique genes per treatment</h3>
<pre class="r"><code>cat(&quot;\n\n#### Cort \n\n&quot;)</code></pre>
<div id="cort-3" class="section level4">
<h4>Cort</h4>
<pre class="r"><code>plt_flipped_cort_only</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-7-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n#### MIF  \n\n&quot;)</code></pre>
</div>
<div id="mif-3" class="section level4">
<h4>MIF</h4>
<pre class="r"><code>plt_flipped_mif_only</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-7-5.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n#### KH \n\n&quot;)</code></pre>
</div>
<div id="kh-3" class="section level4">
<h4>KH</h4>
<pre class="r"><code>plt_flipped_kh_only</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-7-6.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="plots-according-to-document" class="section level1">
<h1>Plots according to document</h1>
<div id="only-those-genes-which-are-significant-in-dex-2h-ns-in-kh1032h-dex-but-are-significant-in-the-same-direction---as-dex-2h-in-mif2h-dex-and-cort2h-dex" class="section level2">
<h2>Only those genes which are significant in Dex 2h, ns in KH103&gt;2h dex but are significant in the same direction (+/-) as Dex 2h in mif&gt;2h dex and cort&gt;2h dex</h2>
<pre class="r"><code>merged162$category1 &lt;- (merged162$FDR &lt; 0.05) &amp; 
  (merged162$FDR_KH &gt; 0.05) &amp; 
  (merged162$FDR_MIF &lt; 0.05) &amp; (merged162$FDR_cort &lt; 0.05) &amp; 
  (merged162$logFC_MIF * merged162$logFC_cort &gt; 0) &amp; # checks that mif and cort have same sign
  (merged162$logFC * merged162$logFC_cort &gt; 0)  # checks that mif and 2h dex have the same sign

rnames_flipped_cat1 &lt;- merged162 %&gt;% dplyr::filter(category1 == TRUE) %&gt;% pull(rowname)

# fig 6c.2a
plot_logFCs_162(rnames_flipped_cat1, &quot;Category 1&quot;)</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-8-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../data/00_paper_figures/6c2a.png&quot;, width = 14, height=5)</code></pre>
</div>
<div id="only-those-genes-which-are-significant-in-dex-2h-ns-in-kh1032h-dex-but-are-significant-in-the-opposite-direction-as-dex-2h-in-mif2h-dex-and-cort1132h-dex" class="section level2">
<h2>Only those genes which are significant in Dex 2h, ns in KH103&gt;2h dex but are significant in the opposite direction as Dex 2h in mif&gt;2h dex and cort113&gt;2h dex</h2>
<pre class="r"><code>merged162$category2 &lt;- (merged162$FDR &lt; 0.05) &amp; 
  (merged162$FDR_KH &gt; 0.05) &amp; 
  (merged162$FDR_MIF &lt; 0.05) &amp; (merged162$FDR_cort &lt; 0.05) &amp; 
  (merged162$logFC_MIF * merged162$logFC_cort &gt; 0) &amp; # checks that mif and cort have same sign
  (merged162$logFC * merged162$logFC_cort &lt; 0)  # checks that mif and 2h dex have different sign

rnames_flipped_cat2 &lt;- merged162 %&gt;% dplyr::filter(category2 == TRUE) %&gt;% pull(rowname)

# fig 6c.2b
plot_logFCs_162(rnames_flipped_cat2, &quot;Category 2&quot;)</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../data/00_paper_figures/6c2b.png&quot;, width = 7, height=5)</code></pre>
</div>
<div id="only-those-genes-which-are-significant-in-dex-2h-ns-or-significant-in-same-direction-as-dex-2h-for-mif2h-dex-and-cort1132h-dex-but-are-significant-in-the-opposite-direction-as-dex-2h-in-kh-1032h-dex" class="section level2">
<h2>Only those genes which are significant in Dex 2h, ns or significant in same direction as Dex 2h for mif&gt;2h dex and cort113&gt;2h dex but are significant in the opposite direction as Dex 2h in KH-103&gt;2h dex</h2>
<pre class="r"><code>merged162$category3 &lt;- (merged162$FDR &lt; 0.05) &amp; 
  (merged162$logFC * merged162$logFC_MIF &gt; 0) &amp; # 2h DEX and MIF point in same direction
  (merged162$logFC * merged162$logFC_cort &gt; 0) &amp; # 2h DEX and cort point in same direction
  (merged162$logFC * merged162$logFC_KH &lt; 0) &amp; # 2h DEX and KH have different logFC signs
  (merged162$FDR_KH &gt; 0) # KH is significant

rnames_flipped_cat3 &lt;- merged162 %&gt;% dplyr::filter(category3 == TRUE) %&gt;% pull(rowname)

# fig 6c.2c
plot_logFCs_162(rnames_flipped_cat3, &quot;Category 3&quot;)</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-10-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../data/00_paper_figures/6c2c.png&quot;, width = 14, height=5)</code></pre>
</div>
<div id="only-those-genes-which-are-significant-in-dex-2h-and-are-significant-in-the-opposite-direction-as-dex-2h-in-all-three-conditions-of-kh-1032h-dex-mif2h-dex-cort1132h-dex" class="section level2">
<h2>Only those genes which are significant in Dex 2h, and are significant in the opposite direction as Dex 2h in all three conditions of KH-103&gt;2h dex, MIF&gt;2h dex, Cort113&gt;2h dex</h2>
<pre class="r"><code>merged162$category4 &lt;- (merged162$FDR &lt; 0.05) &amp; 
  (merged162$logFC * merged162$logFC_KH &lt; 0) &amp; (merged162$FDR_KH &lt; 0.05) &amp; # KH is significant in other direction as 2h DEX
  (merged162$logFC * merged162$logFC_MIF &lt; 0) &amp; (merged162$FDR_MIF &lt; 0.05) &amp; # MIF is significant in other direction as 2h DEX
  (merged162$logFC * merged162$logFC_cort &lt; 0) &amp; (merged162$FDR_cort &lt; 0.05) # cort is significant in other direction as 2h DEX

rnames_flipped_cat4 &lt;- merged162 %&gt;% dplyr::filter(category4 == TRUE) %&gt;% pull(rowname)

# fig 6c.2d
plot_logFCs_162(rnames_flipped_cat4, &quot;Category 4&quot;)</code></pre>
<p><img src="figure/Q3-PROTAC-reversal.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(filename=&quot;../data/00_paper_figures/6c2d.png&quot;, width = 7, height=5)</code></pre>
<div id="save-file-to-excel" class="section level3">
<h3>Save file to excel</h3>
<pre class="r"><code># setup excel file
wb &lt;- createWorkbook()
# add worksheets and add data
addWorksheet(wb, sheetName = &quot;2h&gt;16h&quot;)
addWorksheet(wb, sheetName = &quot;16h&gt;2h&quot;)
writeData(wb, &quot;2h&gt;16h&quot;, merged216)
writeData(wb, &quot;16h&gt;2h&quot;, merged162)
# write final excel to disk
saveWorkbook(wb, &quot;../docs/downloads/Q3_protac_flipped_DE.xlsx&quot;, overwrite=TRUE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.6    hrbrthemes_0.8.6 openxlsx_4.2.5   dplyr_1.0.9     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3      svglite_2.1.0     assertthat_0.2.1  rprojroot_2.0.3  
 [5] digest_0.6.29     utf8_1.2.2        R6_2.5.1          plyr_1.8.7       
 [9] evaluate_0.15     httr_1.4.3        highr_0.9         pillar_1.7.0     
[13] gdtools_0.2.4     rlang_1.0.3       rstudioapi_0.13   extrafontdb_1.0  
[17] whisker_0.4       jquerylib_0.1.4   rmarkdown_2.14    textshaping_0.3.6
[21] labeling_0.4.2    webshot_0.5.3     extrafont_0.18    stringr_1.4.0    
[25] munsell_0.5.0     compiler_4.1.0    httpuv_1.6.5      xfun_0.31        
[29] pkgconfig_2.0.3   systemfonts_1.0.4 htmltools_0.5.2   tidyselect_1.1.2 
[33] tibble_3.1.7      workflowr_1.7.0   viridisLite_0.4.0 fansi_1.0.3      
[37] crayon_1.5.1      withr_2.5.0       later_1.3.0       grid_4.1.0       
[41] jsonlite_1.8.0    Rttf2pt1_1.3.10   gtable_0.3.0      lifecycle_1.0.1  
[45] DBI_1.1.3         git2r_0.30.1      magrittr_2.0.3    scales_1.2.0     
[49] zip_2.2.0         cli_3.3.0         stringi_1.7.6     farver_2.1.1     
[53] reshape2_1.4.4    fs_1.5.2          promises_1.2.0.1  xml2_1.3.3       
[57] bslib_0.3.1       ellipsis_0.3.2    ragg_1.2.2        generics_0.1.3   
[61] vctrs_0.4.1       kableExtra_1.3.4  tools_4.1.0       glue_1.6.2       
[65] purrr_0.3.4       fastmap_1.1.0     yaml_2.3.5        colorspace_2.0-3 
[69] rvest_1.0.2       knitr_1.39        sass_0.4.1       </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
