<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-07-07" />

<title>Q4: GR-binding to PROTAC genes</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Q4: GR-binding to PROTAC genes</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-07-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1targetblank779ceb2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1" target="_blank">779ceb2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1targetblank779ceb2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1" target="_blank">779ceb2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig

Untracked files:
    Untracked:  analysis/08-Differential_binding.Rmd
    Untracked:  data/reddy_bed_DEX6h_r123.bed.gz
    Untracked:  data/results/SE.processed.rds
    Untracked:  data/single_replicate_peaks/

Unstaged changes:
    Modified:   analysis/01_download_files.Rmd
    Modified:   analysis/02_exploratory_analysis.Rmd
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q2-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q4-GR-binding-PROTAC-genes.Rmd
    Modified:   analysis/admin.Rmd
    Modified:   output/Q2_GR_bound_DE_genes.xlsx
    Modified:   output/Q4_KH_DE_GR_bound.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Q4-GR-binding-PROTAC-genes.Rmd</code>) and HTML (<code>docs/Q4-GR-binding-PROTAC-genes.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/analysis/Q4-GR-binding-PROTAC-genes.Rmd" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/Q4-GR-binding-PROTAC-genes.html" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/analysis/Q4-GR-binding-PROTAC-genes.Rmd" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/207afa653f1bc6b51cf5f07acfe8c498be77c57b/docs/Q4-GR-binding-PROTAC-genes.html" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="question-4-binding-of-gr-to-kh-protac-treated-de-genes" class="section level3">
<h3>Question 4: Binding of GR to KH PROTAC treated DE genes?</h3>
<p>Q: Is there a direct binding of GR to the 13 genes shown to be affected by the KH PROTAC?</p>
<p><strong>Approach:</strong></p>
<ol style="list-style-type: decimal">
<li>What are the 13 PROTAC genes?</li>
<li>Does GR bind to them?</li>
</ol>
<pre class="r"><code># load the peaks
peakAnno &lt;- readRDS(&quot;../data/EOS_files/Q2_annotated_peaks.rds&quot;)
# load our DE results
res_int_all &lt;- readRDS(file=&quot;../data/results/DE_our_data.rds&quot;)

# get the data for the 18h KH-PROTAC treatment
kh.protac &lt;- res_int_all[[&quot;18h KH-103&quot;]]

# load the results from the peak frequencies
peakfreq &lt;- readRDS(file=&quot;../data/EOS_files/Q2_gene_peaks.rds&quot;)

# merge peak frequency onto table and add binary variable
peakAnno_reduced &lt;- as.data.frame(peakAnno) %&gt;% 
  dplyr::select(&quot;nearestTSS.gene_name&quot;, &quot;distance2nearestTSS&quot;, &quot;class&quot;) %&gt;% 
  dplyr::rename(Gene=nearestTSS.gene_name) %&gt;%
  dplyr::filter(Gene != &quot;&quot;) %&gt;%
  group_by(Gene) %&gt;%
  dplyr::slice(which.min(abs(distance2nearestTSS)))

kh.protac &lt;- left_join(kh.protac %&gt;% mutate(Gene = rownames(kh.protac)), 
          peakfreq,
          by=&quot;Gene&quot;) %&gt;%
  tidyr::replace_na(list(Freq=0)) %&gt;%
  mutate(gc_bound = Freq &gt; 0) %&gt;%
  dplyr::rename(chip_peaks=Freq) %&gt;% 
  dplyr::select(Gene, everything()) %&gt;%
  dplyr::left_join(peakAnno_reduced, by=&quot;Gene&quot;)

write.xlsx(file=&quot;../output/Q4_KH_DE_GR_bound.xlsx&quot;, kh.protac, rowNames=TRUE)</code></pre>
</div>
<div id="does-gr-gc-bind-to-a-region-closely-associated-with-any-of-these-13-genes" class="section level3">
<h3>Does GR-GC bind to a region closely associated with any of these 13 genes?</h3>
<p>We have 13 genes that are differentially expressed at an &lt;0.05 FDR level. The Reddy lab has analysed where the glucorticoid receptor (NR3C1) binds to after treatment with dexamethasone for 12h. We find that the GR is found to bind close to 5 out of the 13 differentially expressed genes.</p>
<pre class="r"><code># only keep the significant genes
kh.protac.sig &lt;- kh.protac[kh.protac$FDR&lt;0.05,]
kh.protac.sig[,c(&quot;Gene&quot;, &quot;logFC&quot;, &quot;FDR&quot;, &quot;chip_peaks&quot;, &quot;gc_bound&quot;)]</code></pre>
<pre><code>      Gene      logFC          FDR chip_peaks gc_bound
1     PER1  1.1428657 1.723020e-23          3     TRUE
2   TGFBR1  0.5510412 9.540295e-13          1     TRUE
3    FKBP5  0.7301845 2.624094e-10          7     TRUE
4   NBEAL2  0.3850139 1.572408e-04          0    FALSE
5   CAVIN2  0.3182665 6.525299e-04          1     TRUE
6  SLC38A2  0.2546515 7.788073e-04          0    FALSE
7      TEF -0.9140413 1.000325e-03          0    FALSE
8     CRY1 -0.3858283 2.141931e-03          0    FALSE
9    C5AR2  0.6589461 3.944310e-03          0    FALSE
10 ANGPTL4  0.4185862 6.415798e-03          3     TRUE
11    CRY2 -0.4867045 9.726164e-03          0    FALSE
12    CHML  0.2746874 1.352747e-02          0    FALSE
13    PER3 -0.7575232 3.989078e-02          0    FALSE</code></pre>
<pre class="r"><code>plt_gr_binding &lt;- ggplot(kh.protac.sig) +
  geom_point(aes(logFC, -log10(FDR), colour=gc_bound, size=chip_peaks)) +
  labs(title=&quot;GR binding activity for diff. expressed PROTAC genes&quot;) +
  ggrepel::geom_text_repel(data=kh.protac.sig, aes(logFC, -log10(FDR), label=Gene, size=1)) +
  theme_ipsum_rc()

plt_gr_binding</code></pre>
<p><img src="figure/Q4-GR-binding-PROTAC-genes.Rmd/plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-1">
Past versions of plot-1.png
</button>
</p>
<div id="fig-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/figure/Q4-GR-binding-PROTAC-genes.Rmd/plot-1.png" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q4_1_kh_gr_binding.png&quot;, plt_gr_binding)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div id="plot-signal-tracks" class="section level1">
<h1>Plot signal tracks</h1>
<div id="peaks-found-in-bed-file-not-all-13-genes-included" class="section level2">
<h2>Peaks found in bed file (not all 13 genes included)</h2>
<pre class="r"><code>peaks13g &lt;- peakAnno[peakAnno$nearestTSS.gene_name %in% kh.protac.sig$Gene,]


tracks &lt;-  list(&quot;2h-DEX&quot;=&quot;../data/reddy_bigwig_DEX2h_r123.bigWig&quot;,
                &quot;6h-DEX&quot;=&quot;../data/reddy_bigwig_DEX6h_r123.bigWig&quot;,
                &quot;12h-DEX&quot;=&quot;../data/reddy_bigwig_DEX12h_r123.bigWig&quot;)
signals &lt;- signal2Matrix(tracks, peaks13g, w=5, extend=500)</code></pre>
<pre><code>Warning in .checkRegions(filepaths, regions, verbose = verbose, trimOOR = FALSE): 8 region(s) (53%) were excluded because they were out of range of (some of) the file(s).
This usually happens when the genome annotation used for the files differs from that on which the regions were based.</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>png(file=&quot;../docs/assets/Q4_2_kh_peak_tracks.png&quot;, width=12,height=7,units=&quot;in&quot;,res=1200)
h1 &lt;- plotEnrichedHeatmaps(signals) #, left_annotation=peaks13g$nearestTSS.gene_name)#, width=unit(12, &quot;cm&quot;))
h1
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>h1</code></pre>
<p><img src="figure/Q4-GR-binding-PROTAC-genes.Rmd/plot%20signal%20tracks-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-the-signal-around-the-tss-of-each-of-the-13-genes-all-transcripts" class="section level2">
<h2>Plot the signal around the TSS of each of the 13 genes (all transcripts)</h2>
<p>± 10’000 bps from TSS</p>
<pre class="r"><code>ah &lt;- AnnotationHub()</code></pre>
<pre><code>snapshotDate(): 2021-10-20</code></pre>
<pre class="r"><code>ensdb &lt;- ah[[&quot;AH98047&quot;]] # AH89211</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>require(&quot;ensembldb&quot;)</code></pre>
<pre><code>Warning: package &#39;GenomicFeatures&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.1.1</code></pre>
<pre><code>Warning: package &#39;AnnotationFilter&#39; was built under R version 4.1.1</code></pre>
<pre class="r"><code>mygenes.transcripts &lt;-  subset(transcripts(ensdb, columns=c(&quot;tx_id&quot;, &quot;tx_name&quot;,&quot;gene_id&quot;,&quot;gene_name&quot;)), gene_name %in% kh.protac.sig$Gene)
mygenes.tss &lt;-  resize(mygenes.transcripts, width=2000, fix=&#39;start&#39;)
seqlevelsStyle(mygenes.tss) &lt;- &quot;UCSC&quot;</code></pre>
<pre><code>Warning in (function (seqlevels, genome, new_style) : cannot switch some of
GRCh38&#39;s seqlevels from NCBI to UCSC style</code></pre>
<pre class="r"><code>signals2 &lt;- signal2Matrix(tracks, mygenes.tss, w=10, extend=10000)</code></pre>
<pre><code>Warning in .checkRegions(filepaths, regions, verbose = verbose, trimOOR = FALSE): 98 region(s) (87%) were excluded because they were out of range of (some of) the file(s).
This usually happens when the genome annotation used for the files differs from that on which the regions were based.</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>png(file=&quot;../docs/assets/Q4_3_kh_all_txs.png&quot;, width=12,height=7,units=&quot;in&quot;,res=1200)
h2 &lt;- plotEnrichedHeatmaps(signals2) #, left_annotation=peaks13g$nearestTSS.gene_name)#, width=unit(12, &quot;cm&quot;))
h2
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>h2</code></pre>
<p><img src="figure/Q4-GR-binding-PROTAC-genes.Rmd/gene%20TSS%20all%20transcripts-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-the-signal-around-the-tss-of-each-of-the-13-genes-one-transcript-per-gene" class="section level2">
<h2>Plot the signal around the TSS of each of the 13 genes (one transcript per gene)</h2>
<p>± 10’000 bps from TSS</p>
<pre class="r"><code>mygenes_unique &lt;- mygenes.tss[!duplicated(mygenes.tss$gene_name),]

signals3 &lt;- signal2Matrix(tracks, mygenes_unique, w=10, extend=10000)</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX2h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX6h_r123.bigWig</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>png(file=&quot;../docs/assets/Q4_4_kh_unique_gene_tss.png&quot;, width=12,height=7,units=&quot;in&quot;,res=1200)
h3 &lt;- plotEnrichedHeatmaps(signals3)
h3
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>h3</code></pre>
<p><img src="figure/Q4-GR-binding-PROTAC-genes.Rmd/all%20genes%20unique%20TSS-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.18.4        AnnotationFilter_1.18.0 GenomicFeatures_1.46.5 
 [4] AnnotationDbi_1.56.2    Biobase_2.54.0          AnnotationHub_3.2.2    
 [7] BiocFileCache_2.2.1     dbplyr_2.2.1            epiwraps_0.99.42       
[10] EnrichedHeatmap_1.24.0  ComplexHeatmap_2.10.0   GenomicRanges_1.46.1   
[13] GenomeInfoDb_1.30.1     IRanges_2.28.0          S4Vectors_0.32.4       
[16] BiocGenerics_0.40.0     hrbrthemes_0.8.6        ggplot2_3.3.6          
[19] dplyr_1.0.9             openxlsx_4.2.5         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    tidyselect_1.1.2             
  [3] RSQLite_2.2.14                htmlwidgets_1.5.4            
  [5] BiocParallel_1.28.3           munsell_0.5.0                
  [7] ragg_1.2.2                    codetools_0.2-18             
  [9] interp_1.1-2                  withr_2.5.0                  
 [11] colorspace_2.0-3              filelock_1.0.2               
 [13] highr_0.9                     knitr_1.39                   
 [15] rstudioapi_0.13               Rttf2pt1_1.3.10              
 [17] labeling_0.4.2                MatrixGenerics_1.6.0         
 [19] git2r_0.30.1                  GenomeInfoDbData_1.2.7       
 [21] farver_2.1.1                  bit64_4.0.5                  
 [23] rprojroot_2.0.3               vctrs_0.4.1                  
 [25] generics_0.1.3                xfun_0.31                    
 [27] biovizBase_1.42.0             R6_2.5.1                     
 [29] doParallel_1.0.17             clue_0.3-61                  
 [31] locfit_1.5-9.5                bitops_1.0-7                 
 [33] cachem_1.0.6                  DelayedArray_0.20.0          
 [35] assertthat_0.2.1              promises_1.2.0.1             
 [37] BiocIO_1.4.0                  scales_1.2.0                 
 [39] nnet_7.3-17                   gtable_0.3.0                 
 [41] workflowr_1.7.0               rlang_1.0.3                  
 [43] systemfonts_1.0.4             GlobalOptions_0.1.2          
 [45] splines_4.1.0                 rtracklayer_1.54.0           
 [47] extrafontdb_1.0               lazyeval_0.2.2               
 [49] dichromat_2.0-0.1             checkmate_2.1.0              
 [51] BiocManager_1.30.18           yaml_2.3.5                   
 [53] backports_1.4.1               httpuv_1.6.5                 
 [55] Hmisc_4.7-0                   extrafont_0.18               
 [57] tools_4.1.0                   ellipsis_0.3.2               
 [59] jquerylib_0.1.4               RColorBrewer_1.1-3           
 [61] Rcpp_1.0.8.3                  plyr_1.8.7                   
 [63] base64enc_0.1-3               progress_1.2.2               
 [65] zlibbioc_1.40.0               purrr_0.3.4                  
 [67] RCurl_1.98-1.7                prettyunits_1.1.1            
 [69] rpart_4.1.16                  deldir_1.0-6                 
 [71] GetoptLong_1.0.5              cowplot_1.1.1                
 [73] GenomicFiles_1.30.0           SummarizedExperiment_1.24.0  
 [75] ggrepel_0.9.1                 cluster_2.1.3                
 [77] fs_1.5.2                      magrittr_2.0.3               
 [79] magick_2.7.3                  data.table_1.14.2            
 [81] circlize_0.4.15               whisker_0.4                  
 [83] ProtGenerics_1.26.0           matrixStats_0.62.0           
 [85] hms_1.1.1                     mime_0.12                    
 [87] evaluate_0.15                 xtable_1.8-4                 
 [89] XML_3.99-0.10                 jpeg_0.1-9                   
 [91] gridExtra_2.3                 shape_1.4.6                  
 [93] compiler_4.1.0                biomaRt_2.50.3               
 [95] tibble_3.1.7                  crayon_1.5.1                 
 [97] htmltools_0.5.2               later_1.3.0                  
 [99] Formula_1.2-4                 tidyr_1.2.0                  
[101] DBI_1.1.3                     rappdirs_0.3.3               
[103] Matrix_1.4-1                  cli_3.3.0                    
[105] parallel_4.1.0                Gviz_1.38.4                  
[107] pkgconfig_2.0.3               GenomicAlignments_1.30.0     
[109] foreign_0.8-82                xml2_1.3.3                   
[111] foreach_1.5.2                 bslib_0.3.1                  
[113] XVector_0.34.0                stringr_1.4.0                
[115] VariantAnnotation_1.40.0      digest_0.6.29                
[117] Biostrings_2.62.0             rmarkdown_2.14               
[119] htmlTable_2.4.0               edgeR_3.36.0                 
[121] gdtools_0.2.4                 restfulr_0.0.15              
[123] curl_4.3.2                    shiny_1.7.1                  
[125] Rsamtools_2.10.0              rjson_0.2.21                 
[127] lifecycle_1.0.1               jsonlite_1.8.0               
[129] viridisLite_0.4.0             limma_3.50.3                 
[131] BSgenome_1.62.0               fansi_1.0.3                  
[133] pillar_1.7.0                  lattice_0.20-45              
[135] KEGGREST_1.34.0               fastmap_1.1.0                
[137] httr_1.4.3                    survival_3.3-1               
[139] interactiveDisplayBase_1.32.0 glue_1.6.2                   
[141] zip_2.2.0                     UpSetR_1.4.0                 
[143] png_0.1-7                     iterators_1.0.14             
[145] BiocVersion_3.14.0            bit_4.0.4                    
[147] stringi_1.7.6                 sass_0.4.1                   
[149] blob_1.2.3                    textshaping_0.3.6            
[151] latticeExtra_0.6-30           memoise_2.0.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
