<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-07-07" />

<title>Q7: PRX sequences</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Q7: PRX sequences</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-07-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1targetblank779ceb2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1" target="_blank">779ceb2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1targetblank779ceb2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/779ceb2b2c6179b9fdc05cc5d049e81ecdfb6bd1" target="_blank">779ceb2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig

Untracked files:
    Untracked:  analysis/08-Differential_binding.Rmd
    Untracked:  data/reddy_bed_DEX6h_r123.bed.gz
    Untracked:  data/results/SE.processed.rds
    Untracked:  data/single_replicate_peaks/

Unstaged changes:
    Modified:   analysis/01_download_files.Rmd
    Modified:   analysis/02_exploratory_analysis.Rmd
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q2-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q4-GR-binding-PROTAC-genes.Rmd
    Modified:   analysis/admin.Rmd
    Modified:   output/Q2_GR_bound_DE_genes.xlsx
    Modified:   output/Q4_KH_DE_GR_bound.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Q7-PRX-sequences.Rmd</code>) and HTML (<code>docs/Q7-PRX-sequences.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/analysis/Q7-PRX-sequences.Rmd" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Glucorticoid-protacs/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/Q7-PRX-sequences.html" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
<td>
added remaining questions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/207afa653f1bc6b51cf5f07acfe8c498be77c57b/analysis/Q7-PRX-sequences.Rmd" target="_blank">207afa6</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-21
</td>
<td>
added question 3
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="question-7-does-gc-gr-bind-to-prx-sequences" class="section level1">
<h1>Question 7: Does GC-GR bind to PRX sequences?</h1>
<div id="approach" class="section level3">
<h3>Approach</h3>
<p>Q7: Scan open areas (ATAC-seq) for PXR binding (NR1I2). How many of those are bound by GR?</p>
<ol style="list-style-type: decimal">
<li>Scan for PRX sequences.</li>
<li>Is there an overlap between the peaks of GR and the PRX sequences?</li>
</ol>
<pre class="r"><code># ah &lt;- AnnotationHub()
# gr_peaks &lt;- import.bed15(&quot;../data/reddy_bed_DEX12h_r123.bed&quot;, format=&quot;narrowPeak&quot;)
# ensdb &lt;- ah[[&quot;AH98047&quot;]]
# gr_peaks_annotated &lt;- annotateRegions(gr_peaks, ensdb)

gr_peaks &lt;- readRDS(&quot;../data/EOS_files/Q2_annotated_peaks.rds&quot;)
# change from chr (ensembl format) to just numbers
# seqlevels(peaks)&lt;- sub(&#39;chr&#39;,&#39;&#39;,seqlevels(peaks))</code></pre>
<pre class="r"><code>peak_centers &lt;- resize(gr_peaks, fix=&quot;center&quot;, width=100)
# we get the genome sequence from a download from the UCSC, human genome wasnt found in search in annotation hub
genome &lt;- BSgenome::getBSgenome(&quot;BSgenome.Hsapiens.UCSC.hg38&quot;, masked=FALSE, load.only=FALSE)
# we get the sequences corresponding to the peak centers:
peak_seqs &lt;- memes::get_sequence(peak_centers, genome)
peak_seqs</code></pre>
<pre><code>DNAStringSet object of length 7920:
       width seq                                            names               
   [1]   100 TCCTTCAGACGTCTGCCTACAG...GACCTGTGCAAGCACAAGGGA chr2:6186891-6186990
   [2]   100 GTTTATCACTATCTCCCACACC...GTCATAGGCTGATTTCAAAAT chr2:8004379-8004478
   [3]   100 CTGCCTGGAATTCAACTAGAAC...ATACGTGCTGAAAATGCATAC chr2:8121144-8121243
   [4]   100 AAACTAAGAATCAGAGCACTCA...GGAGAGTTTATAGCGCAGAAT chr2:8306813-8306912
   [5]   100 AAAGAGTGATTCATTCTGGTAT...CATGATATGTTTGCCCTGGCT chr2:8307101-8307200
   ...   ... ...
[7916]   100 ATGAACACACCCTGTATGACAA...ACGGCTTTGGCTTTGATGTCA chr18:75490029-75...
[7917]   100 ATAGCTGCTGCTATAGCTGGCG...ACATTCCTTTGGAAGCTTAGA chr18:75923351-75...
[7918]   100 GGCATTTATCCCAGAGAAATGA...AACCAAAACCTGGAAATAATC chr18:77128033-77...
[7919]   100 TCAAGTCTTTGTCATGGAGACC...AGTGTGCACATGTACAGACTG chr18:77560743-77...
[7920]   100 TTTAAATGAAACGGTGAGATGC...TATCGGGAAGCGTTACGTAAG chr18:79791689-79...</code></pre>
</div>
<div id="get-the-motif-for-pxr-nr1i2" class="section level2">
<h2>Get the motif for PXR (NR1I2)</h2>
<pre class="r"><code># we search for &quot;CTCF&quot; in the motif database
motifs &lt;- query(MotifDb, &quot;NR1I2&quot;)
# there are several matching motifs:
names(motifs)</code></pre>
<pre><code>[1] &quot;Hsapiens-HOCOMOCOv10-NR1I2_HUMAN.H10MO.C&quot;              
[2] &quot;Hsapiens-HOCOMOCOv10-NR1I2_HUMAN.H10MO.S&quot;              
[3] &quot;Mmusculus-HOCOMOCOv10-NR1I2_MOUSE.H10MO.C&quot;             
[4] &quot;Mmusculus-HOCOMOCOv10-NR1I2_MOUSE.H10MO.S&quot;             
[5] &quot;Hsapiens-HOCOMOCOv11-core-C-NR1I2_HUMAN.H11MO.0.C&quot;     
[6] &quot;Hsapiens-HOCOMOCOv11-secondary-D-NR1I2_HUMAN.H11MO.1.D&quot;
[7] &quot;Hsapiens-SwissRegulon-NR1I2.SwissRegulon&quot;              </code></pre>
<pre class="r"><code># we select one:
motif &lt;- motifs[[&quot;Hsapiens-HOCOMOCOv11-core-C-NR1I2_HUMAN.H11MO.0.C&quot;]]
# we visualize it:
plt_motif &lt;- view_motifs(motif) +
  labs(title=&quot;PXR motif (NR1I2)&quot;) +
  theme_ipsum_rc()
plt_motif</code></pre>
<p><img src="figure/Q7-PRX-sequences.Rmd/motif-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-1">
Past versions of motif-1.png
</button>
</p>
<div id="fig-motif-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/figure/Q7-PRX-sequences.Rmd/motif-1.png" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q7_1_motif.png&quot;, plt_motif)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div id="how-many-of-the-gr-peaks-contain-a-prx-motif" class="section level2">
<h2>How many of the GR peaks contain a PRX motif?</h2>
<pre class="r"><code># you might eventually need to add the meme_path argument:
moi &lt;- memes::runFimo(peak_seqs,  convert_motifs(motif), meme_path = &quot;/Users/dominiquepaul/.meme/bin&quot;)
moi</code></pre>
<pre><code>GRanges object with 238 ranges and 6 metadata columns:
        seqnames            ranges strand |    motif_id motif_alt_id     score
           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;  &lt;character&gt; &lt;numeric&gt;
    [1]     chr2 20539448-20539466      - |       motif         &lt;NA&gt;   10.6867
    [2]     chr2 23746090-23746108      + |       motif         &lt;NA&gt;   10.6145
    [3]     chr2 28597627-28597645      - |       motif         &lt;NA&gt;   11.9277
    [4]     chr2 30511241-30511259      - |       motif         &lt;NA&gt;   11.7952
    [5]     chr2 33153152-33153170      - |       motif         &lt;NA&gt;   12.2048
    ...      ...               ...    ... .         ...          ...       ...
  [234]    chr18   8730468-8730486      + |       motif         &lt;NA&gt;   10.9880
  [235]    chr18   9905762-9905780      - |       motif         &lt;NA&gt;   15.8193
  [236]    chr18 21439884-21439902      + |       motif         &lt;NA&gt;   11.9398
  [237]    chr18 32383241-32383259      + |       motif         &lt;NA&gt;   14.0482
  [238]    chr18 58362551-58362569      + |       motif         &lt;NA&gt;   10.7229
           pvalue    qvalue    matched_sequence
        &lt;numeric&gt; &lt;numeric&gt;         &lt;character&gt;
    [1]  6.80e-05        NA GGAGCTCCTGGAAGCACAG
    [2]  7.07e-05        NA AAGGCTTCCTTGAGGTTAG
    [3]  3.39e-05        NA TGAGGTGCACAGAGGTTAA
    [4]  3.66e-05        NA TCAGATCACTTGAGGTCAG
    [5]  2.87e-05        NA AGAGCCCAAAAAAGTACAA
    ...       ...       ...                 ...
  [234]  5.78e-05        NA GATAGTTAATAAAGGCCAG
  [235]  2.18e-06        NA AGAGGGTAGAAAAGTTCAC
  [236]  3.37e-05        NA TTAGCTGATTAAAGTACAG
  [237]  8.58e-06        NA AGATATCAGAAAAGTTCAC
  [238]  6.67e-05        NA TTAGTTCCTATGGGGTCAT
  -------
  seqinfo: 22 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code>cat(paste0(&quot;We have &quot;, length(peak_seqs), &quot; GR peaks, of which &quot;, length(moi), &quot; (&quot;, round(length(moi)/length(peak_seqs),4)*100, &quot;%) contain a PRX motif&quot;))</code></pre>
<pre><code>We have 7920 GR peaks, of which 238 (3.01%) contain a PRX motif</code></pre>
<pre class="r"><code>saveRDS(moi, file=&quot;../data/EOS_files/pxr_peaks.rds&quot;)</code></pre>
</div>
<div id="what-do-the-signal-tracks-look-like-around-the-prx-motifs-found-in-the-accessible-dna" class="section level2">
<h2>What do the signal tracks look like around the PRX motifs found in the accessible DNA?</h2>
<pre class="r"><code># Subset for open ATAC-seq data
atac_peaks &lt;- import.bed15(&quot;../data/reddy_bed_ATAC_control.bed.gz&quot;, format=&quot;narrowPeak&quot;)
atac_centers &lt;- resize(atac_peaks, fix=&quot;center&quot;, width=100)
# we get the sequences corresponding to the peak centers:
atac_peak_seqs &lt;- memes::get_sequence(atac_centers, genome)
 
# Subset for PXR motifs
pxr_in_atac &lt;- memes::runFimo(atac_peak_seqs,  convert_motifs(motif), meme_path = &quot;/Users/dominiquepaul/.meme/bin&quot;)
pxr_in_atac</code></pre>
<pre><code>GRanges object with 4510 ranges and 6 metadata columns:
         seqnames              ranges strand |    motif_id motif_alt_id
            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;  &lt;character&gt;
     [1]     chr1 100299915-100299933      + |       motif         &lt;NA&gt;
     [2]     chr1   10459401-10459419      - |       motif         &lt;NA&gt;
     [3]     chr1   10459401-10459419      - |       motif         &lt;NA&gt;
     [4]     chr1 107572804-107572822      - |       motif         &lt;NA&gt;
     [5]     chr1 109838955-109838973      - |       motif         &lt;NA&gt;
     ...      ...                 ...    ... .         ...          ...
  [4506]     chrX     9141669-9141687      - |       motif         &lt;NA&gt;
  [4507]     chrY   10660340-10660358      - |       motif         &lt;NA&gt;
  [4508]     chrY   17817551-17817569      + |       motif         &lt;NA&gt;
  [4509]     chrY   17817551-17817569      + |       motif         &lt;NA&gt;
  [4510]     chrY   17817551-17817569      + |       motif         &lt;NA&gt;
             score    pvalue    qvalue    matched_sequence
         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;         &lt;character&gt;
     [1]  10.19280  8.81e-05        NA TACTGTCAAGGGAGCACAG
     [2]  14.68670  5.37e-06        NA AGGGGGTATAAAAGGTCAT
     [3]  14.68670  5.37e-06        NA AGGGGGTATAAAAGGTCAT
     [4]   9.98795  9.77e-05        NA AATTGTCCAACAAGGTGAC
     [5]  11.71080  3.85e-05        NA CAAGGTGAGAAGAGTTGAA
     ...       ...       ...       ...                 ...
  [4506]   12.4096  2.53e-05        NA GGAATTCCATGGAGGTGAT
  [4507]   11.7108  3.85e-05        NA GGAGTTGATTGAAGTTGAG
  [4508]   10.5783  7.21e-05        NA TAAGTTCATGTGAGCTTAC
  [4509]   10.5783  7.21e-05        NA TAAGTTCATGTGAGCTTAC
  [4510]   10.5783  7.21e-05        NA TAAGTTCATGTGAGCTTAC
  -------
  seqinfo: 24 sequences from an unspecified genome; no seqlengths</code></pre>
<pre class="r"><code># Plot signal tracks
## Load bigwig data
tracks &lt;-  list(#&quot;2h-DEX&quot;=&quot;../data/reddy_bigwig_DEX2h_r123.bigWig&quot;,
                &quot;12h-DEX&quot;=&quot;../data/reddy_bigwig_DEX12h_r123.bigWig&quot;)
motifs_plt &lt;- pxr_in_atac[pxr_in_atac$score &gt; 15]
# configure the chromosome names to not include the &quot;chr&quot;
# seqlevels(motifs_plt) &lt;- stringr::str_replace(seqlevels(motifs_plt), &quot;chr&quot;, &quot;&quot;)
# error occurs here
signals &lt;- signal2Matrix(tracks, motifs_plt, w=5, extend=500)</code></pre>
<pre><code>Reading ../data/reddy_bigwig_DEX12h_r123.bigWig</code></pre>
<pre class="r"><code>plt_signal &lt;- ggplotify::as.ggplot(grid::grid.grabExpr(ComplexHeatmap::draw(plotEnrichedHeatmaps(signals)))) +
  theme_ipsum_rc() +
  labs(title=&quot;Signal tracks for PXR motifs found in accessible DNA&quot;,
       x=&quot;&quot;, y=&quot;&quot;) +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank())
plt_signal</code></pre>
<p><img src="figure/Q7-PRX-sequences.Rmd/signal%20tracks-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signal-tracks-1">
Past versions of “signal tracks-1.png”
</button>
</p>
<div id="fig-no-spaces-signal-tracks-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Glucorticoid-protacs/blob/dd2072ef1ad50bfe6745b0518bcdcdd9add31029/docs/figure/Q7-PRX-sequences.Rmd/signal tracks-1.png" target="_blank">dd2072e</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-06-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(file=&quot;../docs/assets/Q7_2_signal.png&quot;, plt_signal, width=8, height=8)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] BSgenome.Hsapiens.UCSC.hg38_1.4.4 hrbrthemes_0.8.6                 
 [3] universalmotif_1.12.4             BSgenome_1.62.0                  
 [5] ggplot2_3.3.6                     MotifDb_1.36.0                   
 [7] AnnotationHub_3.2.2               BiocFileCache_2.2.1              
 [9] dbplyr_2.2.1                      Biostrings_2.62.0                
[11] XVector_0.34.0                    memes_1.2.5                      
[13] epiwraps_0.99.42                  EnrichedHeatmap_1.24.0           
[15] ComplexHeatmap_2.10.0             rtracklayer_1.54.0               
[17] GenomicRanges_1.46.1              GenomeInfoDb_1.30.1              
[19] IRanges_2.28.0                    S4Vectors_0.32.4                 
[21] BiocGenerics_0.40.0              

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    R.utils_2.12.0               
  [3] tidyselect_1.1.2              RSQLite_2.2.14               
  [5] AnnotationDbi_1.56.2          htmlwidgets_1.5.4            
  [7] BiocParallel_1.28.3           munsell_0.5.0                
  [9] ragg_1.2.2                    codetools_0.2-18             
 [11] interp_1.1-2                  withr_2.5.0                  
 [13] colorspace_2.0-3              Biobase_2.54.0               
 [15] filelock_1.0.2                highr_0.9                    
 [17] knitr_1.39                    rstudioapi_0.13              
 [19] Rttf2pt1_1.3.10               labeling_0.4.2               
 [21] MatrixGenerics_1.6.0          git2r_0.30.1                 
 [23] GenomeInfoDbData_1.2.7        farver_2.1.1                 
 [25] bit64_4.0.5                   rprojroot_2.0.3              
 [27] vctrs_0.4.1                   generics_0.1.3               
 [29] xfun_0.31                     biovizBase_1.42.0            
 [31] ggseqlogo_0.1                 R6_2.5.1                     
 [33] doParallel_1.0.17             splitstackshape_1.4.8        
 [35] clue_0.3-61                   locfit_1.5-9.5               
 [37] AnnotationFilter_1.18.0       gridGraphics_0.5-1           
 [39] bitops_1.0-7                  cachem_1.0.6                 
 [41] DelayedArray_0.20.0           assertthat_0.2.1             
 [43] vroom_1.5.7                   promises_1.2.0.1             
 [45] BiocIO_1.4.0                  scales_1.2.0                 
 [47] nnet_7.3-17                   gtable_0.3.0                 
 [49] processx_3.6.1                ensembldb_2.18.4             
 [51] workflowr_1.7.0               rlang_1.0.3                  
 [53] systemfonts_1.0.4             GlobalOptions_0.1.2          
 [55] splines_4.1.0                 extrafontdb_1.0              
 [57] lazyeval_0.2.2                dichromat_2.0-0.1            
 [59] checkmate_2.1.0               BiocManager_1.30.18          
 [61] yaml_2.3.5                    GenomicFeatures_1.46.5       
 [63] backports_1.4.1               httpuv_1.6.5                 
 [65] Hmisc_4.7-0                   extrafont_0.18               
 [67] tools_4.1.0                   ggplotify_0.1.0              
 [69] ellipsis_0.3.2                jquerylib_0.1.4              
 [71] RColorBrewer_1.1-3            Rcpp_1.0.8.3                 
 [73] plyr_1.8.7                    base64enc_0.1-3              
 [75] progress_1.2.2                zlibbioc_1.40.0              
 [77] purrr_0.3.4                   RCurl_1.98-1.7               
 [79] ps_1.7.1                      prettyunits_1.1.1            
 [81] rpart_4.1.16                  deldir_1.0-6                 
 [83] GetoptLong_1.0.5              cowplot_1.1.1                
 [85] GenomicFiles_1.30.0           SummarizedExperiment_1.24.0  
 [87] cluster_2.1.3                 fs_1.5.2                     
 [89] magrittr_2.0.3                magick_2.7.3                 
 [91] data.table_1.14.2             circlize_0.4.15              
 [93] whisker_0.4                   ProtGenerics_1.26.0          
 [95] matrixStats_0.62.0            pkgload_1.3.0                
 [97] hms_1.1.1                     mime_0.12                    
 [99] evaluate_0.15                 xtable_1.8-4                 
[101] XML_3.99-0.10                 jpeg_0.1-9                   
[103] gridExtra_2.3                 shape_1.4.6                  
[105] testthat_3.1.4                compiler_4.1.0               
[107] biomaRt_2.50.3                tibble_3.1.7                 
[109] crayon_1.5.1                  R.oo_1.25.0                  
[111] htmltools_0.5.2               later_1.3.0                  
[113] tzdb_0.3.0                    Formula_1.2-4                
[115] tidyr_1.2.0                   DBI_1.1.3                    
[117] MASS_7.3-57                   rappdirs_0.3.3               
[119] Matrix_1.4-1                  readr_2.1.2                  
[121] brio_1.1.3                    cli_3.3.0                    
[123] R.methodsS3_1.8.2             parallel_4.1.0               
[125] Gviz_1.38.4                   pkgconfig_2.0.3              
[127] GenomicAlignments_1.30.0      foreign_0.8-82               
[129] xml2_1.3.3                    foreach_1.5.2                
[131] bslib_0.3.1                   yulab.utils_0.0.5            
[133] stringr_1.4.0                 VariantAnnotation_1.40.0     
[135] digest_0.6.29                 rmarkdown_2.14               
[137] htmlTable_2.4.0               edgeR_3.36.0                 
[139] gdtools_0.2.4                 restfulr_0.0.15              
[141] curl_4.3.2                    shiny_1.7.1                  
[143] Rsamtools_2.10.0              rjson_0.2.21                 
[145] lifecycle_1.0.1               jsonlite_1.8.0               
[147] desc_1.4.1                    viridisLite_0.4.0            
[149] limma_3.50.3                  fansi_1.0.3                  
[151] pillar_1.7.0                  lattice_0.20-45              
[153] KEGGREST_1.34.0               fastmap_1.1.0                
[155] httr_1.4.3                    survival_3.3-1               
[157] interactiveDisplayBase_1.32.0 glue_1.6.2                   
[159] UpSetR_1.4.0                  png_0.1-7                    
[161] iterators_1.0.14              cmdfun_1.0.2                 
[163] BiocVersion_3.14.0            bit_4.0.4                    
[165] stringi_1.7.6                 sass_0.4.1                   
[167] blob_1.2.3                    textshaping_0.3.6            
[169] latticeExtra_0.6-30           memoise_2.0.1                
[171] dplyr_1.0.9                  </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
