<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-07-08" />

<title>Differential Binding</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GR-PROTAC-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential Binding</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-07-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>GR-PROTAC-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220515code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220515code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220515)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree0f93e11d370afd43dc357e5febfaaf55f3c52c0ftargetblank0f93e11a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/0f93e11d370afd43dc357e5febfaaf55f3c52c0f" target="_blank">0f93e11</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulGlucorticoidprotacstree0f93e11d370afd43dc357e5febfaaf55f3c52c0ftargetblank0f93e11a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Glucorticoid-protacs/tree/0f93e11d370afd43dc357e5febfaaf55f3c52c0f" target="_blank">0f93e11</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/docs/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/EOS_files/
    Ignored:    data/reddy_bigwig_DEX12h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX2h_r123.bigWig
    Ignored:    data/reddy_bigwig_DEX6h_r123.bigWig
    Ignored:    data/reddy_bigwig_JUNB_DEX12h_r123.bigWig
    Ignored:    data/single_replicate_peaks/.DS_Store
    Ignored:    data/single_replicate_peaks/control/.DS_Store
    Ignored:    data/single_replicate_peaks/control/Chip_GR_control_s2.bigWig
    Ignored:    data/single_replicate_peaks/control/Chip_GR_control_s3.bigWig
    Ignored:    data/single_replicate_peaks/dex12h/.DS_Store
    Ignored:    data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1.bigWig
    Ignored:    data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2.bigWig
    Ignored:    data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s3.bigWig
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/00_Download_files.Rmd
    Untracked:  analysis/01_Results.Rmd
    Untracked:  analysis/Q9-Differential_binding.Rmd
    Untracked:  data/Formatted_Candidate_ETG_20200305_TAD10Kb.tsv
    Untracked:  data/results/dds.rds
    Untracked:  data/single_replicate_peaks/peaks/

Unstaged changes:
    Deleted:    analysis/01_download_files.Rmd
    Deleted:    analysis/02_exploratory_analysis.Rmd
    Deleted:    analysis/08-Differential_binding.Rmd
    Modified:   analysis/Q1-Gene-overlap.Rmd
    Modified:   analysis/Q2-TF-Binding-to-DE-genes.Rmd
    Modified:   analysis/Q3-PROTAC-reversal.Rmd
    Modified:   analysis/Q4-GR-binding-PROTAC-genes.Rmd
    Modified:   analysis/about.Rmd
    Modified:   analysis/admin.Rmd
    Modified:   analysis/index.Rmd
    Modified:   data/results/SE.processed.rds
    Deleted:    data/single_replicate_peaks/control/Chip_GR_control_s2.bam.bai
    Deleted:    data/single_replicate_peaks/control/Chip_GR_control_s2_input.bam.bai
    Deleted:    data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1.bam.bai
    Deleted:    data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2_input.bam.bai
    Deleted:    output/Q2_GR_bound_DE_genes.xlsx
    Deleted:    output/Q4_KH_DE_GR_bound.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="call-peaks-bam---bed" class="section level1">
<h1>Call peaks (bam -&gt; bed)</h1>
<p>Note: We only have two control samples as the first isogenic sample was not uploaded to encode. We call the peaks manually for the 12hDEX and 0h DEX data, as no bed files were uploaded for the 0h DEX data. We call the peak for both groups here to ensure that the peak calling method applied is the same.</p>
<pre class="bash"><code>
macs2 callpeak --outdir ../data/single_replicate_peaks/peaks \
  -n Chip_GR_12hDex_s1 \
  --gsize dm \
  -t ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1.bam \
  -c ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1_input.bam \
  --nomodel \
  --extsize 200

macs2 callpeak --outdir ../data/single_replicate_peaks/peaks \
  -n Chip_GR_12hDex_s2 \
  --gsize dm \
  -t ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2.bam \
  -c ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2_input.bam \
  --nomodel \
  --extsize 200

macs2 callpeak --outdir ../data/single_replicate_peaks/peaks \
  -n Chip_GR_12hDex_s3 \
  --gsize dm \
  -t ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s3.bam \
  -c ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s3_input.bam \
  --nomodel \
  --extsize 200

macs2 callpeak --outdir ../data/single_replicate_peaks/peaks \
  -n Chip_GR_control_s2 \
  --gsize dm \
  -t ../data/single_replicate_peaks/control/Chip_GR_control_s2.bam \
  -c ../data/single_replicate_peaks/control/Chip_GR_control_s2_input.bam \
  --nomodel \
  --extsize 200

macs2 callpeak --outdir ../data/single_replicate_peaks/peaks \
  -n Chip_GR_control_s3 \
  --gsize dm \
  -t ../data/single_replicate_peaks/control/Chip_GR_control_s3.bam \
  -c ../data/single_replicate_peaks/control/Chip_GR_control_s3_input.bam \
  --nomodel \
  --extsize 200</code></pre>
</div>
<div id="differential-binding-analysis" class="section level1">
<h1>Differential binding analysis</h1>
<p>Only the sample 2 and 3 of the 12h Dex treated group have peaks. The control samples 2 and 3 (1 was not uploaded) as well as treatment sample 1 have no peaks.</p>
<pre class="r"><code># load peaks
c2 &lt;- import.bed15(&quot;../data/single_replicate_peaks/peaks/Chip_GR_control_s2_peaks.narrowPeak&quot;, format=&quot;narrowPeak&quot;)
c3 &lt;- import.bed15(&quot;../data/single_replicate_peaks/peaks/Chip_GR_control_s3_peaks.narrowPeak&quot;, format=&quot;narrowPeak&quot;)
t12_1 &lt;- import.bed15(&quot;../data/single_replicate_peaks/peaks/Chip_GR_12hDex_s1_peaks.narrowPeak&quot;, format=&quot;narrowPeak&quot;)
t12_2 &lt;- import.bed15(&quot;../data/single_replicate_peaks/peaks/Chip_GR_12hDex_s2_peaks.narrowPeak&quot;, format=&quot;narrowPeak&quot;)
t12_3 &lt;- import.bed15(&quot;../data/single_replicate_peaks/peaks/Chip_GR_12hDex_s3_peaks.narrowPeak&quot;, format=&quot;narrowPeak&quot;)

peaks &lt;- list(&quot;c2&quot;=c2, &quot;c3&quot;=c3,
              &quot;t12_1&quot;=t12_1, &quot;t12_2&quot;=t12_2, &quot;t12_3&quot;=t12_3)

bamfiles &lt;- c(&quot;../data/single_replicate_peaks/control/Chip_GR_control_s2.bam&quot;,
                 &quot;../data/single_replicate_peaks/control/Chip_GR_control_s3.bam&quot;,
                 &quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1.bam&quot;, 
                 &quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2.bam&quot;, 
                 &quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s3.bam&quot;)</code></pre>
<pre class="r"><code>regionUpset(peaks)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/show%20overlap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>For downstream analysis, we’ll want to have a single set of regions that contains all the relevant regions. Regions that occur only in one sample are not really important, so we could filter them out, and then merge the regions.</p>
<pre class="r"><code>merged_peaks &lt;- reduce(unlist(GRangesList(peaks)), with.revmap=TRUE)
merged_peaks &lt;- granges(merged_peaks[lengths(merged_peaks$revmap)&gt;1]) # we could change the &gt;1 here to require a higher minimum number of samples
merged_peaks</code></pre>
<pre><code>GRanges object with 309 ranges and 0 metadata columns:
        seqnames              ranges strand
           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;
    [1]     chr1     8004944-8005419      *
    [2]     chr1     8182234-8182649      *
    [3]     chr1     8212017-8212287      *
    [4]     chr1     8259072-8259915      *
    [5]     chr1     9436312-9436621      *
    ...      ...                 ...    ...
  [305]     chr9 110912605-110912893      *
  [306]     chr9 115944327-115944940      *
  [307]     chr9 121372531-121372804      *
  [308]     chrX   65772821-65773162      *
  [309]     chrX   72620618-72620828      *
  -------
  seqinfo: 23 sequences from an unspecified genome; no seqlengths</code></pre>
<div id="counting-the-number-of-reads-in-each-peak-in-each-sample" class="section level2">
<h2>Counting the number of reads in each peak, in each sample</h2>
<pre class="r"><code># create a vector with all the peak files in it, and giving it clean names:
# bamfiles &lt;- list.files(&quot;aligned&quot;, pattern=&quot;bam$&quot;, full=TRUE)
# we give the samples clean names:
# names(bamfiles) &lt;- gsub(&quot;\\.bam&quot;,&quot;&quot;,basename(bamfiles))

# for counting, we need to convert our GRanges object to a data.frame with the following specifications:
anno &lt;- cbind(name=as.character(merged_peaks), as.data.frame(merged_peaks))
anno$width &lt;- NULL
colnames(anno) &lt;- c(&quot;GeneID&quot;, &quot;Chr&quot;, &quot;Start&quot;, &quot;End&quot;, &quot;Strand&quot;)

fc &lt;- featureCounts( files=bamfiles,    # the files in which we want to count reads
                     isPairedEnd=FALSE,
                     annot.ext=anno,    # the regions in which we want to count reads
                     readExtension3=50, # extend the reads by 50bp
                     nthreads=3         # multithreading (to speed up)
                    )</code></pre>
<pre><code>
        ==========     _____ _    _ ____  _____  ______          _____  
        =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ 
          =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |
            ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |
              ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |
        ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/
       Rsubread 2.8.2

//========================== featureCounts setting ===========================\\
||                                                                            ||
||             Input files : 5 BAM files                                      ||
||                                                                            ||
||                           Chip_GR_control_s2.bam                           ||
||                           Chip_GR_control_s3.bam                           ||
||                           Chip_GR_12hDex_s1.bam                            ||
||                           Chip_GR_12hDex_s2.bam                            ||
||                           Chip_GR_12hDex_s3.bam                            ||
||                                                                            ||
||              Paired-end : no                                               ||
||        Count read pairs : no                                               ||
||              Annotation : R data.frame                                     ||
||      Dir for temp files : .                                                ||
||                 Threads : 3                                                ||
||                   Level : meta-feature level                               ||
||      Multimapping reads : counted                                          ||
|| Multi-overlapping reads : not counted                                      ||
||   Min overlapping bases : 1                                                ||
||          Read extension : 0 on 5&#39; and 50 on 3&#39; ends                        ||
||                                                                            ||
\\============================================================================//

//================================= Running ==================================\\
||                                                                            ||
|| Load annotation file .Rsubread_UserProvidedAnnotation_pid46018 ...         ||
||    Features : 309                                                          ||
||    Meta-features : 309                                                     ||
||    Chromosomes/contigs : 23                                                ||
||                                                                            ||
|| Process BAM file Chip_GR_control_s2.bam...                                 ||
||    Single-end reads are included.                                          ||
||    Total alignments : 11918412                                             ||
||    Successfully assigned alignments : 872 (0.0%)                           ||
||    Running time : 0.20 minutes                                             ||
||                                                                            ||
|| Process BAM file Chip_GR_control_s3.bam...                                 ||
||    Single-end reads are included.                                          ||
||    Total alignments : 13168495                                             ||
||    Successfully assigned alignments : 1062 (0.0%)                          ||
||    Running time : 0.13 minutes                                             ||
||                                                                            ||
|| Process BAM file Chip_GR_12hDex_s1.bam...                                  ||
||    Single-end reads are included.                                          ||
||    Total alignments : 16653674                                             ||
||    Successfully assigned alignments : 11644 (0.1%)                         ||
||    Running time : 0.19 minutes                                             ||
||                                                                            ||
|| Process BAM file Chip_GR_12hDex_s2.bam...                                  ||
||    Single-end reads are included.                                          ||
||    Total alignments : 14695293                                             ||
||    Successfully assigned alignments : 46289 (0.3%)                         ||
||    Running time : 0.21 minutes                                             ||
||                                                                            ||
|| Process BAM file Chip_GR_12hDex_s3.bam...                                  ||
||    Single-end reads are included.                                          ||
||    Total alignments : 14613584                                             ||
||    Successfully assigned alignments : 51881 (0.4%)                         ||
||    Running time : 0.29 minutes                                             ||
||                                                                            ||
|| Write the final count table.                                               ||
|| Write the read assignment summary.                                         ||
||                                                                            ||
\\============================================================================//</code></pre>
<p>We’re going to package this data into a SummarizedExperiment (SE) object:</p>
<pre class="r"><code>se &lt;- SummarizedExperiment(assays=list(counts=fc$counts), rowRanges=merged_peaks)
colnames(se) &lt;- c(&quot;C2&quot;,&quot;C3&quot;,&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;)
# we keep the counts in the input, but not as a normal sample:
# rowData(se)$input.count &lt;- assay(se)[,&quot;input.bam&quot;]
# se &lt;- se[,colnames(se)!=&quot;input.bam&quot;]
# we assign the conditions:
# se$condition  equivalent to  colData(se)$condition
se$condition &lt;- c(&quot;CTRL&quot;,&quot;12hDEX&quot;)[1+grepl(&quot;T&quot;,colnames(se))]
# is equivalent to:
# se$condition &lt;- c(&quot;CTRL&quot;,&quot;CTRL&quot;,&quot;CTRL&quot;,&quot;MUT&quot;,&quot;MUT&quot;,&quot;MUT&quot;,&quot;MUT&quot;)
se</code></pre>
<pre><code>class: RangedSummarizedExperiment 
dim: 309 5 
metadata(0):
assays(1): counts
rownames(309): chr1:8004944-8005419 chr1:8182234-8182649 ...
  chrX:65772821-65773162 chrX:72620618-72620828
rowData names(0):
colnames(5): C2 C3 T1 T2 T3
colData names(1): condition</code></pre>
</div>
<div id="normalization" class="section level2">
<h2>Normalization</h2>
<p>Let’s have a look at the counts:</p>
<pre class="r"><code>head(assay(se))</code></pre>
<pre><code>                       C2 C3 T1  T2  T3
chr1:8004944-8005419    1  5 38 188 175
chr1:8182234-8182649    1  5 22 152 148
chr1:8212017-8212287    0  4 25  92 119
chr1:8259072-8259915    7  5 98 339 456
chr1:9436312-9436621    0  2 42 162 180
chr1:10388302-10388545  2  2 19  89  93</code></pre>
<pre class="r"><code>head(assays(se)$counts)</code></pre>
<pre><code>                       C2 C3 T1  T2  T3
chr1:8004944-8005419    1  5 38 188 175
chr1:8182234-8182649    1  5 22 152 148
chr1:8212017-8212287    0  4 25  92 119
chr1:8259072-8259915    7  5 98 339 456
chr1:9436312-9436621    0  2 42 162 180
chr1:10388302-10388545  2  2 19  89  93</code></pre>
<p>Let’s plot the counts for the first 500 regions:</p>
<pre class="r"><code>sechm(se, head(row.names(se),500), assayName=&quot;counts&quot;, do.scale=TRUE, top_annotation = c(&quot;condition&quot;))</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Or another way to look at it:</p>
<pre class="r"><code>plot(assay(se)[,1], assay(se)[,2], log=&quot;xy&quot;)</code></pre>
<pre><code>Warning in xy.coords(x, y, xlabel, ylabel, log): 22 x values &lt;= 0 omitted from
logarithmic plot</code></pre>
<pre><code>Warning in xy.coords(x, y, xlabel, ylabel, log): 15 y values &lt;= 0 omitted from
logarithmic plot</code></pre>
<pre class="r"><code>abline(a=0, b=1)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The usual way we represent this kind of relation is through a M-A plot:</p>
<pre class="r"><code>edgeR::maPlot(assay(se)[,1], assay(se)[,2], lowess=TRUE, ylab=&quot;M (log2 foldchange)&quot;, 
              xlab=&quot;A (mean log-count)&quot;); abline(h=0, lty=&quot;dashed&quot;)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The standard approach would be to use the TMM normalization. HOWEVER, because some of our samples appear to have no peaks at all normalisation might distort the counts. We want to replace the library size column by the read counts.</p>
<pre class="r"><code>read_counts &lt;- sapply(bamfiles, function(bamfile){
  aln &lt;- scanBam(bamfile, param=ScanBamParam(what=&quot;qname&quot;))
  length(aln[[1]]$qname)  
})

dds &lt;- DGEList(assay(se), group=se$condition)

# We skip this due to the different peak per sample
# dds &lt;- calcNormFactors(dds) 
dds$libary.size &lt;- read_counts
assays(se)$tmm.logcpm &lt;- log1p(cpm(dds))
sechm(se, head(row.names(se),500), assayName=&quot;tmm.logcpm&quot;, do.scale = TRUE)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This made the 3 controls very consistent with each other, suggesting that the normalization worked.</p>
<p>From this, it looks like some regions have increased YY1 occupancy in the mutant, while others have decreased occupancy, and very little remains unaltered. We can also run the differential analysis and look at the results :</p>
<pre class="r"><code>dds &lt;- estimateDisp(dds)</code></pre>
<pre><code>Using classic mode.</code></pre>
<pre class="r"><code>et &lt;- as.data.frame(topTags(exactTest(dds),n=Inf))
head(et)</code></pre>
<pre><code>                              logFC   logCPM       PValue         FDR
chr5:172869049-172869468  -3.515602 12.49215 1.857146e-05 0.005738581
chr12:6376197-6376471     -8.782395 11.26510 1.090899e-04 0.013341944
chr6:142376184-142376447  -8.697574 11.18713 1.295334e-04 0.013341944
chr11:123122426-123122655 -8.478058 10.96856 5.579489e-04 0.043101555
chrX:65772821-65773162    -1.669320 12.24520 1.768459e-03 0.109290759
chr20:31610720-31611048   -2.668320 11.68709 4.104379e-03 0.185418251</code></pre>
<p>We can represent the results of the analysis as a so-called ‘volcano plot’:</p>
<pre class="r"><code>ggplot(et, aes(logFC, -log10(FDR))) + geom_point() + 
  geom_hline(yintercept=-log10(0.05), linetype=&quot;dashed&quot;)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>(You can use the <code>EnhancedVolcano</code> package to make nicer-looking plots)</p>
<p>At a first glance, there seems to be roughly as many regions that increase in binding as regions that decrease in binding. Does this sound likely to you?</p>
<p>Let’s look again at an MA plot between a ctrl and a mutant:</p>
<pre class="r"><code>edgeR::maPlot(assay(se)[,2], assay(se)[,5], lowess=TRUE); abline(h=0, lty=&quot;dashed&quot;)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="normalization-based-on-top-peaks" class="section level2">
<h2>Normalization based on top peaks</h2>
<pre class="r"><code># we first calculate the peaks&#39; median enrichment against the input
# rowData(se)$medianEnr &lt;- matrixStats::rowMedians(assays(se)$tmm.logcpm - log1p(rowData(se)$input.count))
# we then select the top peaks based on that
# w &lt;- head(order(rowData(se)$medianEnr, decreasing=TRUE), 200)
# we then estimate TMM normalization factors based on that:
# dds2 &lt;- calcNormFactors(dds[w,,keep.lib.sizes=TRUE])
# dds2$samples</code></pre>
<p>Notice that the <code>norm.factors</code> column is different than the one we previous had. We can now inject them into the full object:</p>
<pre class="r"><code># dds$samples$norm.factors &lt;- dds2$samples$norm.factors</code></pre>
<p>We compute again the normalized logcpm and plot our matrix:</p>
<pre class="r"><code>assays(se)$logcpm &lt;- log1p(cpm(dds))
sechm(se, head(row.names(se),500), assayName=&quot;logcpm&quot;, do.scale = TRUE)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can now do the differential analysis using these factors</p>
<pre class="r"><code>dds &lt;- estimateDisp(dds)</code></pre>
<pre><code>Using classic mode.</code></pre>
<pre class="r"><code>et &lt;- as.data.frame(topTags(exactTest(dds),Inf))
head(et)</code></pre>
<pre><code>                              logFC   logCPM       PValue         FDR
chr5:172869049-172869468  -3.515602 12.49215 1.857146e-05 0.005738581
chr12:6376197-6376471     -8.782395 11.26510 1.090899e-04 0.013341944
chr6:142376184-142376447  -8.697574 11.18713 1.295334e-04 0.013341944
chr11:123122426-123122655 -8.478058 10.96856 5.579489e-04 0.043101555
chrX:65772821-65773162    -1.669320 12.24520 1.768459e-03 0.109290759
chr20:31610720-31611048   -2.668320 11.68709 4.104379e-03 0.185418251</code></pre>
<pre class="r"><code>ggplot(et, aes(logFC, -log10(FDR))) + geom_point() + 
  geom_hline(yintercept=-log10(0.05), linetype=&quot;dashed&quot;)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>(See <code>?edgeRUsersGuide()</code> for how to apply edgeR to more complex experimental designs)</p>
<p>We could also visualize the top results in this way:</p>
<pre class="r"><code>sechm(se, head(row.names(et),50), assayName=&quot;logcpm&quot;, do.scale=TRUE, row_title=&quot;Top 50 differentially-bound regions&quot;)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="applying-the-normalization-for-visualization" class="section level2">
<h2>Applying the normalization for visualization</h2>
<p>For comparison, we first plot using background normalization:</p>
<pre class="r"><code>tracks &lt;- list(&quot;C2&quot;=&quot;../data/single_replicate_peaks/control/Chip_GR_control_s2.bigWig&quot;,
               &quot;C3&quot;=&quot;../data/single_replicate_peaks/control/Chip_GR_control_s3.bigWig&quot;,
               &quot;t12_1&quot;=&quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1.bigWig&quot;,
               &quot;t12_2&quot;=&quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2.bigWig&quot;,
               &quot;t12_3&quot;=&quot;../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s3.bigWig&quot;)
m &lt;- signal2Matrix(tracks, merged_peaks)</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/control/Chip_GR_control_s2.bigWig</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/control/Chip_GR_control_s3.bigWig</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s1.bigWig</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s2.bigWig</code></pre>
<pre><code>Reading ../data/single_replicate_peaks/dex12h/Chip_GR_12hDex_s3.bigWig</code></pre>
<pre class="r"><code>plotEnrichedHeatmaps(m)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-18-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>We now apply the normalization factors we calculated:</p>
<pre class="r"><code># we first make sure that the order of the samples matches:
cbind(colnames(dds), names(m))</code></pre>
<pre><code>     [,1] [,2]   
[1,] &quot;C2&quot; &quot;C2&quot;   
[2,] &quot;C3&quot; &quot;C3&quot;   
[3,] &quot;T1&quot; &quot;t12_1&quot;
[4,] &quot;T2&quot; &quot;t12_2&quot;
[5,] &quot;T3&quot; &quot;t12_3&quot;</code></pre>
<pre class="r"><code>m2 &lt;- rescaleSignalMatrices(m, scaleFactors=1/dds$samples$norm.factors)
plotEnrichedHeatmaps(m2)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-19-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Let’s look at it using just the average signal plots:</p>
<pre class="r"><code>d1 &lt;- meltSignals(m)
d1$condition &lt;- gsub(&quot;[0-9]&quot;,&quot;&quot;,d1$sample)
d1$condition &lt;- ifelse(d1$condition==&quot;C&quot;, &quot;Control&quot;, &quot;12hDEX&quot;)
p1 &lt;- ggplot(d1, aes(position, mean, group=sample, colour=condition)) + geom_line() + ggtitle(&quot;Coverage normalization&quot;)
d2 &lt;- meltSignals(m2)
d2$condition &lt;- gsub(&quot;[0-9]&quot;,&quot;&quot;,d2$sample)
d2$condition &lt;- ifelse(d2$condition==&quot;C&quot;, &quot;Control&quot;, &quot;12hDEX&quot;)
p2 &lt;- ggplot(d2, aes(position, mean, group=sample, colour=condition)) + geom_line() + ggtitle(&quot;Top peaks normalization&quot;)
ggpubr::ggarrange(p1, p2, ncol=2)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-20-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="using-background-normalization" class="section level2">
<h2>Using background normalization</h2>
<pre class="r"><code># here I have to specify the chromosomes to use because the bigwig files contain only those
nf &lt;- bwNormFactors(tracks)
m3 &lt;- rescaleSignalMatrices(m, scaleFactors=1/nf)
d3 &lt;- meltSignals(m3)
d3$condition &lt;- gsub(&quot;[0-9]&quot;,&quot;&quot;,d3$sample)
d3$condition &lt;- ifelse(d3$condition==&quot;C&quot;, &quot;Control&quot;, &quot;12hDEX&quot;)
p3 &lt;- ggplot(d3, aes(position, mean, group=sample, colour=condition)) + geom_line() + ggtitle(&quot;Background normalization&quot;)
ggpubr::ggarrange(p2, p3, ncol=2)</code></pre>
<p><img src="figure/Q9-Differential_binding.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggplot2_3.3.6               rtracklayer_1.54.0         
 [3] Rsamtools_2.10.0            Biostrings_2.62.0          
 [5] XVector_0.34.0              edgeR_3.36.0               
 [7] limma_3.50.3                sechm_1.2.0                
 [9] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[11] MatrixGenerics_1.6.0        matrixStats_0.62.0         
[13] Rsubread_2.8.2              epiwraps_0.99.42           
[15] EnrichedHeatmap_1.24.0      ComplexHeatmap_2.10.0      
[17] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[19] IRanges_2.28.0              S4Vectors_0.32.4           
[21] BiocGenerics_0.40.0         dplyr_1.0.9                

loaded via a namespace (and not attached):
  [1] backports_1.4.1          circlize_0.4.15          Hmisc_4.7-0             
  [4] workflowr_1.7.0          BiocFileCache_2.2.1      plyr_1.8.7              
  [7] lazyeval_0.2.2           splines_4.1.0            BiocParallel_1.28.3     
 [10] digest_0.6.29            ensembldb_2.18.4         foreach_1.5.2           
 [13] htmltools_0.5.2          magick_2.7.3             fansi_1.0.3             
 [16] magrittr_2.0.3           checkmate_2.1.0          memoise_2.0.1           
 [19] BSgenome_1.62.0          cluster_2.1.3            doParallel_1.0.17       
 [22] prettyunits_1.1.1        jpeg_0.1-9               colorspace_2.0-3        
 [25] blob_1.2.3               rappdirs_0.3.3           xfun_0.31               
 [28] crayon_1.5.1             RCurl_1.98-1.7           jsonlite_1.8.0          
 [31] survival_3.3-1           VariantAnnotation_1.40.0 iterators_1.0.14        
 [34] glue_1.6.2               registry_0.5-1           gtable_0.3.0            
 [37] zlibbioc_1.40.0          UpSetR_1.4.0             V8_4.2.0                
 [40] GetoptLong_1.0.5         DelayedArray_0.20.0      car_3.1-0               
 [43] shape_1.4.6              abind_1.4-5              scales_1.2.0            
 [46] DBI_1.1.3                rstatix_0.7.0            randomcoloR_1.1.0.1     
 [49] Rcpp_1.0.8.3             viridisLite_0.4.0        progress_1.2.2          
 [52] htmlTable_2.4.0          clue_0.3-61              foreign_0.8-82          
 [55] bit_4.0.4                Formula_1.2-4            htmlwidgets_1.5.4       
 [58] httr_1.4.3               RColorBrewer_1.1-3       ellipsis_0.3.2          
 [61] farver_2.1.1             pkgconfig_2.0.3          XML_3.99-0.10           
 [64] Gviz_1.38.4              nnet_7.3-17              sass_0.4.1              
 [67] dbplyr_2.2.1             deldir_1.0-6             locfit_1.5-9.5          
 [70] utf8_1.2.2               labeling_0.4.2           tidyselect_1.1.2        
 [73] rlang_1.0.3              later_1.3.0              AnnotationDbi_1.56.2    
 [76] munsell_0.5.0            tools_4.1.0              cachem_1.0.6            
 [79] cli_3.3.0                generics_0.1.3           RSQLite_2.2.14          
 [82] broom_1.0.0              evaluate_0.15            stringr_1.4.0           
 [85] fastmap_1.1.0            yaml_2.3.5               knitr_1.39              
 [88] bit64_4.0.5              fs_1.5.2                 purrr_0.3.4             
 [91] KEGGREST_1.34.0          AnnotationFilter_1.18.0  xml2_1.3.3              
 [94] biomaRt_2.50.3           compiler_4.1.0           rstudioapi_0.13         
 [97] filelock_1.0.2           curl_4.3.2               png_0.1-7               
[100] ggsignif_0.6.3           tibble_3.1.7             bslib_0.3.1             
[103] stringi_1.7.6            highr_0.9                GenomicFeatures_1.46.5  
[106] GenomicFiles_1.30.0      lattice_0.20-45          ProtGenerics_1.26.0     
[109] Matrix_1.4-1             vctrs_0.4.1              pillar_1.7.0            
[112] lifecycle_1.0.1          jquerylib_0.1.4          GlobalOptions_0.1.2     
[115] cowplot_1.1.1            data.table_1.14.2        bitops_1.0-7            
[118] seriation_1.3.5          httpuv_1.6.5             R6_2.5.1                
[121] BiocIO_1.4.0             latticeExtra_0.6-30      TSP_1.2-0               
[124] promises_1.2.0.1         gridExtra_2.3            codetools_0.2-18        
[127] dichromat_2.0-0.1        assertthat_0.2.1         rprojroot_2.0.3         
[130] rjson_0.2.21             withr_2.5.0              GenomicAlignments_1.30.0
[133] GenomeInfoDbData_1.2.7   parallel_4.1.0           hms_1.1.1               
[136] rpart_4.1.16             tidyr_1.2.0              rmarkdown_2.14          
[139] carData_3.0-5            Rtsne_0.16               ggpubr_0.4.0            
[142] git2r_0.30.1             biovizBase_1.42.0        base64enc_0.1-3         
[145] interp_1.1-2             restfulr_0.0.15         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
