---
title: "Analysis"
author: "Dominique Paul"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

suppressPackageStartupMessages({
  library(ggplot2)
  library(biomaRt)
  library(dplyr)
  library(SummarizedExperiment)
  library(edgeR)
  library(SEtools)
  library(grid)
  library(VennDiagram)
  library(hrbrthemes)
  library(openxlsx)
  library(stringdist)
  library(rtracklayer)
  library(epiwraps)
  library(kableExtra)
  library(EnrichedHeatmap)
  library(AnnotationHub)
})

source("../code/helper_functions.R")

extrafont::loadfonts()
```

```{r save or load the environment}
# save.image(file='myEnvironment.RData')
# load('myEnvironment.RData')
# quit(save="no")
```

# Overview

The overall goal is to (1) understand the effect of activated glucocorticoid receptors on gene regulation and (2) relate the epigenetic data from the Reddy lab with our expression data. To this end Mahshid and developed seven initial questions which define the primary scope of the analysis. In a discussion on these results further questions arose which are answered further below. As some of these questions/answers relate to results obtained in other work by Katharina, Mahshid and Pierre-Luc the flow of questions may not be fully coherent in themselves.

# Question 1: Similarity of our RNA data with the Reddy data

**Q: Which gene expressions have similar patterns of up-regulation (control vs. Dex) in our data compared to the Reddy data?**

In our experiments we compared samples from a control experiment with samples that were treated for 18 hours with DEX. Besides a control, the Reddy lab performed a similar experiment where they treated cells with DEX for 12 hours. This is the closest comparison to our data.

As a first step we hence want to understand whether the results are similar/comparable.


## Comparison via Venn diagram

```{r extract r objects for Q1}
q1_data <- readRDS(file="../data/EOS_files/Q1_EOS.rds")

n_genes_plot <- dim(q1_data[["logFC_plot_data"]])[1]
n_genes_union <- length(q1_data[["union_DE_genes"]])
```

We first inspect a Venn diagram to understand the overlap of differentially expressed genes from both datasets. **We see that a large part of the genes do overlap, while there are also many genes that are specific to the respective analyses.** However, this is only a binary comparison and so a we next take a look at how the exact log-fold changes compare.

![](assets/1_venn_diagram.pdf)

## Comparison via log-fold changes

Second, we compare the log fold changes for the union of all genes identified as significantly differentially expressed. The union of all DE genes counts `r n_genes_union` genes of which `r n_genes_plot` genes (`r round(100*n_genes_plot/n_genes_union,2)`%) were also in the intersection of both datasets and could be plotted.

**We see that the results of our differential expression (DE) analysis from the 18 DEX treatment compare well with the 12h DEX treatment DE results of the Reddy lab.**

![](assets/2_logFC_Mahshid_Reddy.pdf)

# Question 2: Binding of GR-GC complex to DE genes from our analysis {.tabset}

**Q: For the genes that our RNA-seq data shows to be DE after treatment with DEX as well as the inhibitors and control: do we see any direct TF binding to these sites? For which of these is there a difference compared to the control ChIP-seq experiment?**


**Take-away:** Genes which are differentially expressed at a significant level are associated with GR peaks close to them. Genes with GR binding close to them are also associated with lower p-values in the DE analysis. A higher number of GR peaks close to the TSS of a gene is associated with a higher absolute log-fold change in gene expression. Genes where GR binds close to or directly at the TSS are associated with the higher absolute log-fold changes.

A table with the DE results from our 18h DEX treatment which is augmented by information on peaks identified nearby can be found in the downloads section.

An analysis on peaks which are known to interact with distant genes (long-range interactions) can be found further below in this file.

## Distribution of logFC and p.value

**What does this plot show?**

We select all genes that are differentially expressed - both for Mahshid's data and for the DE analysis on the Reddy data. We then split the data into two groups based on whether the Reddy data showed that the glucocorticoid receptor was found to bind near the gene or not. We plot the distributions of (a) the log fold changes in the DE analysis and (b) the p-value of the DE analysis. 


![](assets/Q2_3_mahshid_reddy.pdf)

## Boxplots by GR peaks number

![](assets/Q2_4_boxplots.pdf)

## Boxplots by type of closest GR peak

_Note:_ NA means that there was no peak for which the gene's TSS was the TSS closest to the peak.

![](assets/Q2_6_boxplots_class.pdf)

## logFC vs. distance to closest peak

![](assets/Q2_5_TSS_distances.pdf)

# Question 3: Inhibition unique to protac compared to Cort and MIF {.tabset}

**Q: Are there any genes that treated by the Protac before/after DEX that could be reversed but which could not by the inhibitors or vice versa?**

We have performed experiments where we first exposed cells to either (1) DEX (2) inhibitors or (3) KH PROTAC for a substantial time and then added inhibitors or the KH PROTAC in case (1) or DEX in cases (2) and (3). We would like to see which genes are normally activated by DEX but where the effect could be blocked by the PROTAC, but NOT the inhibitors.

The plot for the 2h>16h treatment has been split into (1) genes with a positive log-fold change and (2) genes with a negative log-fold change to prevent the plots from becoming even more crowded.

A file with the list of genes can be found in the downloads section.

## 2h > 16h

![](assets/Q3_1_reversal216.pdf)

## 16h > 2h (logFC > 0)

![](assets/Q3_2_reversal162_p1.pdf)

## 16h > 2h (logFC < 0)

![](assets/Q3_3_reversal162_p2.pdf)



# Question 4: Does the GR bind close to any of the 13 DE genes from the 18h KH-PROTAC treatment? {.tabset}

Cells that were exposed to KH-103 for 18h showed 13 genes to be differentially expressed at an <0.05 FDR level. We want to understand whether GR binds close to these genes in a case where such cells are treated with DEX.

The Reddy lab has collected data on where the glucocorticoid receptor (NR3C1) binds to after treatment with dexamethasone for 12h. 

**Take-away:** We find that the GR is found to bind close to five out of the 13 differentially expressed genes from our experiment. Interestingly, we also observe that some GR peaks are stronger after 2 hours of DEX treatment than 12 hours of continuous DEX treatment.

## Table view

```{r q4}
kh.protac <- read.xlsx("../docs/downloads/Q4_KH_DE_GR_bound.xlsx")

# only keep the significant genes
kh.protac.sig <- kh.protac[kh.protac$FDR<0.05,]
kh.protac.sig[,c("Gene", "logFC", "FDR", "chip_peaks", "gc_bound")] %>%
  kableExtra::kable() %>%
  kableExtra::kable_styling()
```

![](assets/Q4_1_kh_gr_binding.pdf)
## p-value vs. logFC

## Signal tracks for peaks that could be matched with the 13 genes

We plot peaks for which the closest TSS was found to be one of the 13 genes. The tracks of the peaks shown are _centered at these peaks_ and _not_ at the TSS of the respective gene.

![](assets/Q4_2_kh_peak_tracks.pdf)

## Tracks around the all TSSs of the 13 genes

Here we first obtain all known transcripts of the 13 genes and plot the signals _centered around the respective TSS sites_. As a gene can have multiple TSSs this plot is a bit crowded.

![](assets/Q4_3_kh_all_txs.pdf)

## Tracks around main TSS for each of the 13 genes

To simplify the previous plot (all TSSs) we just choose a single TSS per gene (the first one in the list) and plot the signal around this TSS.

![](assets/Q4_4_kh_unique_gene_tss.pdf)

# Question 5: Binding to selected genes

**Q: Does GR bind to BRCA1, Hsd11b1 and CH25H?**

These are specific genes requested to be examined by Mahshid. The background of the biological question is not known to me.

**Take-away:** We see that there is one peak in the ChIP data close to the BRCA1 gene and two peaks close to the HSD11B1 gene. There is no peak that is closest to the CH25H gene. However, the CD25H was also not found in the RNA seq data. 

```{r -1}
res_int <- read.xlsx("../docs/downloads/Q2_GR_bound_DE_genes.xlsx")
plt_data <- res_int[res_int$Gene %in% c("BRCA1", "HSD11B1", "CH25H"),c("Gene", "logFC", "FDR", "chip_peaks", "gc_bound")]
plt_data %>%
  ggplot() +
  geom_point(aes(logFC, -log10(FDR), colour=gc_bound, size=chip_peaks)) +
  labs(title="GR binding activity for diff. expressed PROTAC genes") +
  ggrepel::geom_text_repel(data=plt_data, aes(logFC, -log10(FDR), label=Gene, size=2)) +
  theme_ipsum_rc()
```

 CH25H was not found. Is it in the RNA-seq experiment?

```{r -2}
se <- readRDS("../data/EOS_files/Q1_SummExp.rds")
"ENSG00000138135" %in% rowData(se)$gene_id
```

No, it is not in the RNA-seq experiment.

We also check whether the gene is present in the table with the  peaks. If the gene were to be missing in our DE data then a given peak would not show up in the results above, as it could not have been merged onto the DE results.

We search for the closest match to be sure that its not due to capitalisation or a small typo

```{r -3}
# We also have to check the other way around by looking at all the peaks.
# If it was in the peaks then it could not be merged onto the RNA data
peakfreq <- readRDS(file="../data/EOS_files/Q2_gene_peaks.rds")
peakfreq <- peakfreq[2:dim(peakfreq)[1],] # first entry is empty

# we search for the closest match to be sure that its not due to capitalisation or a small typo
peakfreq$Gene[amatch("CH25H", peakfreq$Gene, maxDist = Inf)]

```

**Result:** there is no match for CH25h

 Lastly, we show the information we have for the two genes found in the data:
 
```{r table with kable}
plt_data %>%
  kableExtra::kable() %>%
  kableExtra::kable_styling()
```




# Question 6: Common binding sites of JUNB and GR 

**Q: What are common binding sites between JUNB and GR?** -> Generally not a priority though

**Take-aways:**

- Interestingly there are more peaks after 2h treatment with DEX than 12h of treatment with DEX
- Most of the 12h GR peaks are shared with the JUNB peaks, only a small fraction is unique to the 12h GR ChIP seq data
- The 2h GR data has more peaks that are unique and not shared with the 12h GR or 12h JUNB data

## Unfiltered {.tabset}

```{r q6, results="asis"}
# Load GR
peaks_gr_12h <- import.bed15("../data/reddy_bed_DEX12h_r123.bed", format="narrowPeak")
peaks_gr_2h <- import.bed15("../data/reddy_bed_DEX2h_r123.bed.gz", format="narrowPeak")
# load JUNB
peaks_junb_2h <- import.bed15("../data/reddy_bed_JUNB_DEX2h_r23.bed.gz", format="narrowPeak")
peaks_junb_12h <- import.bed15("../data/reddy_bed_JUNB_DEX12h_r123.bed", format="narrowPeak")

# Show tile diagram
list_of_regions <- list(GR_2h=peaks_gr_2h, GR_12h=peaks_gr_12h, JUNB_12h=peaks_junb_12h) # Does not work with only 
cat("\n\n### Overlap \n\n")
regionOverlaps(list_of_regions)

# Show upset plot
cat("\n\n### Upset plot \n\n")
regionUpset(list_of_regions)
```

## What if we filter the peaks by their score? {.tabset}

**Take-away:** There is no significant change in the results/plots if we filter for a minimum peak score of 900 (max value is 1000).

```{r q6 filtered, results="asis"}
# Does it make a difference if we only choose peaks with high scores?
## What is the range of scores
# qplot(peaks_gr_2h$score, geom="histogram")

# Show tile diagram
list_of_regions_v2 <- list(GR_2h=peaks_gr_2h[peaks_gr_2h$score>900], 
                           GR_12h=peaks_gr_12h[peaks_gr_12h$score>900], 
                           JUNB_12h=peaks_junb_12h[peaks_junb_12h$score>900]) # Does not work with only two 
cat("\n\n### Overlap \n\n")
regionOverlaps(list_of_regions_v2)

# Show upset plot
cat("\n\n### Upset plot \n\n")
regionUpset(list_of_regions_v2)
```

## Which genes have a GR peak AND a JUNB peak within 2500bp distance to the TSS? {.tabset}

Annotate GR 2h and 12h and JUNB 2h

```{r gr peak and junb, results="asis"}
ah <- AnnotationHub()
ensdb <- ah[["AH98047"]] # AH89211 (mouse), AH98047 (human), AH95846 (rat)
gr_peaks_annotated_2h <- annotateRegions(peaks_gr_2h, ensdb)
gr_peaks_annotated_12h <- readRDS("../data/EOS_files/Q2_annotated_peaks.rds")
JUNB_peaks_annotated_2h <- annotateRegions(peaks_junb_2h, ensdb)
JUNB_peaks_annotated_12h <- annotateRegions(peaks_junb_12h, ensdb)

### filter for 2500 bp distance to gene TSS
# GR 2h
gr_peaks_annotated_2h$nearestTSS.gene_name.2500r <- gr_peaks_annotated_2h$nearestTSS.gene_name
gr_peaks_annotated_2h[abs(gr_peaks_annotated_2h$distance2nearestTSS) >= 2500, ]$nearestTSS.gene_name.2500r <- ""
# JUNB 2h 
JUNB_peaks_annotated_2h$nearestTSS.gene_name.2500r <- JUNB_peaks_annotated_2h$nearestTSS.gene_name
JUNB_peaks_annotated_2h[abs(JUNB_peaks_annotated_2h$distance2nearestTSS) >= 2500, ]$nearestTSS.gene_name.2500r <- ""
# JUNB 12h 
JUNB_peaks_annotated_12h$nearestTSS.gene_name.2500r <- JUNB_peaks_annotated_12h$nearestTSS.gene_name
JUNB_peaks_annotated_12h[abs(JUNB_peaks_annotated_12h$distance2nearestTSS) >= 2500, ]$nearestTSS.gene_name.2500r <- ""

# subset
gr_2h_subset <- gr_peaks_annotated_2h[gr_peaks_annotated_2h$nearestTSS.gene_name.2500r != "",]
gr_12h_subset <- gr_peaks_annotated_12h[gr_peaks_annotated_12h$nearestTSS.gene_name.2500r != "",]
junb_2h_subset <- JUNB_peaks_annotated_2h[JUNB_peaks_annotated_2h$nearestTSS.gene_name.2500r != "",]
junb_12h_subset <- JUNB_peaks_annotated_12h[JUNB_peaks_annotated_12h$nearestTSS.gene_name.2500r != "",]

list_of_regions_v3 <- list(GR_2h=gr_2h_subset, 
                           GR_12h=gr_12h_subset,
                           JUNB_2h=junb_2h_subset,
                           JUNB_12h=junb_12h_subset)

cat("\n\n### Overlap \n\n")
regionOverlaps(list_of_regions_v3)

# Show upset plot
cat("\n\n### Upset plot \n\n")
regionUpset(list_of_regions_v3)
```

#### Visualise the genes that have both a JUNB and GR peak within 2500bp of their TSS

```{r gr and junb peak}
gr_junb_2h_intersection_genes <- intersect(gr_2h_subset$nearestTSS.gene_name.2500r, junb_2h_subset$nearestTSS.gene_name.2500r)

# cat(gr_junb_2h_intersection_genes)

length(gr_junb_2h_intersection_genes)
```

There is an overlap of `r length(gr_junb_2h_intersection_genes)` genes.


# Question 7: PXR Sequences

**Q: Does GC-GR bind to PXR sequences?**

Other groups have observed that a GR inhibitor activates CYP genes. CYP genes are target genes of PXR TF that have PXR recognition sites. the hypothesis here is that GR might also bind to PXR sites directly or probably is regulating CYP genes via interaction with PXR TF on these genes. For example, based on our RNA-seq dex triggers many CYP genes and the inhibitors too, but protac doesn't.

The PXR motif looks like this (from Hocomoco):

![](assets/Q7_1_motif.pdf)


```{r grpaks moi}
gr_peaks <- readRDS("../data/EOS_files/Q2_annotated_peaks.rds")
moi <- readRDS(file="../data/EOS_files/pxr_peaks.rds")

# cat(paste0("We have ", length(gr_peaks), " GR peaks, of which ", length(moi), " (", round(length(moi)/length(gr_peaks),4)*100, "%) contain a PXR motif"))
```

### We have `r length(gr_peaks)` GR peaks, of which `r length(moi)` (`r round(length(moi)/length(gr_peaks),4)*100`%) contain a PXR motif

We can also look at the signals around the PXR motifs found in the accessible chromatin (identified using the 12h ATAC seq data from the Reddy lab)

![](assets/Q7_2_signal.pdf)

# Q8: How do our signals compare to our differential expression results?

In the following we plot the signal for a subset of genes, IF these genes are associated with a peak the in the GR data for the 12h DEX ChIP data.

### Signals and DE results for all genes with associated peaks

We first plot the signal tracks for all genes for the sake of completeness. We plot around the TSS of the gene, not around the centre of the peak. Reminder: The peaks are labelled with the gene name the TSS of which is closest to the peak.

```{r q8a}
# load summarised experiment
se <- readRDS(file="../data/results/SE.processed.rds")
# removes the row without name
se <- se[2:dim(se)[1]]
se <- se[,order(se$condType, se$condition, se$batch)]
# change order of conditions
se$condition <- factor(se$condition,
                       c("18h untreated", "18h DMSO", "18h MIF", "18h Cort113", "18h KH-103", 
                  "2h DEX >\n16h DEX+MIF", "2h DEX >\n16h DEX+Cort113", 
                  "2h DEX >\n16h DEX+KH-103", "2h DEX >\n16h DEX+DMSO",
                  "16h MIF  >\n2h DEX+MIF", "16h Cort113 >\n2h DEX+Cort113", 
                  "16h KH-103 >\n2h DEX+KH-103", "16h DMSO >\n2h DEX+DMSO" ))
)
res_int_all <- readRDS(file="../data/results/DE_our_data.rds")
```


```{r}
#' plotSigAndRNA
#' 
#' Plot joint chromatin signals and RNA heatmap
#'
#' @param x Either a list of genes (requires `ensdb` to be given) or a GRanges
#'   with regions of interest, and with the `gene_name` mcol.
#' @param tracks A (named) vector of paths to bigwig files.
#' @param SE A SummarizedExperiment objects with gene names as rows.
#' @param assayName The assay to use in `SE`
#' @param ensdb An `EnsDb` object, ignored if `x` is already a GRanges
#' @param k The optional number of clusters
#' @param extend How much to extend on each side of (the center of) the given regions
#' @param trim Trimming passed to `plotEnrichedHeatmaps`
#' @param seed Seed (for clustering)
#' @param forceSeqlevelsStyle Optional seqLevelStyle to which to force the GRanges
#' @param rna_top_annotation Top annotation for the `SE` object (see `sechm`)
#' @param gaps_at Passed to `sechm`
#' @param widths Respective widths, in centimeters, of the signal and RNA heatmaps
#' @param nf Optional normalization factors for the tracks
#' @param recenterOnPeaks An optional GRanges of peaks on which to recenter when
#'   plotting promoters
#' @param breaks Passed to `sechm`
#' @param mergeDist Distance below which to merge TSS
#' @param retList Logical, whether to return a list of heatmaps instead of hte merge
#' @param ... Passed to `plotEnrichedHeatmaps`
#'
#' @return A HeatmapList
#' @export
plotSigAndRNA <- function(x, tracks, SE, assayName="scaledLFC", ensdb=NULL, k=NULL, 
                          extend=2000, trim=0.99, seed=123, forceSeqlevelsStyle=NULL,
                          rna_top_annotation=c(), gaps_at=NULL, widths=NULL, nf=NULL,
                          recenterOnPeaks=NULL, breaks=NULL, mergeDist=500, use_raster=TRUE,
                          retList=FALSE, ...){
  if(is.character(x)){
    d <- as.integer(floor(mergeDist/2))
    stopifnot(is(ensdb, "EnsDb"))
    x <- intersect(x, row.names(SE))
    all.txs <-  transcripts(ensdb, columns=c("gene_name"))
    x <- promoters(all.txs[which(all.txs$gene_name %in% x)], upstream=d, downstream=d)
    x <- x[!duplicated(granges(x))]
    x <- reduce(split(x,x$gene_name))
    gn <- rep(names(x),lengths(x))
    x <- unlist(x)
    x$gene_name <- gn
    x <- keepSeqlevels(x, seqlevelsInUse(x), pruning.mode = "coarse")
    SE <- SE[as.character(x$gene_name),]
    row.names(SE) <- names(x) <- make.unique(x$gene_name)
    if(!is.null(recenterOnPeaks)){
      if(!is.null(forceSeqlevelsStyle) && forceSeqlevelsStyle=="chr"){
        # bad workaround for the bug of older GenomeInfoDb versions
        seqlevels(recenterOnPeaks) <- gsub("chrchr","chr",paste0("chr",seqlevels(recenterOnPeaks)))
      }
      d <- distanceToNearest(resize(x,width=1L,fix="center"), resize(recenterOnPeaks,width=1L,fix="center"))
      d2 <- rep(0L,length(x))
      d2[from(d)] <- mcols(d)$distance
      d2[which(abs(d2)>min(abs(extend)))] <- 0L
      x <- shift(x, d2+1L)
    }
    if(!is.null(forceSeqlevelsStyle) && forceSeqlevelsStyle=="chr"){ ##
      seqlevels(x) <- gsub("chrchr","chr",paste0("chr",seqlevels(x)))
    }
  }else{
    stopifnot(is(x, "GRanges") && !is.null(x$gene_name))
    x <- x[!is.na(x$gene_name)]
    if(is.null(names(x))) names(x) <- paste0("r",seq_along(x))
    SE <- SE[x$gene_name]
    row.names(SE) <- names(x)
  }
  if(!is.null(forceSeqlevelsStyle) && forceSeqlevelsStyle!="chr")
    seqlevelsStyle(x) <- forceSeqlevelsStyle
  set.seed(seed)
  mat.sig <- signal2Matrix(tracks, x, extend=extend)
  if(!is.null(nf)) mat.sig <- rescaleSignalMatrices(mat.sig, nf)
  SE <- SE[row.names(mat.sig[[1]]),]
  mat.sig <- lapply(mat.sig, FUN=function(x){
    if(length(trim)==1) trim <- sort(c(1-trim, trim))
    q <- quantile(x, trim)
    x[x>q[2]] <- q[2]
    x[x<q[1]] <- q[1]
    x
  })
  mat.rna <- assays(SE)[[assayName]]
  
  cl <- NULL
  if(!is.null(k) && k>1){
    sig.en <- matrix(unlist(lapply(mat.sig, enriched_score)), ncol=length(tracks))
    sig.en <- sechm::safescale(sig.en,byRow=TRUE)
    m <- cbind(sig.en, sechm::safescale(mat.rna,byRow=TRUE))
    cl <- kmeans(m, centers=k, nstart=3)$cluster
  }
  if(is.null(widths)){
    widths <- list(NULL,NULL)
  }else if(!is.list(widths) && is.numeric(widths)){
    widths <- lapply(widths, units="cm", FUN=unit)
  }

  h1 <- plotEnrichedHeatmaps(mat.sig, ..., row_split=cl, width=widths[[1]], use_raster=use_raster)
  h2 <- sechm::sechm(SE, row.names(SE), assayName=assayName, sortRowsOn=NULL, breaks=breaks, use_raster=use_raster,
                     cluster_rows=FALSE, width=widths[[2]], top_annotation=rna_top_annotation, 
                     gaps_at=gaps_at, column_title_gp=gpar(fontsize=10), column_title_rot=90)
  if(retList) return(list(h1,h2))
  h1 + h2
}

selectBestEnhancer <- function(et, peaks){
  et$hasPeak <- overlapsAny(et, peaks)
  et <- et[order(!et$hasPeak, abs(et$distance))]
  et[!duplicated(et$gene_name)]
}

selectBestTSS <- function(genes=NULL, peaks, ensdb, forceSeqlevelsStyle="chr"){
  all.txs <-  transcripts(ensdb, columns=c("gene_name"))
  if(!is.null(genes)) all.txs <- all.txs[which(all.txs$gene_name %in% genes)]
  x <- promoters(all.txs, upstream=1, downstream=1)
  if(!is.null(forceSeqlevelsStyle)){
    if(forceSeqlevelsStyle=="chr"){
      seqlevels(x) <- gsub("chrchr","chr",paste0("chr",seqlevels(x)))
    }else{
      seqlevelsStyle(x) <- forceSeqlevelsStyle
    }
  }
  x <- keepSeqlevels(x, seqlevelsInUse(x), pruning.mode = "coarse")
  d <- distanceToNearest(x, peaks)
  x$dist <- Inf
  x$dist[from(d)] <- mcols(d)$distance
  x <- x[order(abs(x$dist))]
  x <- x[!duplicated(x$gene_name)]
  sort(x)
}


```

## Fig 6c.1 (Q8).

Heatmap visualizing conditions dmso, untreated, 2h dex, KH103 >2h dex, MIF>2h dex, Cort113>2h dex, also show next to it GR tracks at TSS +/-2.5kb and at enhancer, the map needs to be clustered in 3 clusters– hopefully according to the genes that are not inhibited by a given inhibitor (message: KH103 if given before dex, blocks dex induced changes better than mif and cort113) (message 2: The genes that are not blocked cluster by inhibitor – we might be able to infer the way the inhibitor profile/specificity)

```{r plot_signals_and_sechm function, fig.width=8, fig.height=6}
#ah <- AnnotationHub(cache = "/common/AnnotationHub/", localHub = TRUE)
ah <- AnnotationHub()
ensdb <- ah[["AH89426"]]

peaks <- lapply(c("../data/reddy_bed_DEX2h_r123.bed.gz",
                  "../data/reddy_bed_DEX6h_r123.bed.gz"),
                FUN=function(x) import(x, format="NarrowPeak"))
peaks2h <- peaks[[1]]
peaks <- reduce(unlist(GRangesList(peaks)))
tss <- selectBestTSS(NULL, peaks, ensdb)
names(tss) <- tss$gene_name

d <- distanceToNearest(peaks2h, tss)
peaks2h$gene_name <- NA_character_
peaks2h$gene_name[from(d)] <- tss[to(d)]$gene_name



tracks <-  list("Control"="../data/single_replicate_peaks/control/Chip_GR_control_s3.bigWig",
                "30min-DEX"="../data/reddy_bed_DEX30min_r123.bigWig",
                "2h-DEX"="../data/reddy_bigwig_DEX2h_r123.bigWig",
                "6h-DEX"="../data/reddy_bigwig_DEX6h_r123.bigWig",
                "12h-DEX"="../data/reddy_bigwig_DEX12h_r123.bigWig")

nf <- bwNormFactors(tracks)


et <- read.delim("../data/enhancerTargets.tsv.gz")
et <- et[which(et$log2.pval.adapt>1),]
et <- GRanges(et$chr, IRanges(et$start, et$end), gene_name=et$symbol, distance=et$distance)
enh <- selectBestEnhancer(et, peaks)
names(enh) <- enh$gene_name

dea2h <- rowData(se)[["DEA.16h DMSO > 2h DEX+DMSO"]]
degs2h <- row.names(dea2h)[abs(dea2h$logFC)>log2(1.2) & dea2h$FDR<0.05]
degs2h <- intersect(degs2h[!is.na(degs2h)], intersect(names(tss),names(enh)))

cond_subset <- c("18h untreated", "18h DMSO", "16h DMSO >\n2h DEX+DMSO", "16h KH-103 >\n2h DEX+KH-103", "16h MIF  >\n2h DEX+MIF", "16h Cort113 >\n2h DEX+Cort113")
se2 <- se[,se$condition %in% cond_subset]


hmWrap <- function(x, i=1:3, se2, do.draw=FALSE, trim=c(0,0.99), ...){
  x <- intersect(x, intersect(names(enh),names(tss)))
  h1 <- plotSigAndRNA(enh[x], tracks=tracks[i], top_annotation=FALSE, SE=se2, ensdb = ensdb, widths=list(1,3), nf=nf[i], forceSeqlevelsStyle = "chr", trim=trim, extend=2500, rna_top_annotation=NA, gaps_at="condition", ..., retList=TRUE, axis_name_rot=90, scale_title="Enhancer")
  h2 <- plotSigAndRNA(tss[x], tracks=tracks[i], top_annotation=FALSE, SE=se2, ensdb = ensdb, widths=list(1,3), nf=nf[i], forceSeqlevelsStyle = "chr", axis_name="TSS", trim=trim, extend=2500, rna_top_annotation=NA, gaps_at="condition", ..., retList=TRUE, scale_title="TSS", axis_name_rot=90, breaks=0.99)
  h <- h1[[1]]+h2[[1]]+h2[[2]]
  if(do.draw!=TRUE) return(h)
  draw(h, padding = unit(c(1, 1, 2, 1), "cm"), merge=TRUE)  
}

h1 <- hmWrap(degs2h, se2=se2, k=4, row_title="Altered after 2h DEX", assayName = "log2FC")
pdf("../data/00_paper_figures/6c1.pdf", width=8, height=6)
draw(h1, padding = unit(c(0.2, 0.2, 2, 0.2), "cm"), merge=TRUE)
dev.off()
```


### Looking at peaks

```{r peak-centered, fig.width=8, fig.height=6}
plotSigAndRNA(peaks2h[which(peaks2h$gene_name %in% degs2h)], SE=se2, tracks=tracks[1:3], top_annotation=FALSE, k=3, row_title="Activated after 2h DEX (GR peaks)", nf=nf[1:3], widths = list(1,3), forceSeqlevelsStyle = "chr", axis_name="TSS", trim=c(0,0.98), extend=2500, rna_top_annotation=NA, gaps_at="condition")
```

Note that this is restricted to genes that are the closets gene to a GR peak, and showing signal at that peak.

## Fig 6d.1 (Q8). 

Heatmap visualizing conditions dmso, untreated, dex 18h, 2h dex>KH103, 2h dex>MIF, 2h dex>Cort113, also show next to it GRtracks at TSS +/-2.5kb and at enhancer, the map needs to be clustered in 3 clusters– hopefully according to the genes that are not inhibited by a given inhibitor (message1: KH103 if given after dex, blocks less good probably compared to mif and cort113) (message 2: The genes that are not blocked cluster by inhibitor – we might be able to infer the way the inhibitor profile/specificity)

```{r, fig.width=9, fig.height=6}
deas <- grep("^DEA\\.", colnames(rowData(se)), value=TRUE)
deas <- lapply(setNames(deas, gsub("^DEA\\.", "", deas)),
               FUN=function(x) rowData(se)[[x]])
degs <- lapply(deas, FUN=function(x){
  row.names(x)[which(x$FDR<0.05)]
})

dea16h <- deas$`2h DEX > 16h DEX+DMSO`
degs16h <- row.names(dea16h)[abs(dea16h$logFC)>log2(1.2) & dea16h$FDR<0.01]
degs16h <- degs16h[!is.na(degs16h)]

inhibDegs2 <- unique(intersect(degs16h, degs[["inhibited.after"]]))

cond_subset <- c("18h untreated", "18h DMSO", "2h DEX >\n16h DEX+DMSO", "2h DEX >\n16h DEX+KH-103", "2h DEX >\n16h DEX+MIF", "2h DEX >\n16h DEX+Cort113")
se2 <- se[,se$condition %in% cond_subset]

h <- hmWrap(degs16h, se2=se2, i=c(1,3:5), row_title="Activated after 12h DEX", k=4, assayName = "log2FC")

pdf("../data/00_paper_figures/6d1.pdf", width=9, height=6)
draw(h, padding = unit(c(0.2, 0.2, 2, 0.2), "cm"), merge=TRUE)
dev.off()
```

## Fig 7b.2 (Q8). 




```{r, fig.width=8, fig.height=6}
inhibDegs <- unique(c(intersect(degs2h, 
                                unlist(degs[c("inhibited by KH103.prior",
                                              "inhibited by others.prior")])),
                      unlist(degs[c("KHonly","otherInhibs")])))

cond_subset <- c("18h untreated", "18h DMSO", "16h DMSO >\n2h DEX+DMSO", "16h KH-103 >\n2h DEX+KH-103", "16h MIF  >\n2h DEX+MIF", "16h Cort113 >\n2h DEX+Cort113")
se2 <- se[,se$condition %in% cond_subset]


h <- hmWrap(inhibDegs, se2=se2, row_title="Different across inhibitors", k=3)

pdf("../data/00_paper_figures/7b1_differentAcrossInhibitors.pdf", width=8, height=6)
draw(h, padding = unit(c(0.2, 0.2, 2, 0.2), "cm"), merge=TRUE)
dev.off()
```



## Fig 7b.2 (Q8). 

Heatmap of those 13 genes that are significant in KH-103 alone condition comparing with untreated/dmso/Dex2h/Dex 18h (maybe also MIF/Cort113) alongside their TSS centered waterfall (TLS1 selected TSS) +- 2500 bp of the GR Chipseq of Reddy for the list (maybe if possible also including the long range enhancer info as well?). (message: a. these few genes that are changed in kh103 are also mostly changed in other inhibitor and dex conditions too, b. only 5 of those changed genes have direct GR mediation (except some have GR around enhancer) and c. marking up TEF and Cry2 as the only ones that are specifically changed only by KH103)



```{r, fig.width=9, fig.height=5}
cond_subset <- c("18h untreated", "18h DMSO", "18h KH-103",  "18h Cort113","18h MIF",
                 "16h DMSO >>\2h DEX+DMSO","2h DEX >\n16h DEX+DMSO")
se2 <- se[,se$condition %in% cond_subset]

idegs <- unique(unlist(degs[c("18h KH-103", "18h Cort113", "18h MIF")]))

hi <- hmWrap(idegs, se2=se2, i=c(1,3:5), row_title="Triggered by the inhibitors", k=4)


pdf("../data/00_paper_figures/fig7b2.pdf", width=8, height=6)
draw(hi, padding = unit(c(0.1, 0.1, 2, 0.1), "cm"), merge=TRUE)
dev.off()
draw(hi, padding = unit(c(0.1, 0.1, 2, 0.1), "cm"), merge=TRUE)


hk <- hmWrap(degs[["18h KH-103"]], se2=se2, i=c(1,3:5), row_title="Triggered by KH-103")
pdf("../data/00_paper_figures/fig7b1.pdf", width=8, height=5)
draw(hk, padding = unit(c(0.1, 0.1, 2, 0.1), "cm"), merge=TRUE)
dev.off()

```


