---
title: "Analysis"
author: "Dominique Paul"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

suppressPackageStartupMessages({
  library(ggplot2)
  library(biomaRt)
  library(dplyr)
  library(SummarizedExperiment)
  library(edgeR)
  library(SEtools)
  library(grid)
  library(VennDiagram)
  library(hrbrthemes)
  library(openxlsx)
  library(stringdist)
  library(rtracklayer)
  library(epiwraps)
})

source("../code/helper_functions.R")
```

# Question 1: Similarity of our RNA data with the Reddy data

**Q: Which gene expressions have similar patterns of up-regulation (control vs. Dex) in our data compared to the Reddy data?**

In our experiments we compared samples from a control experiment with samples that were treated for 18 hours with DEX. Besides a control, the Reddy lab performed a similar experiment where they treated cells with DEX for 12 hours. This is the closest comparison to our data.

*Planned approach:* Check which genes are differentially expressed in their data and create a Venn-diagram for comparable experiments with the differentially expressed genes from our analysis.

## Results

```{r extract r objects for Q1}
q1_data <- readRDS(file="../data/EOS_files/Q1_EOS.rds")
names(q1_data)

n_genes_plot <- dim(q1_data[["logFC_plot_data"]])[1]
n_genes_union <- length(q1_data[["union_DE_genes"]])
```

We first inspect a Venn diagram to understand the overlap of differentially expressed genes from both datasets. We see that a large part of the genes do overlap, while there are also many genes that are specific to the respective analyses.

![](assets/1_venn_diagram.png)

Second, we compare the log fold changes for the union of all genes identified as significantly differentially expressed. The union of all DE genes counts `r n_genes_union` genes of which `r n_genes_plot` genes (`r round(n_genes_plot/n_genes_union,2)`%) were also in the intersection of both datasets and could be plotted.

![](assets/2_logFC_Mahshid_Reddy.png)

# Question 2: Binding of GR-GC complex to DE genes from our analysis

**Q: For the genes that our RNA-seq data shows to be DE after treatment with DEX as well as the inhibitors and control: do we see any direct TF binding to these sites? For which of these is there a difference compared to the control ChIP-seq experiment?**

We see a significant change in terms of both the p-value and the logFC for genes that are bound by the GC-GR complex.

Plot: we select all genes that are differentially expressed - both for M = shid's data and for the DE analysis on the Reddy data. We then split the data into two groups based on whether the Reddy data showed that the glucorticoid receptor was found to bind near the gene or not. We plot the distributions of (a) the log fold changes in the DE analysis and (b) the p-value of the DE analysis. 

**Idea for next step:** Does it make sense to look at pathways of associated genes?

![](assets/Q2_3_mahshid_reddy.png)

# Question 3: Inhibition unique to protac compared to Cort and MIF {.tabset}

Are there any genes that treated by the Protac before/after DEX that could be reversed but which could not by the inhibitors or vice versa? 

We have performed experiments where we first exposed cells to either (1) DEX (2) inhibitors or (3) KH PROTAC for a substantial time and then added inhibitors or the KH PROTAC in case (1) or DEX in cases (2) and (3). We would like to see whether the genes which were activated by DEX but the returned to a normal expression by the KH PROTAC, where a similar treatment by the inhibitors could not lead to such an effect and vice versa.

## 2h > 16h

![](assets/Q3_1_reversal216.png)

## 2h > 16h (logFC > 0)

![](assets/Q3_2_reversal162_p1.png)

## 2h > 16h (logFC < 0)

![](assets/Q3_3_reversal162_p2.png)

Download excel file with data [here](assets/protact_flipped_DE.xlsx)

# Question 4: Does the GR bind close to any of the 13 DE genes from the 12h KH-PROTAC treatment?

We have 13 genes that are differentially expressed at an <0.05 FDR level. The Reddy lab has collected data on where the glucorticoid receptor (NR3C1) binds to after treatment with dexamethasone for 12h. We find that the GR is found to bind close to five out of the 13 differentially expressed genes.

```{r}
kh.protac <- read.xlsx("../output/Q4_KH_DE_GR_bound.xlsx")

# only keep the significant genes
kh.protac.sig <- kh.protac[kh.protac$FDR<0.05,]
kh.protac.sig[,c("Gene", "logFC", "FDR", "chip_peaks", "gc_bound")] %>%
  kableExtra::kable() %>%
  kableExtra::kable_styling()
```

![](assets/Q4_1_kh_gr_binding.png)

# Question 5: Binding to selected genes

**Q: Does GR bind to BRCA1, Hsd11b1 and CH25H?**

--> We see that there is one peak in the ChIP data close to the BRCA1 gene and two close to the HSD11B1 gene. There is no peak that is closest to the CH25H gene. However, the CD25H was also not found in the RNA seq data. 

```{r Q8}
res_int <- read.xlsx("../output/Q2_GR_bound_DE_genes.xlsx")
plt_data <- res_int[res_int$Gene %in% c("BRCA1", "HSD11B1", "CH25H"),c("Gene", "logFC", "FDR", "chip_peaks", "gc_bound")]
plt_data %>%
  ggplot() +
  geom_point(aes(logFC, -log10(FDR), colour=gc_bound, size=chip_peaks)) +
  labs(title="GR binding activity for diff. expressed PROTAC genes") +
  ggrepel::geom_text_repel(data=plt_data, aes(logFC, -log10(FDR), label=Gene, size=2)) +
  theme_ipsum_rc()


# CH25H was not found. Is it in the RNA-seq experiment?
se <- readRDS("../data/EOS_files/Q1_SummExp.rds")
"ENSG00000138135" %in% rowData(se)$gene_id

# We also have to check the other way around by looking at all the peaks.
# If it was in the peaks then it could not be merged onto the RNA data
peakfreq <- readRDS(file="../data/EOS_files/Q2_gene_peaks.rds")
peakfreq <- peakfreq[2:dim(peakfreq)[1],]

# we search for the closest match to be sure that its not due to capitalisation or a small typo
peakfreq$Gene[amatch("CH25H", peakfreq$Gene, maxDist = Inf)]
# --> No match for CH25h

# show results
# peakfreq[peakfreq$Gene %in% c("BRCA1", "HSD11B1", "CH25H"),] %>%
#   kableExtra::kable() %>%
#   kableExtra::kable_styling()

plt_data %>%
  kableExtra::kable() %>%
  kableExtra::kable_styling()
```




# Question 6: Common binding sites of JUNB and GR

**Q: What are common binding sites between JUNB and GR?** -> Generally not a priority though

**Take-aways:**

- Interestingly there are more peaks after 2h treatment with DEX than 12h of treatment with DEX
- Most of the 12h GR peaks are shared with the JUNB peaks, only a small fraction is unique to the 12h GR ChIP seq data
- The 2h GR data has more peaks that are unique and not shared with the 12h GR or 12h JUNB data

```{r q6}
# Load GR
peaks_gr_12h <- import.bed15("../data/reddy_bed_DEX12h_r123.bed", format="narrowPeak")
peaks_gr_2h <- import.bed15("../data/reddy_bed_DEX2h_r123.bed.gz", format="narrowPeak")
# load JUNB
peaks_junb_12h <- import.bed15("../data/reddy_bed_JUNB_DEX12h_r123.bed", format="narrowPeak")

# Show tile diagram
list_of_regions <- list(GR_2h=peaks_gr_2h, GR_12h=peaks_gr_12h, JUNB_12h=peaks_junb_12h) # Does not work with only two
regionOverlaps(list_of_regions)

# Show upset plot
regionUpset(list_of_regions)
```

### What if we filter the peaks by their score?

There is no significant change in the results/plots if we filter for a minimum peak score of 900 (max value is 1000).

```{r q6 filtered}
# Does it make a difference if we only choose peaks with high scores?
## What is the range of scores
# qplot(peaks_gr_2h$score, geom="histogram")

# Show tile diagram
list_of_regions_v2 <- list(GR_2h=peaks_gr_2h[peaks_gr_2h$score>900], 
                           GR_12h=peaks_gr_12h[peaks_gr_12h$score>900], 
                           JUNB_12h=peaks_junb_12h[peaks_junb_12h$score>900]) # Does not work with only two 
regionOverlaps(list_of_regions_v2)

# Show upset plot
regionUpset(list_of_regions_v2)
```

# Question 7: PXR Sequences

**Q: Does GC-GR bind to PXR sequences?**

Other groups have observed that a GR inhibitor activates CYP genes. CYP genes are target genes of PXR TF that have PXR recognition sites. the hypothesis here is that GR might also bind to PXR sites directly or probably is regulating CYP genes via interaction with PXR TF on these genes. For example, based on our RNA-seq dex triggers many CYP genes and the inhibitors too, but protac doesn't.

The PXR motif looks like this (from Hocomoco):

![](assets/Q7_1_motif.png)


```{r}
gr_peaks <- readRDS("../data/EOS_files/Q2_annotated_peaks.rds")
moi <- readRDS(file="../data/EOS_files/pxr_peaks.rds")

cat(paste0("We have ", length(gr_peaks), " GR peaks, of which ", length(moi), " (", round(length(moi)/length(gr_peaks),4)*100, "%) contain a PXR motif"))
```

### We have `r length(gr_peaks)` GR peaks, of which `r length(moi)` (`r round(length(moi)/length(gr_peaks),4)*100`%) contain a PXR motif

We can also look at the signals around the PXR motifs found in the accessible chromatin (identified using the 12h ATAC seq data from the Reddy lab)

![](assets/Q7_2_signal.png)










#
