---
title: "About"
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# todo lecture presentation

Q7: Scan open areas (ATAC-seq) for PXR binding (NR1I2). How many of those are bound by GR?

Create boxplots:
  x: number of peaks
  y: distribution of logFCs
  
Plot signals for DE genes with BED data

16 dmso + dex -> could look at that

# Analysing the use of PROTACs to target glucorticoid receptors

## Background

### Glucorticoids

Glucorticoids (a portmanteau of glucose + cortex + steroids; abbreviated GC) are a form of steroids which play an important role as a feedback mechanism in the human immune system. They are generated in the adrenal cortex and are transported throughout the body where they diffuse through the cell membrane of the cell and bind to glucorticoid receptors (GR) which reside in the cytosol. These activated GC-GR complexes can either indirectly or directly influence gene expression. Direct regulation involves homo-dimerisation of the receptor which then acts as a transcription factor (TF) in the nucleus to regulate gene expression (transactivation). In the absence of the GC-GR complex other TFs are able to function normally However, when GR is activated (in form of a GC-GR complex) it can interact with these TFs and hinder them from binding to DNA (transrepression). The GC-GR complex is particularly known for up-regulating the expression of anti-inflammatory proteins and repression of pro-inflammatory genes. 

Cortisol, a stress hormone, is the best-known glucorticoid and also the most important one. It is essential for many cardiovascular, metabolic, immunologic and homeostatic functions.

### PROTACs

Proteolysis targeting chimeras (PROTACs) are small molecules with two active domains bound together by a so-called linker domain. They are able of catalysing the targeted degradation of proteins. One of the two active domains binds the protein of interest. While inhibitors block the functioning of proteins by binding to the active domain, PROTACs can instead bind to any part of the protein. PROTACs are thus easier to construct in the sense that the active domain can have many forms and need fit the active domain of the target. The other active domain of the PROTAC binds to a protein (ER ubiquitin ligase) which is responsible for adding ubiquitin molecules to proteins, which are read by the cell as marks for destruction. Proteins marked for destruction eventually bind to the proteasome which subsequently destroys said proteins.

Often, to block the functioning of proteins we use inhibitors in biology. However, inhibitors have to bind to the active site of a protein to work, while PROTACs only need to bind to the targeted protein with high selectivity rather than fully block the active domain, as mentioned above. Furthermore, one inhibitor has to stay bound to the targeted protein to block its function. To fully block the protein from functioning, we hence would need at least as many inhibitors as there are proteins that we want to inhibit.  PROTACs on the other hand can direct the destruction of multiple proteins before they themselves are degraded. PROTACs can thus be administered at lower dosages compared to inhibitors to achieve similar effects.

Being administerable at a lower dosage, being more selective and not necessarily having to bind to the active domain makes PROTACs attractive drug candidates.


## What is the overall research goal of the project?

The goal of the project is to understand (1) the contribution of GR to the transcriptional stress response in the hippocampus in vivo and (2) show that PROTACs can be used in-vivo as a tool in neuroscience to normalise stress response.

## What has been done so far?

Katharina and Mahshid have performed experiments on mice where they treated mice with combinations of (1) a synthetic form of a GC called Dexamethasone (DEX) (2) a PROTAC inhibitor (KH-103) and (3) GR inhibitors. They have collected RNA-seq data to understand what the effects of treatment are - which was analysed by Pierre-Luc - and have determined differentially expressed genes.

[TODO: Part on PL's analysis]

## What is my contribution?

The [Tim Reddy lab](http://reddylab.org/) has done related research and has performed RNA-seq, ChIP-seq, and ATAC-seq experiments on the same cell line and under similar conditions. While we only have RNA-seq data, the Reddy lab has ChIP-seq experiments for two proteins, one of which being the GR, and DNA accessibility data. 

This subanalysis of the project will analyse the data from the Reddy lab to complement and extend findings from our own experiments.

## What data will I be working with?

### Our data

Mahshid and Katharina have collected RNA-seq data for the following treatment conditions of 

**Control treatments: 18h treatment with inhibitor or PROTAC only**

1. Inhibitor 1: "18h Cort113"
1. Inhibitor 2: "18h MIF"
1. PROTAC only: "18h KH-103""16h DMSO > 2h DEX+DMSO"

**Switched treatments: 16h+2h** 

We first treat with the inhibitor, PROTAC or DMSO as control and the add DEX for the last two hours

1. Inhibitor 1: "16h Cort113 > 2h DEX+Cort113"
1. Inhibitor 2: "16h MIF  > 2h DEX+MIF"
1. PROTAC: "16h KH-103 > 2h DEX+KH-103"
1. Control with DMSO: 16h DMSO > 2h DEX+DMSO"

**Switched Treatment: 2h+16h**

We first treat with DEX for 2 hours and then add the inhibitor PROTAC for another 16 hours

1. Inhibitor 1 "2h DEX > 16h DEX+Cort113"
1. Inhibitor 2: "2h DEX > 16h DEX+MIF"
1. PROTAC: "2h DEX > 16h DEX+KH-103"
1. DEX only: "2h DEX > 16h DEX+DMSO"


### Reddy lab data

The experiments from the Reddy lab include:

1. **RNA - DEX vs control:** One RNA-seq experiment with a control group and dexamethasone treatment for multiple time intervals, up to 12 hours
1. **ChIP - NR3C1 & JUNB:** Two ChIP-seq experiments. One is for NR3C1 - which is as I understand is the only relevant (?) GR in the cell. The second ChIP-seq experiment is for JUNB. JUNB appears to be a transcription factor subunit.
1. **ATAC-seq - DEX exposure:** experiment with a control as well as data for exposure of the cell culture to DEX for multiple intervals.


## What will the first questions be that I will be analysing?

### Question 1: Similarity of our RNA data with the Reddy data

(DONE)  Q: Which gene expressions have similar patterns of up-regulation (control vs. Dex) in our data compared to the Reddy data? 

*Planned approach:* Check which genes are differentially expressed in the Reddy data and create a Venn-diagram for comparable experiments with the differentially expressed genes from our analysis. Also, create a plot

### Question 2: Binding of GR-GC complex to DE genes from our analysis

(Discuss in call) Q: For the genes that our RNA-seq data shows to be DE after treatment with DEX as well as the inhibitors and control: do we see any direct TF binding to these sites? For which of these is there a difference compared to the control ChIP-seq experiment?

### Question 3: For our RNA: are there any genes that treated by the PROTAC before/after DEX that could be reversed but which could not by the inhibitors or vice versa?

(DONE) We have performed experiments where we first exposed cells to either (1) DEX (2) inhibitors or (3) KH PROTAC for a substantial time and then added inhibitors or the KH PROTAC in case (1) or DEX in cases (2) and (3). We would like to see whether the genes which were activated by DEX but the returned to a normal expression by the KH PROTAC, where a similar treatment by the inhibitors could not lead to such an effect and vice versa.

### Question 4: Binding of GR to KH PROTAC treated DE genes?

(DONE) Q: Is there a direct binding of GR to the 13 genes shown to be affected by the KH PROTAC?

### Question 5: Binding to selected genes

(DONE)Q: Does GR bind to BRCA1, Hsd11b1 and CH25H?

### Question 6: Common binding sites of JUNB and GR

Q: What are common binding sites between JUNB and GR? -> Generally not a priority though

### Question 7: PXR Sequences

Q: Does GC-GR bind to PXR sequences

Scan for PXR sequences. See if GR binds there




## Abbreviations

GC = Glucorticoid

GR = Glucorticoid receptor

PROTAC = Proteolysis Targeting Chimera

DEX = Dexamethasone

TF = Transcription Factor






