---
title: 'Q4: GR-binding to PROTAC genes'
author: "Dominique Paul"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

suppressPackageStartupMessages({
  library(openxlsx)
  library(dplyr)
  library(ggplot2)
  library(hrbrthemes)
  library(epiwraps)
  library(AnnotationHub)
})

hrbrthemes::import_roboto_condensed()
extrafont::loadfonts()
```

### Question 4: Binding of GR to KH PROTAC treated DE genes?

Q: Is there a direct binding of GR to the 13 genes shown to be affected by the KH PROTAC?

**Approach:** 

1. What are the 13 PROTAC genes?
1. Does GR bind to them?


```{r load data}
# load the peaks
peakAnno <- readRDS("../data/EOS_files/Q2_annotated_peaks.rds")
# load our DE results
res_int_all <- readRDS(file="../data/results/DE_our_data.rds")

# get the data for the 18h KH-PROTAC treatment
kh.protac <- res_int_all[["18h KH-103"]]

# load the results from the peak frequencies
peakfreq <- readRDS(file="../data/EOS_files/Q2_gene_peaks.rds")

# merge peak frequency onto table and add binary variable
peakAnno_reduced <- as.data.frame(peakAnno) %>% 
  dplyr::select("nearestTSS.gene_name", "distance2nearestTSS", "class") %>% 
  dplyr::rename(Gene=nearestTSS.gene_name) %>%
  dplyr::filter(Gene != "") %>%
  group_by(Gene) %>%
  dplyr::slice(which.min(abs(distance2nearestTSS)))

kh.protac <- left_join(kh.protac %>% mutate(Gene = rownames(kh.protac)), 
          peakfreq,
          by="Gene") %>%
  tidyr::replace_na(list(Freq=0)) %>%
  mutate(gc_bound = Freq > 0) %>%
  dplyr::rename(chip_peaks=Freq) %>% 
  dplyr::select(Gene, everything()) %>%
  dplyr::left_join(peakAnno_reduced, by="Gene")

write.xlsx(file="../docs/downloads/Q4_KH_DE_GR_bound.xlsx", kh.protac, rowNames=TRUE)
```

### Does GR-GC bind to a region closely associated with any of these 13 genes?

We have 13 genes that are differentially expressed at an <0.05 FDR level. The Reddy lab has analysed where the glucorticoid receptor (NR3C1) binds to after treatment with dexamethasone for 12h. We find that the GR is found to bind close to 5 out of the 13 differentially expressed genes.

```{r keep significant genes}
# only keep the significant genes
kh.protac.sig <- kh.protac[kh.protac$FDR<0.05,]
kh.protac.sig[,c("Gene", "logFC", "FDR", "chip_peaks", "gc_bound")]
```

```{r plot}
plt_gr_binding <- ggplot(kh.protac.sig) +
  geom_point(aes(logFC, -log10(FDR), colour=gc_bound, size=chip_peaks)) +
  labs(title="GR binding activity for diff. expressed PROTAC genes") +
  ggrepel::geom_text_repel(data=kh.protac.sig, aes(logFC, -log10(FDR), label=Gene, size=1)) +
  theme_ipsum_rc()

plt_gr_binding

ggsave(file="../docs/assets/Q4_1_kh_gr_binding.pdf", plt_gr_binding)
```

# Plot signal tracks 

## Peaks found in bed file (not all 13 genes included)

Visualisations are centred around peaks, but annotated by the closest gene. The visualised tracks are NOT centred around the respective genes' TSS.

```{r plot signal tracks, fig.width=12, fig.height=7}
peaks13g <- peakAnno[peakAnno$nearestTSS.gene_name %in% kh.protac.sig$Gene,]

names(peaks13g) <- make.unique(peaks13g$nearestTSS.gene_name)

tracks <-  list("2h-DEX"="../data/reddy_bigwig_DEX2h_r123.bigWig",
                "6h-DEX"="../data/reddy_bigwig_DEX6h_r123.bigWig",
                "12h-DEX"="../data/reddy_bigwig_DEX12h_r123.bigWig")
signals <- signal2Matrix(tracks, peaks13g, w=5, extend=500)
```

```{r plot signal tracks 2, fig.width=12, fig.height=7}
h1 <- plotEnrichedHeatmaps(signals, show_row_names=TRUE, row_names_side="left", row_order=order(rownames(signals[["2h-DEX"]]))) #, left_annotation=peaks13g$nearestTSS.gene_name)#, width=unit(12, "cm"))
h1

ggsave(file="../docs/assets/Q4_2_kh_peak_tracks.pdf", width=12,height=7)
```

## Plot the signal around the TSS of each of the 13 genes (all transcripts)

± 10'000 bps from TSS

```{r gene TSS all transcripts, fig.width=12, fig.height=7}
ah <- AnnotationHub()
ensdb <- ah[["AH98047"]] # AH89211

mygenes.transcripts <-  subset(transcripts(ensdb, columns=c("tx_id", "tx_name", "tx_biotype", "tx_support_level", "gene_id","gene_name")), gene_name %in% kh.protac.sig$Gene)
mygenes.tss <-  resize(mygenes.transcripts, width=2000, fix='start')
seqlevelsStyle(mygenes.tss) <- "UCSC"

signals2 <- signal2Matrix(tracks, mygenes.tss, w=10, extend=10000)
h2 <- plotEnrichedHeatmaps(signals2) #, left_annotation=peaks13g$nearestTSS.gene_name)#, width=unit(12, "cm"))
h2

ggsave(file="../docs/assets/Q4_3_kh_all_txs.pdf", width=12,height=7)
```


## Plot the signal around the TSS of each of the 13 genes (one transcript per gene)

± 10'000 bps from TSS

```{r all genes unique TSS, fig.width=12, fig.height=7}
mygenes_unique <- mygenes.tss[!duplicated(mygenes.tss$gene_name),]

names(mygenes_unique) <- mygenes_unique$gene_name
signals3 <- signal2Matrix(tracks, mygenes_unique, w=10, extend=10000)

h3 <- plotEnrichedHeatmaps(signals3, show_row_names=TRUE, row_names_side="left", row_order=order(rownames(signals3[["2h-DEX"]]))) 
h3

ggsave(file="../docs/assets/Q4_4_kh_unique_gene_tss.pdf", width=12,height=7)
```

