---
title: 'Q7: PRX sequences'
author: "Dominique Paul"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

suppressPackageStartupMessages({
  library(rtracklayer)
  library(epiwraps)
  library(GenomicRanges)
  library(memes)
  library(Biostrings)
  library(AnnotationHub)
  library(MotifDb)
  library(BSgenome)
  library(universalmotif)
})
```

# Question 7: Does GC-GR bind to PRX sequences?

### Approach

Q7: Scan open areas (ATAC-seq) for PXR binding (NR1I2). How many of those are bound by GR?

1. Scan for PRX sequences. 
2. Is there an overlap between the peaks of GR and the PRX sequences?

```{r load peaks}
peaks <- import.bed15("../data/reddy_bed_ATAC_control.bed.gz", format="narrowPeak")

# change from chr (ensembl format) to just numbers
# seqlevels(peaks)<- sub('chr','',seqlevels(peaks))
```


```{r scan peaks for PXR sequence}
peak_centers <- resize(peaks, fix="center", width=100)
# we get the genome sequence from a download from the UCSC, human genome wasnt found in search in annotation hub
genome <- BSgenome::getBSgenome("BSgenome.Hsapiens.UCSC.hg38", masked=FALSE, load.only=FALSE)
# we get the sequences corresponding to the peak centers:
peak_seqs <- memes::get_sequence(peak_centers, genome)
peak_seqs[1:1000]
```


## Get the motif for PXR (NR1I2)

```{r}
# we search for "CTCF" in the motif database
motifs <- query(MotifDb, "NR1I2")
# there are several matching motifs:
names(motifs)
# we select one:
motif1 <- motifs[["Hsapiens-HOCOMOCOv11-core-C-NR1I2_HUMAN.H11MO.0.C"]]
# we visualize it:
view_motifs(motif)
```


## Scanning a set of sequences using the Meme suite

```{r}
moi <- memes::runFimo(peak_seqs[1:1000],  convert_motifs(motif))
# you might eventually need to add the meme_path argument:
moi <- memes::runFimo(peak_seqs,  convert_motifs(motif), meme_path = "/home/pierre-luc/bin/")
```
